---
title: "Nematostella TOMOseq"
output: 
  BiocStyle::html_document:
        toc: true
        toc_float: true
        highlight: tango
        code_folding: show
  BiocStyle::pdf_document:
        toc: true
        highlight: tango
---

```{r setup working dict}


setwd("/g/arendt/gerber/Analyses/NematostellaTomoSeq/")

```

```{r load libaries, warning=FALSE, message=FALSE, results="hide"}
suppressPackageStartupMessages({
  library(Seurat)
  library(RColorBrewer)
  library(cowplot)
  library(viridis)
  library(patchwork)
  library(future)
  library(tidyverse)
  library(plyr)
  })

```

```{r setup parallelization}
library(future)
#parallize for scaling
plan("multiprocess", workers = 45)
plan()

#set maximum to 50GB for each worker
options(future.globals.maxSize = 50000 * 1024^2)
```

#Load TOMO data

```{r}
tomo = readRDS("/g/arendt/gerber/Analyses/NematostellaTomoSeq/tomoseq_animals_seurat.rds")
```

#All slices

```{r}
#remove bad samples
#b3-MKK-1-uncut
#b3-MTJ-47-48hpa
#b3-MT2-39-24hpa
#b1 all

select.cells = colnames(tomo)[!tomo@meta.data$sample_name %in% c("b3-MKK-1-uncut","b3-MTJ-47-48hpa","b3-MT2-39-24hpa","b1-1-uncut","b1-2-uncut")]

tomo = subset(tomo, cells = select.cells)




##############

#parallize scaling
library(future)
plan("multiprocess", workers = 1)
plan()

#set maximum to 10GB for each worker
options(future.globals.maxSize = 10000 * 1024^2)

all.genes <- rownames(tomo)
tomo = ScaleData(  tomo, features = all.genes ,vars.to.regress = c("nFeature_RNA"))

#run PCA on all genes
#RunPCA(object, assay = NULL, features = NULL,  npcs = 50, rev.pca = FALSE, weight.by.var = TRUE, verbose = TRUE,  ndims.print = 1:5, nfeatures.print = 30, reduction.name = "pca",  reduction.key = "PC_", seed.use = 42)

tomo = RunPCA(tomo,features = all.genes,npcs = 50)

#plot PCA heatmaps
#DimHeatmap(pbmc, dims = 1, cells = 500, balanced = TRUE)

library(RColorBrewer)
library(ggplot2)

pdf("tomo_PCAheatmaps.pdf",width=30,height=20)
DimHeatmap(tomo, dims = 1:16, cells = 20, balanced = TRUE, ncol = 4,  fast = F) 
DimHeatmap(tomo, dims = 17:32, cells = 200, balanced = TRUE, ncol = 4,  fast = F) 
DimHeatmap(tomo, dims = 33:48, cells = 200, balanced = TRUE,  ncol = 4, fast = F) 
dev.off()

#elbow plot
pdf("tomo_PCelbow.pdf",width=20,height=10)
ElbowPlot(tomo, ndims = 50)
dev.off()

#BSS1[["spatial"]] <- CreateDimReducObject(embeddings = dim,  key = "Spatial_", assay = "RNA")

#Cluster cells
tomo<- FindNeighbors(tomo, dims = 1:30)
tomo <- FindClusters(tomo, resolution = 0.6)


# Perform Hierarchical Clustering
library(tidyverse)
library(factoextra)
Data.hc <- tomo@reductions$pca@cell.embeddings[,c(1:30)] %>% get_dist(method = "pearson") %>% hclust(method = 'ward.D') ## PCs
Hcls <- data.frame(Clusters = as.factor(cutree(Data.hc, k = 10)))
Hcls <- Hcls %>% mutate(Wells = colnames(tomo))

tomo@meta.data$Hclust = Hcls$Clusters

tomo = RunUMAP(tomo, dims = 1:30,min.dist = 0.03)

#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("tomo_Embedding_PC30_res0.6.pdf",width=6.8,height=5)
DimPlot(object = tomo, reduction = 'umap', pt.size = 1)
DimPlot(object = tomo, reduction = 'umap', pt.size = 1, group.by = "sequencing_sample_id")
DimPlot(object = tomo, reduction = 'umap', pt.size = 1, group.by = "batch")
DimPlot(object = tomo, reduction = 'umap', pt.size = 1, group.by = "sample_name")
FeaturePlot(tomo, reduction = 'umap', pt.size = 1, features = c("nFeature_RNA","nCount_RNA"),order = F, cols = c(brewer.pal(9,"Greys")[3],brewer.pal(9,"Blues")[3:9]))
dev.off()



saveRDS(tomo,"./tomo.RDS")
```

###Harmony

```{r}
library(harmony)

tomo.harm = RunHarmony(object = tomo,group.by.vars= "sample_name", assay.use = "RNA", dims.use = 1:30, plot_convergence = TRUE, max.iter.cluster = 20 ,max.iter.harmony=50,theta=0.1,lambda=0.01,seed=TRUE)
#Default lambda=1
#Default theta=2

#Cluster cells
tomo.harm<- FindNeighbors(tomo.harm, dims = c(2:30), reduction = "harmony")
tomo.harm <- FindClusters(tomo.harm, resolution = 0.3)

#run UMAP
tomo.harm = RunUMAP(tomo.harm,dims = c(2:30), reduction = "harmony", min.dist = 0.03)
#Default min.dist=0.3

library(tidyverse)
library(factoextra)
Data.hc <- tomo.harm@reductions$harmony@cell.embeddings[,c(2:30)] %>% get_dist(method = "pearson") %>% hclust(method = 'ward.D') ## PCs
Hcls <- data.frame(Clusters = as.factor(cutree(Data.hc, k = 9)))
Hcls <- Hcls %>% mutate(Wells = colnames(tomo.harm))

tomo.harm@meta.data$Hclust = Hcls$Clusters


mitosis <- list(intersect(mitosis[[1]], rownames(tomo.harm)))
tomo.harm <- AddModuleScore(
  object = tomo.harm,
  features = mitosis,
  name = 'mitosis'
)



tomo.harm@meta.data$slice_sum[tomo.harm@meta.data$slice_index %in% c(1,2,3,4,5,6,7,8,9,10)] = "slice10"
tomo.harm@meta.data$slice_sum[tomo.harm@meta.data$slice_index %in% c(11,12,13,14,15,16,17,18,19,20)] = "slice20"
tomo.harm@meta.data$slice_sum[tomo.harm@meta.data$slice_index %in% c(21,22,23,24,25,26,27,28,29,30)] = "slice30"
tomo.harm@meta.data$slice_sum[tomo.harm@meta.data$slice_index %in% c(31,32,33,34,35,36,37,38,39,40)] = "slice40"
tomo.harm@meta.data$slice_sum[tomo.harm@meta.data$slice_index %in% c(41,42,43,44,45,46,47,48,49,50)] = "slice50"
tomo.harm@meta.data$slice_sum[tomo.harm@meta.data$slice_index %in% c(51,52,53,54,55,56,57,58,59,60)] = "slice60"
tomo.harm@meta.data$slice_sum[tomo.harm@meta.data$slice_index %in% c(61,62,63,64,65,66,67,68,69,70)] = "slice70"
tomo.harm@meta.data$slice_sum[tomo.harm@meta.data$slice_index %in% c(71,72,73,74,75,76,77,78,79,80)] = "slice80"
tomo.harm@meta.data$slice_sum[tomo.harm@meta.data$slice_index %in% c(81,82,83,84,85,86,87,88,89,90)] = "slice90"
tomo.harm@meta.data$slice_sum[tomo.harm@meta.data$slice_index %in% c(91,92,93,94,95,96)] = "slice96"

tomo.harm@meta.data$cut_info = "cut"
tomo.harm@meta.data$cut_info[tomo.harm@meta.data$sample_name %in% tomo.harm@meta.data$sample_name[grep("uncut",tomo.harm@meta.data$sample_name)]] = "uncut"

#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

my_spectral_palette = colorRampPalette(colors = brewer.pal(9, "Set1"))

pdf("tomo_Harmony_Embedding_PC2_30_res0.3.pdf",width=6,height=5)
DimPlot(object = tomo.harm,label = T,shuffle=T, reduction = 'umap', pt.size = 2)
DimPlot(object = tomo.harm, label = F,shuffle=T,reduction = 'umap', pt.size = 2, group.by = "seurat_clusters",cols = c("darkgoldenrod3","red3","salmon","turquoise4","dodgerblue"))
DimPlot(object = tomo.harm, label = F,shuffle=T,reduction = 'umap', pt.size = 2, group.by = "Hclust")
DimPlot(object = tomo.harm, label = F,shuffle=T,reduction = 'umap', pt.size = 2, group.by = "sequencing_sample_id")
DimPlot(object = tomo.harm, label = F,shuffle=T,reduction = 'umap', pt.size = 2, group.by = "batch")
DimPlot(object = tomo.harm,label = F,shuffle=T, reduction = 'umap',group.by = "sample_name", pt.size = 2)
DimPlot(object = tomo.harm,label = F,shuffle=T, reduction = 'umap',group.by = "time_point", pt.size = 2)
DimPlot(object = tomo.harm,label = F,shuffle=T, reduction = 'umap',group.by = "slice_index", pt.size = 2)
DimPlot(object = tomo.harm,label = F,shuffle=T, reduction = 'umap',group.by = "slice_sum", pt.size = 2,cols = viridis_pal(option = "magma")(10))
DimPlot(object = tomo.harm,label = F,shuffle=T, reduction = 'umap',group.by = "cut_info", pt.size = 2)
FeaturePlot(tomo.harm, reduction = 'umap', pt.size = 1, features = c("nCount_RNA","nFeature_RNA","mitosis1"),order = T, cols = c("gray",rev(viridis_pal(option = "inferno")(10))))
dev.off()

pdf("tomo_Harmony_Feature_Mitosis.pdf",width=20,height=5)
FeaturePlot(tomo.harm, reduction = 'umap', pt.size = 2, features = c("mitosis1"),order = T,split.by = "time_point", cols = c("gray",rev(viridis_pal(option = "viridis")(10))))
dev.off()



pdf("test.pdf",width=6,height=5)
DimPlot(object = tomo.harm,label = T,shuffle=T, reduction = 'umap', pt.size = 2)
DimPlot(object = tomo.harm, label = F,shuffle=T,reduction = 'umap', pt.size = 2, group.by = "sequencing_sample_id")
DimPlot(object = tomo.harm, label = F,shuffle=T,reduction = 'umap', pt.size = 2, group.by = "batch")
DimPlot(object = tomo.harm,label = F,shuffle=T, reduction = 'umap',group.by = "sample_name", pt.size = 2)
DimPlot(object = tomo.harm,label = F,shuffle=T, reduction = 'umap',group.by = "time_point", pt.size = 2)
DimPlot(object = tomo.harm,label = F,shuffle=T, reduction = 'umap',group.by = "slice_index", pt.size = 2)
DimPlot(object = tomo.harm,label = F,shuffle=T, reduction = 'umap',group.by = "slice_sum", pt.size = 2,cols = viridis_pal(option = "magma")(10))
DimPlot(object = tomo.harm,label = F,shuffle=T, reduction = 'umap',group.by = "cut_info", pt.size = 2)
FeaturePlot(tomo.harm, reduction = 'umap', pt.size = 1, features = c("nCount_RNA","nFeature_RNA"),order = T, cols = c("gray",rev(viridis_pal(option = "inferno")(10))))
dev.off()

#plot counts as violin

pdf("tomo_Harmony_Vln_nCounts.pdf",width=5,height=5)
VlnPlot(tomo.harm, c("nCount_RNA","nFeature_RNA"), group.by = "time_point",pt.size = 0, cols = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))
dev.off()

saveRDS(tomo.harm,"./tomo_Harmony.RDS")


```

###Pseudotime

```{r}
######Calc pseudo time


harmony.embedding = Embeddings(tomo.harm, "harmony")

#make scaled values positive
for (i in 1:ncol(harmony.embedding)) {
  harmony.embedding[,i] = harmony.embedding[,i] + abs(min(harmony.embedding[,i]))
}

harmony.embedding = as.data.frame(harmony.embedding,stringsAsFactor = F)

meta.data = FetchData(object = tomo.harm, vars = c('seurat_clusters', 'orig.ident','UMAP_1','UMAP_2',"sequencing_sample_id","batch","sample_name","time_point","slice_index","slice_sum"))


#destiny

library("destiny")
library("Rcpp")


cells_bottom = rownames(meta.data)[meta.data$seurat_clusters %in% c(1)]
cells_center1 = rownames(meta.data)[meta.data$seurat_clusters %in% c(0)]
cells_center2 = rownames(meta.data)[meta.data$seurat_clusters %in% c(3,2)]
cells_top = rownames(meta.data)[meta.data$seurat_clusters %in% c(4)]


#cells_bottom = rownames(meta.data)[meta.data$UMAP_1 > 2]
#cells_center1 = rownames(meta.data)[meta.data$UMAP_1 < 2 & meta.data$UMAP_1 > (-2)]
#cells_center2 = rownames(meta.data)[meta.data$UMAP_1 < (-2) & meta.data$UMAP_2 < 0]
#cells_top = rownames(meta.data)[meta.data$UMAP_1 < (-2) & meta.data$UMAP_2 > 0 ]


#calculate pseudo time separate on branches

harmony.embedding.bottom = as.data.frame(harmony.embedding[cells_bottom,],stringsAsFactors = F)
harmony.embedding.center1 = as.data.frame(harmony.embedding[cells_center1,],stringsAsFactors = F)
harmony.embedding.center2 = as.data.frame(harmony.embedding[cells_center2,],stringsAsFactors = F)
harmony.embedding.top = as.data.frame(harmony.embedding[cells_top,],stringsAsFactors = F)

library("destiny")

#bottom
DatasetDM <- DiffusionMap(harmony.embedding.bottom[,2:30])

dpt = DPT(DatasetDM )

pseudotime_bottom = dpt[random_root(dpt), ]

harmony.embedding.bottom$pt = as.numeric(pseudotime_bottom)
harmony.embedding.bottom = harmony.embedding.bottom[order(harmony.embedding.bottom$pt),]

harmony.embedding.bottom$pt_adj = rev(harmony.embedding.bottom$pt)
harmony.embedding.bottom$pt_adj = harmony.embedding.bottom$pt_adj

harmony.embedding.bottom$rank = rev(1:nrow(harmony.embedding.bottom))


#center1
DatasetDM <- DiffusionMap(harmony.embedding.center1[,2:30])

dpt = DPT(DatasetDM )

pseudotime_center1 = dpt[random_root(dpt), ]

harmony.embedding.center1$pt = as.numeric(pseudotime_center1)
harmony.embedding.center1 = harmony.embedding.center1[order(harmony.embedding.center1$pt),]

harmony.embedding.center1$pt_adj = (harmony.embedding.center1$pt)
harmony.embedding.center1$pt_adj = harmony.embedding.center1$pt_adj+  max(harmony.embedding.bottom$pt_adj)


harmony.embedding.center1$rank = (1:nrow(harmony.embedding.center1))
harmony.embedding.center1$rank = harmony.embedding.center1$rank  + nrow(harmony.embedding.bottom)


#center2
DatasetDM <- DiffusionMap(harmony.embedding.center2[,2:30])

dpt = DPT(DatasetDM )

pseudotime_center2 = dpt[random_root(dpt), ]

harmony.embedding.center2$pt = as.numeric(pseudotime_center2)
harmony.embedding.center2 = harmony.embedding.center2[order(harmony.embedding.center2$pt),]

harmony.embedding.center2$pt_adj = rev(harmony.embedding.center2$pt)
harmony.embedding.center2$pt_adj = harmony.embedding.center2$pt_adj + max(harmony.embedding.bottom$pt_adj) + max(harmony.embedding.center1$pt_adj)
harmony.embedding.center2$rank = rev(1:nrow(harmony.embedding.center2))

harmony.embedding.center2$rank = harmony.embedding.center2$rank  + nrow(harmony.embedding.bottom) + nrow(harmony.embedding.center1)

#top
DatasetDM <- DiffusionMap(harmony.embedding.top[,2:30])

dpt = DPT(DatasetDM)

pseudotime_top = dpt[random_root(dpt)]

harmony.embedding.top$pt = as.numeric(pseudotime_top)
harmony.embedding.top = harmony.embedding.top[order(harmony.embedding.top$pt),]

harmony.embedding.top$pt_adj = (harmony.embedding.top$pt)
harmony.embedding.top$pt_adj = harmony.embedding.top$pt_adj + max(harmony.embedding.center2$pt_adj) + max(harmony.embedding.bottom$pt_adj) + max(harmony.embedding.center1$pt_adj)
harmony.embedding.top$rank = rev(1:nrow(harmony.embedding.top))
harmony.embedding.top$rank = harmony.embedding.top$rank   + nrow(harmony.embedding.center2) + nrow(harmony.embedding.bottom)+ nrow(harmony.embedding.center1)


pseudotime = rbind(harmony.embedding.bottom,harmony.embedding.center1,harmony.embedding.center2,harmony.embedding.top,stringsAsFactors = F)

plot.pt = cbind(meta.data,rank = pseudotime[match(rownames(meta.data),rownames(pseudotime)),"rank"],pt_adj = pseudotime[match(rownames(meta.data),rownames(pseudotime)),"pt_adj"], stringsAsFactors = F)


tomo.harm@meta.data$pseudo_rank = plot.pt$rank
tomo.harm@meta.data$pseudo_raw = plot.pt$pt_adj

### set some parameters
# cells
cellcex = 1
#color
colramp = colorRampPalette(viridis(12))
colgrad = colramp(101)

pdf("./tomo_Harmony_Embedding_UMAP_PT_Rank.pdf",width=8,height=8)

#ranks
  # normalize expression [0-1]
  exp = plot.pt [,"rank"]/max(plot.pt [,"rank"])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  plot.pt [,"bin"] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(plot.pt$UMAP_1, plot.pt$UMAP_2, cex=cellcex, pch=16, col=colgrad[plot.pt[,"bin"]], xlab="", ylab="", main = "pseudotime rank")
    
    
#raw pt
     # normalize expression [0-1]
  exp = plot.pt [,"pt_adj"]/max(plot.pt [,"pt_adj"])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  plot.pt [,"bin"] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(plot.pt$UMAP_1, plot.pt$UMAP_2, cex=cellcex, pch=16, col=colgrad[plot.pt[,"bin"]], xlab="", ylab="", main = "pseudotime raw")
    
dev.off()

saveRDS(tomo.harm,"./tomo_Harmony.RDS")



#plot cluster marker as heatmap along pseudotime


genes = readRDS("./uncut_Harmony_ClusterMarker_final.RDS")

#hr <- hclust(as.dist(1-cor(t(data),method="pearson")), method="ward")

#data = data[match(names(sort(cutree(hr,k=4))),rownames(data)),]

#if (!require(devtools)) {
#  install.packages("devtools")
#}

#devtools::install_github("xmc811/Scillus", ref = "development")
library(Scillus)


pdf("tomo_Harmony_Heatmap_UncutClusterMarker_final.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = genes,
              sort_var = c("pseudo_rank"),
              anno_var = c("time_point","sample_name","pseudo_rank","slice_sum","seurat_clusters"),
              anno_colors = list("Set1","Greys", "RdYlBu" ,"YlGn","Set3"), hm_colors = viridis_pal(option = "plasma")(3), hm_limit = c(-0.5,0,1))
ggplot(FetchData(tomo.harm,c("slice_sum","pseudo_rank")),aes(x = pseudo_rank, color = slice_sum, fill = slice_sum )) + geom_histogram(bins = nrow(FetchData(tomo.harm,c("slice_sum","pseudo_rank")))) + scale_fill_manual(values = rev(colorRampPalette(brewer.pal(9,"YlGn"))(9)))
dev.off()


```

##Check alignment of pseudo and slice index

```{r}

tmp = FetchData(tomo.harm,c("sample_id","time_point","pseudo_rank","slice_index"))
tmp = list()
for (i in unique(tomo.harm@meta.data$sample_id)) {
  tmp[[i]] = FetchData(subset(tomo.harm,subset = sample_id == i),c("sample_id","time_point","pseudo_rank","slice_index","nCount_RNA"))
  tmp[[i]] = tmp[[i]][order(tmp[[i]][,"pseudo_rank"]),]
  tmp[[i]]$pseudo_rank_slice =  1:length(tmp[[i]]$pseudo_rank)
  tmp[[i]] = tmp[[i]][order(tmp[[i]][,"slice_index"]),]
  tmp[[i]]$slice_index_slice =  1:length(tmp[[i]]$pseudo_rank)
  tmp[[i]]$grouping = factor(1:length(tmp[[i]]$pseudo_rank))
}

tmp = do.call("rbind", tmp)

tmp$diff = abs(tmp$slice_index_slice - tmp$pseudo_rank_slice)

plot.df = melt(tmp[,c("pseudo_rank_slice","slice_index_slice","grouping","sample_id","time_point")],id.vars = c("grouping","sample_id","time_point"))

plot.df$sample_id = as.factor(plot.df$sample_id)
plot.df$sample_id <- factor(plot.df$sample_id, levels = c("HH7T7BGXF-39","HH7T7BGXF-40","HH7T7BGXF-42","HLLWFBGXH-9","HLMT2BGXH-37","HLMTJBGXH-1","HLLWFBGXH-17","HLMKKBGXH-2","HLMMWBGXH-4","HLMTJBGXH-40","HLLWFBGXH-18","HLMKKBGXH-3","HLMMWBGXH-42","HLMKKBGXH-4","HLMMWBGXH-6","HLMT2BGXH-48","HLLWFBGXH-13","HLMMWBGXH-7","HLMT2BGXH-43","HLMTJBGXH-16"))

tmp$sample_id = as.factor(tmp$sample_id)
tmp$sample_id <- factor(tmp$sample_id, levels = c("HH7T7BGXF-39","HH7T7BGXF-40","HH7T7BGXF-42","HLLWFBGXH-9","HLMT2BGXH-37","HLMTJBGXH-1","HLLWFBGXH-17","HLMKKBGXH-2","HLMMWBGXH-4","HLMTJBGXH-40","HLLWFBGXH-18","HLMKKBGXH-3","HLMMWBGXH-42","HLMKKBGXH-4","HLMMWBGXH-6","HLMT2BGXH-48","HLLWFBGXH-13","HLMMWBGXH-7","HLMT2BGXH-43","HLMTJBGXH-16"))

pdf("./tomo_Harmony_PT_index_comparison.pdf",width=15,height=15)
ggplot(plot.df,aes(x= value,y = variable, color = time_point, group = grouping)) + geom_point() + geom_line() + facet_wrap(~sample_id, ncol = 6) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])) 
ggplot(tmp,aes(x= slice_index_slice,y = diff, fill = time_point)) + geom_col() + facet_wrap(~sample_id, ncol = 6) + scale_fill_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])) 
ggplot(tmp,aes(x= slice_index_slice,y = nCount_RNA, fill = time_point)) + geom_col() + facet_wrap(~sample_id, scales = "free_y", ncol = 6) + scale_fill_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])) 
ggplot(tmp,aes(x= diff,y = nCount_RNA, color = time_point)) + geom_point() + facet_wrap(~sample_id, scales = "free_y", ncol = 6) + geom_smooth(method='lm', formula= y~x, se = F) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])) 
ggplot(tmp,aes(x= diff,y = nCount_RNA, color = time_point)) + geom_point()  + geom_smooth(method='lm', formula= y~x, se = F) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])) 
ggplot(tmp,aes(x= pseudo_rank_slice,y = slice_index_slice,color = time_point)) + geom_point() + facet_wrap(~sample_id, ncol = 6) + geom_smooth(method='lm', formula= y~x) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])) 
dev.off()


#Get R and function for regression line and R value

model = list()

for (i in 1:length(unique(tmp$sample_id))) {
  tmp.sub = tmp[tmp$sample_id %in% unique(tmp$sample_id)[i],]
  model[[as.character(unique(tmp$sample_id))[i]]] <- summary(lm(pseudo_rank_slice ~ slice_index_slice, data = tmp.sub))

}

saveRDS(model,"./tomo_Harmony_PT_index_comparison_statistics.RDS")
  

aggregate(tmp$diff, list(tmp$time_point), FUN=mean) 

#Plot Fig1 heatmap marker for individual poyps


genes = readRDS("./uncut_Harmony_ClusterMarker_final.RDS")

#devtools::install_github("xmc811/Scillus", ref = "development")
library(Scillus)


pdf("tomo_Harmony_Heatmap_UncutClusterMarker_SliceIndex_byPolyp.pdf",width=40,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = genes,
              sort_var = c("sample_id","slice_index"),
              anno_var = c("slice_index","sample_name","pseudo_rank","slice_sum","seurat_clusters"),
              anno_colors = list("YlGn","Greys", "RdYlBu" ,"YlGn","Set3"), hm_colors = viridis_pal(option = "plasma")(3), hm_limit = c(-0.5,0,1))
dev.off()


```

##SPRING


```{r}

harmony.embedding = Embeddings(tomo.harm, "harmony")

#make scaled values positive
for (i in 1:ncol(harmony.embedding)) {
  harmony.embedding[,i] = harmony.embedding[,i] + abs(min(harmony.embedding[,i]))
}

harmony.embedding = as.data.frame(harmony.embedding,stringsAsFactor = F)


#extract norm data from Seurat object

norm.data = GetAssayData(object = tomo.harm, slot = "data")
norm.data = as.matrix(norm.data)


meta.data = FetchData(object = tomo.harm, vars = c('ident',"seurat_clusters", 'orig.ident','UMAP_1','UMAP_2',"sequencing_sample_id","batch","sample_name","time_point","slice_index","slice_sum"))

##############################
#SPRING
##############################

### export for SPRING (export norma and not scaled data)

#cutoff 0.8
y=as.data.frame(t(meta.data[,c('seurat_clusters','orig.ident','slice_sum',"sample_name")]),stringsAsFactors=F)

write.table(y,sep = ",",col.names = F,file = "SPRING_Export_tomoAll_meta.csv")

x = as.data.frame(t(harmony.embedding[,c(2:30)]))

write.table(x,sep = ",",col.names = F,file ="SPRING_Export_tomoAll_HarmonyMatrix.csv")

z=rownames(x)

write.table(z,col.names = F,row.names = F,sep = ",",file = "SPRING_Export_tomoAll_GeneList.csv")

###Run with : 10 PCs gene /10 neighbors 

#import SPRING tables

coords = read.csv("/g/arendt/gerber/Analyses/NematostellaTomoSeq/tomoAll_coordinates.txt",header = F)
colnames(coords) = c("Cell","x","y")
edges = read.csv("/g/arendt/gerber/Analyses/NematostellaTomoSeq/tomoAll_edge_list.txt",header = F)
colnames(edges) = c("start","end")

rownames(coords) = rownames(harmony.embedding)

### set some parameters
# cells
colstart = "darkblue"
colmed1 = "green"
colmed2 = "yellow"
colmed3 = "orange"
colend = "firebrick"
cellcex = 2
# edges
linecol = adjustcolor("black", alpha.f=0.4)
#linecol = adjustcolor("gray")
linewidth = 0.75

#flip
coords$y = coords$y * -1

coords$rank=tomo.harm@meta.data$pseudo_rank
coords$slice=tomo.harm@meta.data$slice_sum
coords$slice =  as.numeric(gsub("slice","",coords$slice))
coords$ident=tomo.harm@meta.data$seurat_clusters

coords$color[coords$ident=="0"]= "darkgoldenrod3"
coords$color[coords$ident=="1"]="red3"
coords$color[coords$ident=="2"]="salmon"
coords$color[coords$ident=="3"]= "turquoise4"
coords$color[coords$ident=="4"]="dodgerblue"


plot.data = coords
plot.data$order = sample(1:nrow(plot.data),replace = F)
plot.data = plot.data[order(plot.data$order),]
#plot.data$Cell = rownames(plot.data)

pdf("./tomo_Harmony_SPRING_ident.pdf",width=8,height=5)
plot(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=plot.data$color , xlab="", ylab="")

edge_start_coords = plot.data[match(edges$start, plot.data$Cell), c("x","y")]
edge_end_coords = plot.data[match(edges$end, plot.data$Cell), c("x","y")]
segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
	x1=edge_end_coords[,"x"], y1=edge_end_coords[,"y"], lwd=linewidth, col=linecol)

points(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=plot.data$color)

dev.off()

#time point

coords$time_point=tomo.harm@meta.data$time_point

coords$color[coords$time_point=="uncut"]= c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])[1]
coords$color[coords$time_point=="12hpa"]= c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])[2]
coords$color[coords$time_point=="24hpa"]= c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])[3]
coords$color[coords$time_point=="48hpa"]= c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])[4]
coords$color[coords$time_point=="96hpa"]= c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])[5]


plot.data = coords
plot.data$order = sample(1:nrow(plot.data),replace = F)
plot.data = plot.data[order(plot.data$order),]
#plot.data$Cell = rownames(plot.data)

pdf("./tomo_Harmony_SPRING_time_point.pdf",width=8,height=5)
plot(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=plot.data$color , xlab="", ylab="")

edge_start_coords = plot.data[match(edges$start, plot.data$Cell), c("x","y")]
edge_end_coords = plot.data[match(edges$end, plot.data$Cell), c("x","y")]
segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
	x1=edge_end_coords[,"x"], y1=edge_end_coords[,"y"], lwd=linewidth, col=linecol)

points(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=plot.data$color)

dev.off()



# cells
cellcex = 2
#color

#colramp = colorRampPalette(c("dodgerblue","violetred2","darkgoldenrod3"))(101)
colramp = colorRampPalette(rev(brewer.pal(n = 9, name = "YlGnBu"))[1:8])(101)
colgrad = colramp

#Get plot for colorpalette
pdf("./PT.scale.pdf",width=8,height=4)
FeaturePlot(tomo.harm, reduction = 'umap', pt.size = 1, features = c("nFeature_RNA","nCount_RNA"),order = F, cols = colorRampPalette(c("dodgerblue","violetred2","darkgoldenrod3"))(101))
FeaturePlot(uncut, reduction = 'umap', pt.size = 1, features = c("nFeature_RNA","nCount_RNA"),order = F, cols = colorRampPalette(rev(brewer.pal(n = 9, name = "YlGnBu"))[1:8])(101))
dev.off()

pdf("./tomo_Harmony_SPRING_PT.pdf",width=8,height=5)

  # normalize expression [0-1]
  exp = plot.data[,"rank"]/max(plot.data[,"rank"])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  plot.data[,"bin"] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=colgrad[plot.data[,"bin"]], xlab="", ylab="", main = "pseudotime")

    
edge_start_coords = plot.data[match(edges$start, plot.data$Cell), c("x","y")]
edge_end_coords = plot.data[match(edges$end, plot.data$Cell), c("x","y")]
segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
	x1=edge_end_coords[,"x"], y1=edge_end_coords[,"y"], lwd=linewidth, col=linecol)

    points(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=colgrad[plot.data[,"bin"]], xlab="", ylab="")

dev.off()

cellcex = 2
#color

#colramp = colorRampPalette(c("dodgerblue","violetred2","darkgoldenrod3"))(101)
colramp =  colorRampPalette(rev(brewer.pal(n = 9, name = "YlGn"))[1:8])(101)
colgrad = colramp

#Get plot for colorpalette
pdf("./Bin.scale.pdf",width=8,height=4)
FeaturePlot(uncut, reduction = 'umap', pt.size = 1, features = c("nFeature_RNA","nCount_RNA"),order = F, cols = colorRampPalette(c("dodgerblue","violetred2","darkgoldenrod3"))(101))
FeaturePlot(uncut, reduction = 'umap', pt.size = 1, features = c("nFeature_RNA","nCount_RNA"),order = F, cols = colorRampPalette(rev(brewer.pal(n = 9, name = "YlGn"))[1:8])(101))
dev.off()

pdf("./tomo_Harmony_SPRING_SliceBin.pdf",width=8,height=5)

  # normalize expression [0-1]
  exp = plot.data[,"slice"]/max(plot.data[,"slice"])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  plot.data[,"bin"] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=colgrad[plot.data[,"bin"]], xlab="", ylab="", main = "pseudotime")

    
edge_start_coords = plot.data[match(edges$start, plot.data$Cell), c("x","y")]
edge_end_coords = plot.data[match(edges$end, plot.data$Cell), c("x","y")]
segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
	x1=edge_end_coords[,"x"], y1=edge_end_coords[,"y"], lwd=linewidth, col=linecol)

    points(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=colgrad[plot.data[,"bin"]], xlab="", ylab="")

dev.off()

#plot genes on Embedding

#plot for paper
tmp=c("NVEC200-000872.1",
"NVEC200-010351.1",
"NVEC200-012164.1",
"NVEC200-000491.1",
"NVEC200-006580.1",
"NVEC200-015113.1",
"NVEC200-015112.1", 
"NVEC200-001364.1",
"NVEC200-001363.1", 
"NVEC200-001362.1", 
"NVEC200-006274.1",
"NVEC200-013190.1",
"NVEC200-013831.1",
"NVEC200-001043.1",
"NVEC200-005938.1",
"NVEC200-001872.1")


meta.data = FetchData(object = tomo.harm, vars = c('seurat_clusters', 'orig.ident',tmp))

tmp.name=c("Anthox7",
"cdh1",
"FoxB",
"GBX",
"M-lim",
"NvnAChRaC",
"NvnAChRaD",
"OtxA",
"OtxB",
"OtxC",
"Sox2",
"SoxE",
"TCF",
"Anthox1a",
"metalloproteinase inhibitor 3",
"uromodulin_b")
names(tmp.name) = tmp

#color.palette = c(colorRampPalette(brewer.pal(9,"Greys")[9:3])(n = 30),colorRampPalette(brewer.pal(9,"Reds")[3:9])(n = 69))

# create 200 colors (colstart --> colend)
colramp1 = colorRampPalette(brewer.pal(9,"Greys")[9:2])
colramp2 = colorRampPalette(brewer.pal(9,"Reds")[2:9])

colgrad = c(colramp1(50),colramp2(51))

colgrad = viridis_pal(option = "plasma")(101)

cellcex = 2.5
# edges
linecol = adjustcolor("black")
linewidth = 0.5

##########

pdf("./tomo_SPRING_PaperMarker.pdf",width=20,height=13)

par(mfrow=c(4,4))

for (i in colnames(meta.data[,tmp])) {

  # normalize expression [0-1]
  exp = meta.data[,i]/max(meta.data[,i])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  coords[,i] = round(exp, digits=2) *100 +1
  
  coords = coords[order(coords[,i]),]
  #plot gradient

    plot(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="", main = tmp.name[i],bty="n",yaxt="n",xaxt="n")

    edge_start_coords = coords[match(edges$start, coords$Cell), c("x","y")]
    edge_end_coords = coords[match(edges$end, coords$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="")

    coords = coords[match(colnames(tomo.harm),rownames(coords)),]
}

dev.off()


#plot all


tmp=c("NVEC200-001043.1",
"NVEC200-000874.1",
"NVEC200-000872.1",
"NVEC200-010351.1",
"NVEC200-010468.1",
"NVEC200-006988.1",
"NVEC200-001297.1",
"NVEC200-007533.1",
"NVEC200-014111.1",
"NVEC200-006032.1",
"NVEC200-006030.1",
"NVEC200-006642.1",
"NVEC200-006028.1",
"NVEC200-003168.1",
"NVEC200-012164.1",
"NVEC200-009995.1",
"NVEC200-011699.1",
"NVEC200-000491.1",
"NVEC200-002987.1",
"NVEC200-000883.1",
"NVEC200-006580.1",
"NVEC200-005938.1",
"NVEC200-001459.1",
"NVEC200-001458.1",
"NVEC200-009287.1",
"NVEC200-008102.1",
"NVEC200-015113.1",
"NVEC200-015112.1",
"NVEC200-012338.1",
"NVEC200-001364.1",
"NVEC200-001363.1",
"NVEC200-001362.1",
"NVEC200-013954.1",
"NVEC200-011655.1",
"NVEC200-008902.1",
"NVEC200-009545.1",
"NVEC200-003200.1",
"NVEC200-009182.1",
"NVEC200-006274.1",
"NVEC200-001846.1",
"NVEC200-013190.1",
"NVEC200-004425.1",
"NVEC200-006033.1",
"NVEC200-013831.1",
"NVEC200-014326.1",
"NVEC200-008616.1",
"NVEC200-008615.1",
"NVEC200-001014.1",
"NVEC200-000260.1",
"NVEC200-009768.1",
"NVEC200-008308.1",
"NVEC200-000261.1",
"NVEC200-010238.1",
"NVEC200-008832.1",
"NVEC200-005807.1",
"NVEC200-005805.1",
"NVEC200-005804.1",
"NVEC200-001872.1",
"NVEC200-011709.1",
"NVEC200-008022.1",
"NVEC200-012414.1")

tmp.name=c("Anthox1a",
"Anthox6",
"Anthox7",
"cdh1",
"cdh3",
"Churchill",
"Ci",
"delta",
"FGF1A",
"FGF8A",
"FGF8B",
"FGFRa",
"FGFRb",
"Fox1",
"FoxB",
"FoxC ",
"FoxD",
"GBX",
"Hedgehog2",
"HLXB9",
"M-lim",
"MMP 3 inhibitor",
"MoxB_a",
"MoxB_b",
"NCam1",
"NvnAChRaB",
"NvnAChRaC",
"NvnAChRaD",
"NvnAChRaU",
"OtxA",
"OtxB",
"OtxC",
"PaxA",
"PaxD3",
"PL10",
"Repo",
"snailA",
"Sox1",
"Sox2",
"Sox3",
"SoxE",
"SoxF1",
"Sprouty",
"TCF",
"Thiamine enzyme",
"wnt 5",
"wnt 7",
"wnt1",
"wnt10",
"wnt11",
"wnt2",
"wnt6",
"wnt8b",
"wntA",
"ZicC",
"ZicD",
"ZicE",
"Uromodulin_a",
"otp",
"wnt16",
"RGM")
names(tmp.name) = tmp

#color.palette = c(colorRampPalette(brewer.pal(9,"Greys")[9:3])(n = 30),colorRampPalette(brewer.pal(9,"Reds")[3:9])(n = 69))

# create 200 colors (colstart --> colend)
colramp1 = colorRampPalette(brewer.pal(9,"Greys")[9:2])
colramp2 = colorRampPalette(brewer.pal(9,"Reds")[2:9])

colgrad = c(colramp1(50),colramp2(51))

colgrad = viridis_pal(option = "plasma")(101)

cellcex = 2.5
# edges
linecol = adjustcolor("black")
linewidth = 0.5


meta.data = FetchData(object = tomo.harm, vars = c('seurat_clusters', 'orig.ident'))
tmp.mtx = GetAssayData(tomo.harm,slot = "scale.data")
meta.data = cbind(meta.data,as.data.frame(t(tmp.mtx[tmp,])))

##########

pdf("./tomo_Harmony_SPRING_AllInsituMarker_scaled.pdf",width=30,height=22)

par(mfrow=c(8,8))

for (i in colnames(meta.data[,tmp])) {

  # normalize expression [0-1]
  meta.data[,i] = meta.data[,i] + abs(min(meta.data[,i]))
  exp = meta.data[,i]/max(meta.data[,i])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  coords[,i] = round(exp, digits=2) *100 +1
  
  coords = coords[order(coords[,i]),]
  #plot gradient

    plot(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="", main = tmp.name[i],bty="n",yaxt="n",xaxt="n")

    edge_start_coords = coords[match(edges$start, coords$Cell), c("x","y")]
    edge_end_coords = coords[match(edges$end, coords$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="")

    coords = coords[match(colnames(tomo.harm),rownames(coords)),]
}

dev.off()


meta.data = FetchData(object = tomo.harm, vars = c('seurat_clusters', 'orig.ident'))
tmp.mtx = GetAssayData(tomo.harm,slot = "data")
meta.data = cbind(meta.data,as.data.frame(t(as.matrix(tmp.mtx[tmp,]))))

pdf("./tomo_Harmony_SPRING_AllInsituMarker_norm.pdf",width=25,height=22)

par(mfrow=c(8,8))

for (i in colnames(meta.data[,tmp])) {

  # normalize expression [0-1]
  exp = meta.data[,i]/max(meta.data[,i])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  coords[,i] = round(exp, digits=2) *100 +1
  
  coords = coords[order(coords[,i]),]
  #plot gradient

    plot(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="", main = tmp.name[i],bty="n",yaxt="n",xaxt="n")

    edge_start_coords = coords[match(edges$start, coords$Cell), c("x","y")]
    edge_end_coords = coords[match(edges$end, coords$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="")

    coords = coords[match(colnames(tomo.harm),rownames(coords)),]
}

dev.off()



```

###Markers

```{r}
gene_annotation = read.csv("./tomo_assays_rna_metafeatures.csv")

Idents(object = tomo.harm) <-  tomo.harm@meta.data$seurat_clusters

markers <- FindAllMarkers(tomo.harm, only.pos = TRUE,  logfc.threshold = 0.3)
markers$diff = markers$pct.1 - markers$pct.2

markers.anno = markers
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id")
markers.anno = markers.anno  %>% arrange(cluster , desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_AllSeuratClusterMarker.tsv")


#plot with annotation
gene_ids =c("NVEC200-007938.1",
"NVEC200-006310.1",
"NVEC200-007657.1",
"NVEC200-001774.1",
"NVEC200-005045.1",
"NVEC200-012721.1",
"NVEC200-000574.1",
"NVEC200-003926.1",
"NVEC200-007145.1",
"NVEC200-004198.1",
"NVEC200-003956.1",
"NVEC200-004328.1",
"NVEC200-005871.1",
"NVEC200-002920.1",
"NVEC200-007658.1",
"NVEC200-006293.1",
"NVEC200-005409.1",
"NVEC200-000071.1",
"NVEC200-007344.1",
"NVEC200-008391.1",
"NVEC200-000603.1",
"NVEC200-009271.1",
"NVEC200-004490.1",
"NVEC200-001362.1",
"NVEC200-006554.1",
"NVEC200-000775.1",
"NVEC200-002050.1",
"NVEC200-002734.1",
"NVEC200-006858.1",
"NVEC200-002496.1",
"NVEC200-001506.1",
"NVEC200-010860.1",
"NVEC200-000296.1",
"NVEC200-006821.1")


gene_names =c(
"Gsx1",
"Fezf2",
"Gpx5",
"XPNPEP1",
"Tpcn1",
"XPO6",
"TUBA1D",
"fezf2",
"EDIL3",
"Capsl",
"Pnliprp1",
"HNRNPL",
"Tmprss9",
"FRRS1",
"Gm2a",
"GBA3",
"CTRB2",
"SIX4",
"TMPRSS11D",
"TLL1",
"Itga9",
"Btg1",
"Edil3",
"OTX1",
"MYC",
"Flna",
"TJP1",
"blt801",
"SLC16A3",
"PF13-0198",
"ART3",
"LAF3",
"LWamide neuropeptides",
"skeletal organic matrix protein 5")

gg_Fig <- FeaturePlot(tomo.harm, pt.size = 2, features = gene_ids,order = T, cols = c("gray",rev(viridis_pal(option = "mako")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) & NoLegend() & NoAxes()})

pdf("tomo_Harmony_Embedding_HclustClusterMarkerAnno.pdf",width=25,height=20)
CombinePlots( gg_Fig )
dev.off()


#plot with annotation
gene_ids =c(
"NVEC200-009271.1",
"NVEC200-006430.1",
"NVEC200-014823.1",
"NVEC200-005937.1",
"NVEC200-006423.1",
"NVEC200-003174.1",
"NVEC200-007376.1",
"NVEC200-007166.1",
"NVEC200-002459.1",
"NVEC200-002734.1",
"NVEC200-004198.1",
"NVEC200-005968.1",
"NVEC200-011460.1",
"NVEC200-013726.1",
"NVEC200-010443.1",
"NVEC200-002310.1",
"NVEC200-014047.1",
"NVEC200-003642.1",
"NVEC200-011925.1",
"NVEC200-000547.1",
"NVEC200-001405.1",
"NVEC200-005216.1",
"NVEC200-002397.1",
"NVEC200-011742.1",
"NVEC200-008616.1",
"NVEC200-002307.1",
"NVEC200-003388.1",
"NVEC200-003926.1",
"NVEC200-000574.1",
"NVEC200-005104.1",
"NVEC200-012414.1",
"NVEC200-001119.1",
"NVEC200-012745.1",
"NVEC200-005455.1",
"NVEC200-007938.1",
"NVEC200-006310.1",
"NVEC200-003993.1",
"NVEC200-007657.1",
"NVEC200-001253.1",
"NVEC200-006391.1",
"NVEC200-000758.1",
"NVEC200-004893.1",
"NVEC200-004526.1",
"NVEC200-000574.1",
"NVEC200-013268.1",
"NVEC200-004653.1",
"NVEC200-012175.1",
"NVEC200-006239.1")


gene_names =c(
"Btg1",
"hmx3",
"PSAP",
"Hoxc4",
"nkx-2.5",
"SELENOP",
"CalpB",
"Trafd1",
"faxc",
"blt801",
"Capsl",
"Gpx2",
"Ctrb1",
"AQP4",
"fabL",
"OVCH1",
"Mmp19",
"Cpa2",
"DSCAM",
"aguA",
"OIT3",
"PRY2",
"ADAM9",
"Npr1",
"Wnt5a",
"CUBlike",
"50 kDa hatching enzyme",
"fezf2",
"TUBA1D",
"PKD1L2",
"RGMB",
"CD63",
"pkd2",
"Csrp2",
"Gsx1",
"Fezf2",
"LYPLAL1",
"Gpx5",
"C1qb",
"NKX6-2",
"COL17A1",
"Adgrd1",
"Tekt1",
"TUBA1D",
"kcnk9",
"TEKT2",
"CHRNA9",
"KCND2")

gg_Fig <- FeaturePlot(tomo.harm, pt.size = 2, features = gene_ids,order = T, cols = c("gray",rev(viridis_pal(option = "mako")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) & NoLegend() & NoAxes()})

pdf("tomo_Harmony_Embedding_SeuratClusterMarkerAnno.pdf",width=25,height=20)
CombinePlots( gg_Fig )
dev.off()



#plot with annotation
gene_ids =c("NVEC200-014282.1",
"NVEC200-012734.1",
"NVEC200-007620.1",
"NVEC200-006579.1",
"NVEC200-012735.1",
"NVEC200-000612.1",
"NVEC200-002278.1",
"NVEC200-005841.1",
"NVEC200-003858.1",
"NVEC200-008100.1",
"NVEC200-003362.1",
"NVEC200-000942.1",
"NVEC200-002307.1",
"NVEC200-003388.1",
"NVEC200-013945.1",
"NVEC200-005748.1",
"NVEC200-013180.1",
"NVEC200-007065.1",
"NVEC200-007110.1",
"NVEC200-001093.1",
"NVEC200-014405.1",
"NVEC200-007443.1",
"NVEC200-012327.1",
"NVEC200-012414.1",
"NVEC200-009768.1",
"NVEC200-000731.1",
"NVEC200-008186.1")


gene_names =c(
"Col7a1",
"ACT1",
"FOS-FOX",
"CSRP2",
"act-3",
"sqh",
" CUBlike_2278",
"Klkb1",
"Hatching enzyme Paracentrotus",
"FBLN2",
"CPA4",
"tut1",
"CUBlike_2307",
"Hatching enzyme Hemicentrotus ",
"ADAM17",
"Myl9",
"actb",
"Calml3",
"NOVA1",
"CNR11",
"Pxn",
"NEFH",
"Mctp1",
"RGMB",
"wnt4a",
"VCAN",
"BRN3")

gg_Fig <- FeaturePlot(tomo.harm, pt.size = 2, features = gene_ids,order = T, cols = c("gray",rev(viridis_pal(option = "mako")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) & NoLegend() & NoAxes()})

pdf("tomo_Harmony_Embedding_SeuratClusterMarkerAnno_avgFCsort.pdf",width=25,height=20)
CombinePlots( gg_Fig )
dev.off()



#Plot markers Danila identified
genes = c("NVEC200-008961.1",
"NVEC200-004758.1",
"NVEC200-010547.1",
"NVEC200-004699.1",
"NVEC200-013420.1",
"NVEC200-005885.1",
"NVEC200-001850.1",
"NVEC200-001506.1",
"NVEC200-009111.1",
"NVEC200-002739.1",
"NVEC200-000357.1",
"NVEC200-011114.1",
"NVEC200-004088.1",
"NVEC200-006835.1",
"NVEC200-007376.1",
"NVEC200-008361.1",
"NVEC200-005413.1",
"NVEC200-001093.1",
"NVEC200-000882.1",
"NVEC200-010810.1",
"NVEC200-011748.1",
"NVEC200-003361.1",
"NVEC200-009216.1",
"NVEC200-010524.1",
"NVEC200-011105.1",
"NVEC200-003928.1",
"NVEC200-004068.1",
"NVEC200-003718.1",
"NVEC200-009385.1",
"NVEC200-009713.1",
"NVEC200-008552.1",
"NVEC200-007145.1",
"NVEC200-007568.1",
"NVEC200-015129.1",
"NVEC200-007091.1",
"NVEC200-001874.1",
"NVEC200-006080.1",
"NVEC200-001602.1",
"NVEC200-015237.1",
"NVEC200-008300.1",
"NVEC200-002436.1",
"NVEC200-010180.1",
"NVEC200-014321.1",
"NVEC200-000531.1",
"NVEC200-008855.1",
"NVEC200-011425.1",
"NVEC200-003970.1",
"NVEC200-004711.1",
"NVEC200-010860.1",
"NVEC200-009422.1",
"NVEC200-003438.1",
"NVEC200-001625.1",
"NVEC200-003483.1",
"NVEC200-010800.1",
"NVEC200-014901.1",
"NVEC200-004408.1",
"NVEC200-002875.1",
"NVEC200-000292.1",
"NVEC200-008767.1",
"NVEC200-007537.1",
"NVEC200-001252.1",
"NVEC200-002111.1",
"NVEC200-013652.1",
"NVEC200-008961.1",
"NVEC200-000259.1",
"NVEC200-002800.1",
"NVEC200-003335.1",
"NVEC200-009192.1",
"NVEC200-013545.1",
"NVEC200-012953.1",
"NVEC200-005509.1",
"NVEC200-012412.1",
"NVEC200-013550.1",
"NVEC200-010451.1",
"NVEC200-002496.1",
"NVEC200-005192.1",
"NVEC200-001883.1",
"NVEC200-002101.1",
"NVEC200-001847.1",
"NVEC200-004463.1",
"NVEC200-005075.1",
"NVEC200-008011.1",
"NVEC200-005812.1",
"NVEC200-003934.1",
"NVEC200-003511.1",
"NVEC200-006561.1",
"NVEC200-012975.1",
"NVEC200-004586.1",
"NVEC200-013600.1",
"NVEC200-011019.1",
"NVEC200-005871.1",
"NVEC200-010328.1",
"NVEC200-013972.1",
"NVEC200-007471.1",
"NVEC200-008858.1",
"NVEC200-000611.1",
"NVEC200-008026.1",
"NVEC200-002544.1",
"NVEC200-002534.1",
"NVEC200-012988.1")

pdf("tomo_Harmony_Embedding_DanilaRegionalMarker_Feature.pdf",width=10,height=100)
FeaturePlot(tomo.harm, pt.size = 1, features = genes,split.by = "time_point", order = T, cols = c("gray",rev(viridis_pal(option = "mako")(12))), repel=T) & NoLegend() & NoAxes()
dev.off()



data = as.matrix(GetAssayData(object = tomo.harm, slot = "data"))
data = data[genes,]

hr <- hclust(as.dist(1-cor(t(data),method="pearson")), method="ward")

data = data[match(names(sort(cutree(hr,k=4))),rownames(data)),]

#if (!require(devtools)) {
#  install.packages("devtools")
#}

#devtools::install_github("xmc811/Scillus", ref = "development")
library(Scillus)


pdf("tomo_Harmony_DanilaRegionalMarker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = unique(rownames(data)),
              sort_var = c("time_point","pseudo_rank"),
              anno_var = c("time_point","sample_name","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1","Set3", "RdYlBu" ,"RdYlBu",c("yellow","red")), hm_colors = viridis_pal(option = "viridis")(3), hm_limit = c(-1,0,1))
dev.off()

#very interestingv genes

genes = c("NVEC200-009713.1",
"NVEC200-014321.1",
"NVEC200-006561.1",
"NVEC200-006835.1",
"NVEC200-006080.1",
"NVEC200-000531.1",
"NVEC200-008011.1",
"NVEC200-008855.1",
"NVEC200-001602.1",
"NVEC200-008767.1",
"NVEC200-002534.1",
"NVEC200-010524.1",
"NVEC200-013550.1",
"NVEC200-002436.1",
"NVEC200-003335.1",
"NVEC200-009422.1",
"NVEC200-002739.1",
"NVEC200-003438.1",
"NVEC200-002800.1",
"NVEC200-009192.1",
"NVEC200-001093.1",
"NVEC200-003718.1",
"NVEC200-004758.1",
"NVEC200-013420.1")


data = as.matrix(GetAssayData(object = tomo.harm, slot = "data"))
data = data[genes,]

hr <- hclust(as.dist(1-cor(t(data),method="pearson")), method="ward")

data = data[match(names(sort(cutree(hr,k=10))),rownames(data)),]

#if (!require(devtools)) {
#  install.packages("devtools")
#}

#devtools::install_github("xmc811/Scillus", ref = "development")
library(Scillus)


pdf("tomo_Harmony_DanilaRegionalMarkerVeryInt_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = unique(rownames(data)),
              sort_var = c("time_point","pseudo_rank"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "Spectral" ,"RdYlBu",c("yellow","red")), hm_colors = viridis_pal(option = "viridis")(3), hm_limit = c(-1,0,1))
dev.off()

#read in WoundDE genes by Danila

genes.wound = read.csv("./WoundDEDanila.csv", header = F)
genes.wound[1,1] = "NVEC200-000209.1"

data = as.matrix(GetAssayData(object = tomo.harm, slot = "data"))
data = data[genes.wound[,1],]

hr <- hclust(as.dist(1-cor(t(data),method="pearson")), method="ward")

data = data[match(names(sort(cutree(hr,k=15))),rownames(data)),]


pdf("tomo_Harmony_DanilaWoundMarker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = unique(rownames(data)),
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "Spectral" ,"RdYlBu",c("yellow","red")), hm_colors = viridis_pal(option = "viridis")(3), hm_limit = c(-1,0,1))
dev.off()


pdf("tomo_Harmony_DanilaWoundMarker_Feature.pdf",width=10,height=200)
FeaturePlot(tomo.harm,raster = F, pt.size = 1, features = unique(rownames(data)),split.by = "time_point", order = T, cols = c("gray",rev(viridis_pal(option = "mako")(12))), repel=T) & NoLegend() & NoAxes()
dev.off()

#Separate by region/cluster and timepoint 

tomo.harm@meta.data$cluster_time = paste(tomo.harm@meta.data$seurat_clusters,tomo.harm@meta.data$time_point,sep = "_")

Idents(object = tomo.harm) <- tomo.harm@meta.data$cluster_time

markers <- FindAllMarkers(tomo.harm, only.pos = TRUE,  logfc.threshold = 0.3)
markers$diff = markers$pct.1 - markers$pct.2

markers.anno = markers
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(cluster , desc(diff))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_ClusterTimePointMarker.tsv")

markers.ClusterTime = markers.anno

#look for timepoint specific markers and remove them from the previous list

Idents(object = tomo.harm) <- tomo.harm@meta.data$time_point

markers <- FindAllMarkers(tomo.harm, only.pos = TRUE,  logfc.threshold = 0.3)
markers$diff = markers$pct.1 - markers$pct.2

markers.anno = markers
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(cluster , desc(diff))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_TimePointMarker.tsv")

markers.Time = markers.anno

#intersect to remove timpoint specific markers
markers.ClusterTime.filter = markers.ClusterTime[!markers.ClusterTime$gene %in% markers.Time$gene,]

write_delim(markers.ClusterTime.filter,delim = "\t","./tomo_Harmony_filterClusterTimePointMarker.tsv")

markers.top50<- markers.ClusterTime.filter %>% arrange(cluster , desc(avg_log2FC)) %>% group_by(cluster) %>% slice(1:50)
markers.top50 = unique(markers.top50$gene)


#order slices by pseudotime and timepoint


tmp = FetchData(subset(tomo.harm,subset = time_point == "96hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
meta.order = tmp
tmp = FetchData(subset(tomo.harm,subset = time_point == "48hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)
tmp = FetchData(subset(tomo.harm,subset = time_point == "24hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)
tmp = FetchData(subset(tomo.harm,subset = time_point == "12hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)

tmp = FetchData(uncut,c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)

meta.order = meta.order[match(colnames(tomo.harm),rownames(meta.order)),]

tomo.harm@meta.data$order_time = meta.order$order


data = as.matrix(GetAssayData(object = tomo.harm, slot = "data"))
data = data[markers.top50,]

hr <- hclust(as.dist(1-cor(t(data),method="pearson")), method="ward")

data = data[match(names(sort(cutree(hr,k=10))),rownames(data)),]


#tomo.harm@meta.data$barcode = NULL

pdf("tomo_Harmony_scHeatmap_ClusterTimeMarker.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = rownames(data),
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "Spectral" ,"RdYlBu",c("yellow","red")), hm_colors = rev(viridis_pal(option = "viridis")(12)), hm_limit = c(-1,0,1))
dev.off()

Idents(object = tomo.harm) <- tomo.harm@meta.data$seurat_clusters



#Compare regions with each other across time points

#Oral

tomo.harm@meta.data$cluster_time = paste(tomo.harm@meta.data$seurat_clusters,tomo.harm@meta.data$time_point,sep = "_")

Idents(object = tomo.harm) <- tomo.harm@meta.data$cluster_time

markers = FindMarkers(tomo.harm,c("3_uncut","4_uncut"),c("4_96hpa","4_48hpa","4_24hpa","4_12hpa","3_96hpa","3_48hpa","3_24hpa","3_12hpa","1_uncut","1_96hpa","1_48hpa","1_24hpa","1_12hpa"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_UncutTipMarker.tsv")

#this gives us uncut specific and regeneration specific tip markers
#Are there time point specific tip marker


markers = FindMarkers(tomo.harm,c("4_12hpa","3_12hpa"),c("4_96hpa","4_48hpa","4_24hpa","3_96hpa","3_48hpa","3_24hpa","3_12hpa","3_uncut","4_uncut","1_uncut","1_96hpa","1_48hpa","1_24hpa","1_12hpa"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_12hpaTipMarker.tsv")


markers = FindMarkers(tomo.harm,c("4_24hpa","3_24hpa"),c("4_96hpa","4_48hpa","3_96hpa","3_48hpa","3_12hpa","3_uncut","4_uncut","4_12hpa","3_12hpa","1_uncut","1_96hpa","1_48hpa","1_24hpa","1_12hpa"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_24hpaTipMarker.tsv")


markers = FindMarkers(tomo.harm,c("3_48hpa","4_48hpa"),c("4_96hpa","3_96hpa","3_12hpa","3_uncut","4_uncut","4_12hpa","3_12hpa","4_24hpa","3_24hpa","1_uncut","1_96hpa","1_48hpa","1_24hpa","1_12hpa"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_48hpaTipMarker.tsv")


markers = FindMarkers(tomo.harm,c("4_96hpa","3_96hpa"),c("3_12hpa","3_uncut","4_uncut","4_12hpa","3_12hpa","4_24hpa","3_24hpa","3_48hpa","4_48hpa","1_uncut","1_96hpa","1_48hpa","1_24hpa","1_12hpa"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_96hpaTipMarker.tsv")

#Compare regions with each other across time points

#Aboral with tip exclusion

tomo.harm@meta.data$cluster_time = paste(tomo.harm@meta.data$seurat_clusters,tomo.harm@meta.data$time_point,sep = "_")

Idents(object = tomo.harm) <- tomo.harm@meta.data$cluster_time

markers = FindMarkers(tomo.harm,c("1_uncut","1_uncut"),c("1_96hpa","1_48hpa","1_24hpa","1_12hpa","4_96hpa","4_48hpa","4_24hpa","4_12hpa","4_uncut","3_96hpa","3_48hpa","3_24hpa","3_12hpa","3_uncut"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_UncutFootMarker.tsv")

#this gives us uncut specific and regeneration specific tip markers
#Are there time point specific tip marker


markers = FindMarkers(tomo.harm,c("1_12hpa"),c("1_96hpa","1_48hpa","1_24hpa","1_uncut","4_96hpa","4_48hpa","4_24hpa","4_12hpa","4_uncut","3_96hpa","3_48hpa","3_24hpa","3_12hpa","3_uncut"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_12hpaFootMarker.tsv")


markers = FindMarkers(tomo.harm,c("1_24hpa"),c("1_96hpa","1_48hpa","1_12hpa","1_uncut","4_96hpa","4_48hpa","4_24hpa","4_12hpa","4_uncut","3_96hpa","3_48hpa","3_24hpa","3_12hpa","3_uncut"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_24hpaFootMarker.tsv")


markers = FindMarkers(tomo.harm,c("1_48hpa"),c("1_96hpa","1_24hpa","1_12hpa","1_uncut","4_96hpa","4_48hpa","4_24hpa","4_12hpa","4_uncut","3_96hpa","3_48hpa","3_24hpa","3_12hpa","3_uncut"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_48hpaFootMarker.tsv")


markers = FindMarkers(tomo.harm,c("1_96hpa"),c("1_48hpa","1_24hpa","1_12hpa","1_uncut","4_96hpa","4_48hpa","4_24hpa","4_12hpa","4_uncut","3_96hpa","3_48hpa","3_24hpa","3_12hpa","3_uncut"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_96hpaFootMarker.tsv")


#Get time point specific marker

Idents(object = tomo.harm) <- tomo.harm@meta.data$time_point

#uncut/regeneration specific
markers = FindAllMarkers(tomo.harm,only.pos = TRUE,  logfc.threshold = 0.3)

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(cluster , desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_TimePointSpecMarker.tsv")


#Compare regeneration time points against reference

Idents(object = tomo.harm) <- tomo.harm@meta.data$time_point

#12 specific
markers = FindMarkers(tomo.harm,c("12hpa"),c("uncut"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_12vs0Marker.tsv")

#24 specific
markers = FindMarkers(tomo.harm,c("24hpa"),c("uncut"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_24vs0Marker.tsv")

#48 specific
markers = FindMarkers(tomo.harm,c("48hpa"),c("uncut"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_48vs0Marker.tsv")


#96 specific
markers = FindMarkers(tomo.harm,c("96hpa"),c("uncut"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_96vs0Marker.tsv")

#Compare time points by binning time points

Idents(object = tomo.harm) <- tomo.harm@meta.data$time_point

#uncut/regeneration specific
markers = FindMarkers(tomo.harm,c("uncut"),c("12hpa","24hpa","48hpa","96hpa"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_UncutSpecMarker.tsv")

#uncut/12hpa specific
markers = FindMarkers(tomo.harm,c("uncut","12hpa"),c("24hpa","48hpa","96hpa"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_Uncut_12hpa_SpecMarker.tsv")

#12hpa/24hpa specific
markers = FindMarkers(tomo.harm,c("12hpa","24hpa"),c("uncut","48hpa","96hpa"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_12hpa_24hpa_SpecMarker.tsv")

#24hpa/48hpa specific
markers = FindMarkers(tomo.harm,c("24hpa","48hpa"),c("uncut","12hpa","96hpa"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_24hpa_48hpa_SpecMarker.tsv")

#48hpa/96hpa specific
markers = FindMarkers(tomo.harm,c("48hpa","96hpa"),c("24hpa","uncut","12hpa"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_48hpa_96hpa_SpecMarker.tsv")

#uncut/96hpa specific
markers = FindMarkers(tomo.harm,c("uncut","96hpa"),c("48hpa","24hpa","12hpa"))

markers.anno = markers
markers.anno$gene = rownames(markers.anno)
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_uncut_96hpa_SpecMarker.tsv")



gene_annotation = read.csv("./tomo_assays_rna_metafeatures.csv")

Idents(object = tomo.harm) <-  tomo.harm@meta.data$seurat_clusters

markers <- FindAllMarkers(tomo.harm, only.pos = TRUE,  logfc.threshold = 0.3)
markers$diff = markers$pct.1 - markers$pct.2

markers.anno = markers
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id")
markers.anno = markers.anno  %>% arrange(cluster , desc(avg_log2FC))

write_delim(markers.anno,delim = "\t","./tomo_Harmony_AllSeuratClusterMarker.tsv")


```

###Search for proliferation marker

```{r}
#check first how known cell cycle markers are expressed across the embedding

#NVEC200_000589_1.1	top2a
#NVEC200_008052_1.1	hmgb2
#NVEC200_006181_1.1	mki67
#NVEC200_010739_1.1	aurka
#NVEC200_010955_1.1	cenpa
#NVEC200-008980_1.1 cdca8

cc =c("NVEC200-008052.1","NVEC200-006181.1","NVEC200-010739.1","NVEC200-010955.1","NVEC200-008980.1")


pdf("tomo_Harmony_Feature_AllCdk.pdf",width=15,height=10)
FeaturePlot(tomo.harm, pt.size = 2, features = gene_annotation[grep("cdk",gene_annotation$gene_name),"gene_id"],order = T, cols = c(viridis_pal(option = "plasma")(50)), repel=T, min.cutoff = "q1") & NoLegend() & NoAxes()
dev.off()
  

#screen manually nematostella annotation list of g2m genes from seurat

#intersect with var genes

#tomo.harm = FindVariableFeatures(
#  tomo.harm,
#  nfeatures = 5000)

#top10 <- head(VariableFeatures(diatom), 10)

gene_ids =c("NVEC200-0000589.1",
            "NVEC200-008052.1",
            "NVEC200-006181.1",
            "NVEC200-010739.1",
            "NVEC200-010955.1",
            "NVEC200-008980.1",
            "NVEC200-014852.1",
            "NVEC200-005759.1",
            "NVEC200-000144.1",
            "NVEC200-010279.1",
            "NVEC200-014354.1",
            "NVEC200-011097.1")
gene_names =c(
"top2a",
"hmgb2",
"mki67",
"aurka",
"cenpa",
"cdca8",
"cdk1",
"tacc3",
"nusap1",
"anln",
"ndc80",
"ttk")

gg_Fig <- FeaturePlot(tomo.harm, pt.size = 2, features = gene_ids,order = T, cols = c(viridis_pal(option = "plasma")(50)), repel=T, min.cutoff = "q5")
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) & NoLegend() & NoAxes()})

pdf("tomo_Harmony_Feature_ManualCC.pdf",width=15,height=10)
CombinePlots( gg_Fig )
dev.off()


pdf("tomo_Harmony_Feature_AllCdk.pdf",width=15,height=10)
FeaturePlot(tomo.harm, pt.size = 2, features = gene_annotation[grep("cdk",gene_annotation$gene_name),"gene_id"],order = T, cols = c(viridis_pal(option = "plasma")(50)), repel=T, min.cutoff = "q1") & NoLegend() & NoAxes()
dev.off()


#Make a score

tomo.harm = AddModuleScore(tomo.harm,list(cc.score = gene_ids),name= c("cc.score"))
  

gene_ids =c(#"NVEC200-0000589.1",
            "NVEC200-008052.1",
            "NVEC200-006181.1",
            "NVEC200-010739.1",
            "NVEC200-010955.1",
            "NVEC200-008980.1",
            "NVEC200-014852.1",
            "NVEC200-005759.1",
            "NVEC200-000144.1",
            "NVEC200-010279.1",
            "NVEC200-014354.1",
            "NVEC200-011097.1",
            "cc.score1")
gene_names =c(
#"top2a",
"hmgb2",
"mki67",
"aurka",
"cenpa",
"cdca8",
"cdk1",
"tacc3",
"nusap1",
"anln",
"ndc80",
"ttk",
"cc.score")

gg_Fig <- FeaturePlot(tomo.harm, pt.size = 2, features = gene_ids,order = T, cols = c(viridis_pal(option = "plasma")(50)), repel=T, min.cutoff = "q5")
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) & NoLegend() & NoAxes()})

pdf("tomo_Harmony_Feature_ManualCC.pdf",width=15,height=10)
CombinePlots( gg_Fig )
dev.off()



gene_ids =c(#"NVEC200-0000589.1",
            "NVEC200-008052.1",
            "NVEC200-006181.1",
            "NVEC200-010739.1",
            "NVEC200-010955.1",
            "NVEC200-008980.1",
            "NVEC200-014852.1",
            "NVEC200-005759.1",
            "NVEC200-000144.1",
            "NVEC200-010279.1",
            "NVEC200-014354.1",
            "NVEC200-011097.1")
a =c(
#"top2a",
"hmgb2",
"mki67",
"aurka",
"cenpa",
"cdca8",
"cdk1",
"tacc3",
"nusap1",
"anln",
"ndc80",
"ttk")

data = as.matrix(GetAssayData(object = tomo.harm, slot = "scale.data"))
data = data[gene_ids,]
#data = rbind(data,t(FetchData(tomo.harm,"cc.score1")))

hr <- hclust(as.dist(1-cor(t(data),method="spearman")), method="ward")

data = data[match(rownames(data)[hr$order],rownames(data)),]


#downsample and extract names to have an even heatmap
set.seed(24)
tmp.uncut <- subset(tomo.harm, subset = time_point == "uncut")
tmp.uncut = colnames(tmp.uncut[, sample(colnames(tmp.uncut), size =119, replace=F)])
tmp.12hpa <- subset(tomo.harm, subset = time_point == "12hpa")
tmp.12hpa <- colnames(tmp.12hpa[, sample(colnames(tmp.12hpa), size =119, replace=F)])
tmp.24hpa <- subset(tomo.harm, subset = time_point == "24hpa")
tmp.24hpa <- colnames(tmp.24hpa[, sample(colnames(tmp.24hpa), size =119, replace=F)])
tmp.48hpa <- subset(tomo.harm, subset = time_point == "48hpa")
tmp.48hpa <- colnames(tmp.48hpa[, sample(colnames(tmp.48hpa), size =119, replace=F)])
tmp.96hpa <- subset(tomo.harm, subset = time_point == "96hpa")
tmp.96hpa <- colnames(tmp.96hpa[, sample(colnames(tmp.96hpa), size =119, replace=F)])

subsample.slices = c(tmp.uncut,tmp.12hpa,tmp.24hpa,tmp.48hpa,tmp.96hpa)
  
  
#tmp.tomo.harm@meta.data$time_point <- factor(x = tmp.tomo.harm@meta.data$time_point, levels = c("uncut", "12hpa", "24hpa", "48hpa", "96hpa")) # change the order of the factor levels

pdf("tomo_Harmony_ManualCC_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = subset(tomo.harm, cells = subsample.slices), 
              markers = rownames(data),
              sort_var = c("time_point","pseudo_rank"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "YlGnBu","YlGn",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.5,0,1))

plot(hr, labels = NULL, hang = 0.1, 
     main = "Cluster dendrogram")
heatmap.2(as.matrix(data),trace="none",density="none",Colv=F, dendrogram="row",Rowv=as.dendrogram(hr), key=T,scale="none")

dev.off()

data = as.matrix(GetAssayData(object = tomo.harm, slot = "scale.data"))
data = data[gene_ids,]
data = rbind(data,t(FetchData(tomo.harm,"cc.score1")))

gene.labs = c(a,"cc.score")
names(gene.labs) = rownames(data)
#gene.labs = gene.labs[match(rownames(data),names(gene.labs))]

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_ids,"cc.score1")),id.vars=c("time_point","order_time_scale"))

pdf("./tomo_Harmony_ManualCC_LinePlot.pdf",width=10,height=8)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.4)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  

ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_smooth(se = F,span = 0.4)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()

#in heatmap upper branch of tree shows interesting pattern --> plot these genes individually




```


###Plot In situ markers

```{r}
gene_id = c("NVEC200-014078.1", 
"NVEC200-000872.1",
"NVEC200-010351.1",
"NVEC200-012164.1",
"NVEC200-000491.1",
"NVEC200-006580.1",
"NVEC200-015113.1",
"NVEC200-015112.1", 
"NVEC200-001364.1",
"NVEC200-001363.1", 
"NVEC200-001362.1", 
"NVEC200-006274.1",
"NVEC200-013190.1",
"NVEC200-013831.1",
"NVEC200-001043.1",
"NVEC200-005938.1",
"NVEC200-002437.1",
"NVEC200-001872.1")
a = c("AnthoRFamide",
"Anthox7",
"cdh1",
"FoxB",
"GBX",
"M-lim",
"NvnAChRaC",
"NvnAChRaD",
"OtxA",
"OtxB",
"OtxC",
"Sox2",
"SoxE",
"TCF",
"Anthox1a",
"metalloproteinase inhibitor 3",
"uromodulin_a",
"uromodulin_b")


tmp = FetchData(subset(tomo.harm,subset = time_point == "96hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = tmp
tmp = FetchData(subset(tomo.harm,subset = time_point == "48hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)
tmp = FetchData(subset(tomo.harm,subset = time_point == "24hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)
tmp = FetchData(subset(tomo.harm,subset = time_point == "12hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)

tmp = FetchData(subset(tomo.harm,subset = time_point == "uncut"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)

tmp = FetchData(uncut,c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)

meta.order = meta.order[match(colnames(tomo.harm),rownames(meta.order)),]

tomo.harm@meta.data$order_time = meta.order$order
tomo.harm@meta.data$order_time_scale = meta.order$order.scale

pdf("tomo_Harmony_InSituPaperMarker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = gene_id,
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "Spectral" ,"RdYlBu",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.1,0,1))
dev.off()

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_id)),id.vars=c("time_point","order_time_scale"))

gene.labs = a
names(gene.labs) = gene_id

                                                                         
pdf("./tomo_Harmony_InSituPaperMarker_LinePlot.pdf",width=15,height=13)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.5)  + xlab("PT")  + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])) 
dev.off()


#Plot All

a = c("Anthox1a",
"Anthox6",
"Anthox7",
"cdh1",
"cdh3",
"Churchill",
"Ci",
"delta",
"FGF1A",
"FGF8A",
"FGF8B",
"FGFRa",
"FGFRb",
"Fox1",
"FoxB",
"FoxC ",
"FoxD",
"GBX",
"Hedgehog2",
"HLXB9",
"M-lim",
"MMP 3 inhibitor",
"MoxB_a",
"MoxB_b",
"NCam1",
"NvnAChRaB",
"NvnAChRaC",
"NvnAChRaD",
"NvnAChRaU",
"OtxA",
"OtxB",
"OtxC",
"PaxA",
"PaxD3",
"PL10",
"Repo",
"snailA",
"Sox1",
"Sox2",
"Sox3",
"SoxE",
"SoxF1",
"Sprouty",
"TCF",
"Thiamine enzyme",
"wnt 5",
"wnt 7",
"wnt1",
"wnt10",
"wnt11",
"wnt2",
"wnt6",
"wnt8b",
"wntA",
"ZicC",
"ZicD",
"ZicE",
"Uromodulin_a",
"otp",
"wnt16",
"RGM")

gene_id = c("NVEC200-001043.1",
"NVEC200-000874.1",
"NVEC200-000872.1",
"NVEC200-010351.1",
"NVEC200-010468.1",
"NVEC200-006988.1",
"NVEC200-001297.1",
"NVEC200-007533.1",
"NVEC200-014111.1",
"NVEC200-006032.1",
"NVEC200-006030.1",
"NVEC200-006642.1",
"NVEC200-006028.1",
"NVEC200-003168.1",
"NVEC200-012164.1",
"NVEC200-009995.1",
"NVEC200-011699.1",
"NVEC200-000491.1",
"NVEC200-002987.1",
"NVEC200-000883.1",
"NVEC200-006580.1",
"NVEC200-005938.1",
"NVEC200-001459.1",
"NVEC200-001458.1",
"NVEC200-009287.1",
"NVEC200-008102.1",
"NVEC200-015113.1",
"NVEC200-015112.1",
"NVEC200-012338.1",
"NVEC200-001364.1",
"NVEC200-001363.1",
"NVEC200-001362.1",
"NVEC200-013954.1",
"NVEC200-011655.1",
"NVEC200-008902.1",
"NVEC200-009545.1",
"NVEC200-003200.1",
"NVEC200-009182.1",
"NVEC200-006274.1",
"NVEC200-001846.1",
"NVEC200-013190.1",
"NVEC200-004425.1",
"NVEC200-006033.1",
"NVEC200-013831.1",
"NVEC200-014326.1",
"NVEC200-008616.1",
"NVEC200-008615.1",
"NVEC200-001014.1",
"NVEC200-000260.1",
"NVEC200-009768.1",
"NVEC200-008308.1",
"NVEC200-000261.1",
"NVEC200-010238.1",
"NVEC200-008832.1",
"NVEC200-005807.1",
"NVEC200-005805.1",
"NVEC200-005804.1",
"NVEC200-001872.1",
"NVEC200-011709.1",
"NVEC200-008022.1",
"NVEC200-012414.1")


tmp = FetchData(subset(tomo.harm,subset = time_point == "96hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = tmp
tmp = FetchData(subset(tomo.harm,subset = time_point == "48hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)
tmp = FetchData(subset(tomo.harm,subset = time_point == "24hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)
tmp = FetchData(subset(tomo.harm,subset = time_point == "12hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)

tmp = FetchData(uncut,c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)

meta.order = meta.order[match(colnames(tomo.harm),rownames(meta.order)),]

tomo.harm@meta.data$order_time = meta.order$order
tomo.harm@meta.data$order_time_scale = meta.order$order.scale

pdf("tomo_Harmony_InSituMarkerAll_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = gene_id,
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "Spectral" ,"RdYlBu",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.1,0,1))
dev.off()

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_id)),id.vars=c("time_point","order_time_scale"))

gene.labs = a
names(gene.labs) = gene_id

pdf("./tomo_Harmony_InSituMarkerAll_LinePlot.pdf",width=25,height=20)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.5)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])) 
dev.off()

```

##Plot Danila Gaussian marker

```{r}
Gaussian.marker = read.delim("/g/arendt/gerber/Analyses/NematostellaTomoSeq/Danila_Gaussian_Markers.tsv",sep = "\t",header = T)

Gaussian.marker = merge(Gaussian.marker,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
Gaussian.marker = merge(Gaussian.marker,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
Gaussian.marker = merge(Gaussian.marker,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)

write_delim(Gaussian.marker,delim = "\t","/g/arendt/gerber/Analyses/NematostellaTomoSeq/Danila_Gaussian_Markers_Annotation.tsv")

#####
Gaussian.marker = as.data.frame(read_delim(delim = "\t","/g/arendt/gerber/Analyses/NematostellaTomoSeq/Danila_Gaussian_Markers_Annotation.tsv"))
Gaussian.marker = Gaussian.marker[,c("v3.Ontology_term","gene","gene_name_short.x")]
Gaussian.marker = Gaussian.marker[!duplicated(Gaussian.marker),]
Gaussian.marker = Gaussian.marker[Gaussian.marker$gene %in% rownames(tomo.harm),]


tmp = FetchData(subset(tomo.harm,subset = time_point == "96hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = tmp
tmp = FetchData(subset(tomo.harm,subset = time_point == "48hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)
tmp = FetchData(subset(tomo.harm,subset = time_point == "24hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)
tmp = FetchData(subset(tomo.harm,subset = time_point == "12hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)
tmp = FetchData(subset(tomo.harm,subset = time_point == "uncut"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)


#tmp = FetchData(uncut,c("time_point","pseudo_rank"))
#tmp = tmp[order(tmp$pseudo_rank),]
#tmp$order = 1:nrow(tmp)
#tmp$order.scale = tmp$order/max(tmp$order)
#meta.order = rbind(meta.order,tmp,stringsAsFactors = F)

meta.order = meta.order[match(colnames(tomo.harm),rownames(meta.order)),]

tomo.harm@meta.data$order_time = meta.order$order
tomo.harm@meta.data$order_time_scale = meta.order$order.scale

pdf("tomo_Harmony_DanilaGaussianMarker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = Gaussian.marker$gene,
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "Spectral" ,"RdYlBu",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.1,0,1))
dev.off()

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",Gaussian.marker$gene)),id.vars=c("time_point","order_time_scale"))

gene.labs = Gaussian.marker$gene_name_short.x
names(gene.labs) = Gaussian.marker$gene

pdf("./tomo_Harmony_DanilaGaussianMarker_LinePlot.pdf",width=26,height=35)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.5)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])) 
dev.off()


```



####GOseq

```{r}
gtf = read.delim("/g/arendt/gerber/Genomes/Nematostella/NVEC200.20200813.gtf",header = F, sep = "\t")
gtf = gtf[gtf$V3 == "transcript",]
gtf$length = gtf$V5 - gtf$V4
gtf$gene = sapply(strsplit(gtf$V9,split = ";"),"[[",1)
gtf$gene = sapply(strsplit(gtf$gene,split = " "),"[[",2)
gtf$gene = gsub("NVEC200_","NVEC200-",gtf$gene)
gtf$gene = gsub("_1.1",".1",gtf$gene)

length.df = gtf[,c("gene","length")]

#get all markers

markers.all = merge(gene_annotation,blast_annotation, by="gene_id" ,all.x = T,all.y = T)
markers.all = merge(markers.all,v1_annotation, by.x="gene_id" , by.y="Nvec200",all.x = T,all.y = T)
markers.all = markers.all[,c("gene_id","v3.Ontology_term")]
markers.all = na.omit(markers.all)
markers.all = markers.all[!duplicated(markers.all),]

#link many GOs to one gene id on separate rows

GO.list.mod = data.frame(gene_id = NA,v3.Ontology_term = NA)

for (i in 1:nrow(markers.all)) {
  a = unique(strsplit(markers.all[i,"v3.Ontology_term"], "\\,")[[1]])
  if (length(a) > 1) {
    for(j in 1:length(a)) {
      GO.list.mod = rbind(GO.list.mod,c(markers.all[i,"gene_id"],a[j]),stringsAsFactors = F)
    }
    
  } else {
    GO.list.mod = rbind(GO.list.mod,markers.all[i,c("gene_id","v3.Ontology_term")],stringsAsFactors = F)
  }
} 
GO.list.mod = na.omit(GO.list.mod)


#link GO terms with IDs

Gaussian.marker = as.data.frame(read_delim(delim = "\t","/g/arendt/gerber/Analyses/NematostellaTomoSeq/Danila_Gaussian_Markers_Annotation.tsv"))
Gaussian.marker = Gaussian.marker[,c("v3.Ontology_term","gene")]
Gaussian.marker = Gaussian.marker[!duplicated(Gaussian.marker),]

#Prepare for Goseq
assayed.genes = intersect(rownames(tomo.harm),unique(GO.list.mod$gene_id))

gene.vector=as.integer(assayed.genes %in% Gaussian.marker$gene)
names(gene.vector)=assayed.genes
head(gene.vector)

gene.lengths <- length.df$length
names(gene.lengths) <- length.df$gene

# subset by assayed genes and genes with GO.
gene.lengths <- gene.lengths[assayed.genes]
gene.lengths <- gene.lengths[which(names(gene.lengths) %in% unique(GO.list.mod$gene_id))]
#gene.lengths <- gene.lengths[which(names(gene.lengths) %in% unique(GO.list.mod$transcript_ID))]

library(goseq)

pwf <- nullp(gene.vector, bias.data=gene.lengths, plot.fit = F)

go <- goseq(pwf, gene2cat = GO.list.mod)

enriched_go <- go %>% filter(over_represented_pvalue < 0.02)

write.csv(file = "./tomo_Harmony_GOseq_DanilaGaussian.csv",enriched_go[,c("category","over_represented_pvalue","numDEInCat")],row.names = F)

#TUB1 looks promising
#find more correlated genes with the interesting heat shock markers

matrix_mod<-as.matrix(GetAssayData(object = tomo.harm, slot = "data"))
gene<-as.numeric(matrix_mod["NVEC200-003362.1",])
correlations<-sort(apply(matrix_mod,1,function(x){cor(gene,x)}))
correlations = as.data.frame(correlations)
correlations$gene = rownames(correlations)
correlations = merge(correlations,gene_annotation, by.x="gene" , by.y="gene_id")
correlations = merge(correlations,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
correlations = merge(correlations,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
correlations = correlations[order(-correlations$correlations),]  


write_delim(correlations,delim = "\t","./tomo_Harmony_CPA4_corGenes.tsv")
  
# run GoSeq for the cor genes > 0.5
correlations.marker = correlations
correlations.marker = correlations.marker[correlations.marker$correlations > 0.3,c("gene","gene_name_short.x","v3.Ontology_term")]
correlations.marker = correlations.marker[!is.na(correlations.marker$v3.Ontology_term),]
correlations.marker = correlations.marker[!duplicated(correlations.marker),]

#Prepare for Goseq
assayed.genes = intersect(rownames(tomo.harm),unique(GO.list.mod$gene_id))

gene.vector=as.integer(assayed.genes %in% correlations.marker$gene)
names(gene.vector)=assayed.genes
head(gene.vector)

gene.lengths <- length.df$length
names(gene.lengths) <- length.df$gene

# subset by assayed genes and genes with GO.
gene.lengths <- gene.lengths[assayed.genes]
gene.lengths <- gene.lengths[which(names(gene.lengths) %in% unique(GO.list.mod$gene_id))]
#gene.lengths <- gene.lengths[which(names(gene.lengths) %in% unique(GO.list.mod$transcript_ID))]

library(goseq)

pwf <- nullp(gene.vector, bias.data=gene.lengths, plot.fit = F)

go <- goseq(pwf, gene2cat = GO.list.mod)

enriched_go <- go %>% filter(over_represented_pvalue < 0.05)

write.csv(file = "./tomo_Harmony_GOseq_DanilaGaussian_CPA4cor.csv",enriched_go[,c("category","over_represented_pvalue","numDEInCat")],row.names = F)

gene_id = correlations.marker$gene
a = correlations.marker$gene_name_short.x

data = as.matrix(GetAssayData(object = tomo.harm, slot = "scale.data"))
data = data[gene_id,]

hr <- hclust(as.dist(1-cor(t(data),method="spearman")), method="ward")

data = data[match(rownames(data)[hr$order],rownames(data)),]


#downsample and extract names to have an even heatmap
set.seed(24)
tmp.uncut <- subset(tomo.harm, subset = time_point == "uncut")
tmp.uncut = colnames(tmp.uncut[, sample(colnames(tmp.uncut), size =119, replace=F)])
tmp.12hpa <- subset(tomo.harm, subset = time_point == "12hpa")
tmp.12hpa <- colnames(tmp.12hpa[, sample(colnames(tmp.12hpa), size =119, replace=F)])
tmp.24hpa <- subset(tomo.harm, subset = time_point == "24hpa")
tmp.24hpa <- colnames(tmp.24hpa[, sample(colnames(tmp.24hpa), size =119, replace=F)])
tmp.48hpa <- subset(tomo.harm, subset = time_point == "48hpa")
tmp.48hpa <- colnames(tmp.48hpa[, sample(colnames(tmp.48hpa), size =119, replace=F)])
tmp.96hpa <- subset(tomo.harm, subset = time_point == "96hpa")
tmp.96hpa <- colnames(tmp.96hpa[, sample(colnames(tmp.96hpa), size =119, replace=F)])

subsample.slices = c(tmp.uncut,tmp.12hpa,tmp.24hpa,tmp.48hpa,tmp.96hpa)
  
  
#tmp.tomo.harm@meta.data$time_point <- factor(x = tmp.tomo.harm@meta.data$time_point, levels = c("uncut", "12hpa", "24hpa", "48hpa", "96hpa")) # change the order of the factor levels

pdf("tomo_Harmony_GO_Gaussian_CPA4cor_Marker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = subset(tomo.harm, cells = subsample.slices), 
              markers = rownames(data),
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "YlGnBu","YlGn",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.5,0,1))

plot(hr, labels = NULL, hang = 0.1, 
     main = "Cluster dendrogram")
heatmap.2(as.matrix(data),trace="none",density="none",Colv=F, dendrogram="row",Rowv=as.dendrogram(hr), key=T,scale="none")

dev.off()


gene.labs = a
names(gene.labs) = gene_id

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_id)),id.vars=c("time_point","order_time_scale"))

pdf("./tomo_Harmony_GO_Gaussian_CPA4cor_Marker_LinePlot.pdf",width=20,height=20)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.4)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()

```


GOseq on specific GO clusters

```{r}
gtf = read.delim("/g/arendt/gerber/Genomes/Nematostella/NVEC200.20200813.gtf",header = F, sep = "\t")
gtf = gtf[gtf$V3 == "transcript",]
gtf$length = gtf$V5 - gtf$V4
gtf$gene = sapply(strsplit(gtf$V9,split = ";"),"[[",1)
gtf$gene = sapply(strsplit(gtf$gene,split = " "),"[[",2)
gtf$gene = gsub("NVEC200_","NVEC200-",gtf$gene)
gtf$gene = gsub("_1.1",".1",gtf$gene)

length.df = gtf[,c("gene","length")]

#get all markers

markers.all = merge(gene_annotation,blast_annotation, by="gene_id" ,all.x = T,all.y = T)
markers.all = merge(markers.all,v1_annotation, by.x="gene_id" , by.y="Nvec200",all.x = T,all.y = T)
markers.all = markers.all[,c("gene_id","v3.Ontology_term")]
markers.all = na.omit(markers.all)
markers.all = markers.all[!duplicated(markers.all),]

#link many GOs to one gene id on separate rows

GO.list.mod = data.frame(gene_id = NA,v3.Ontology_term = NA)

for (i in 1:nrow(markers.all)) {
  a = unique(strsplit(markers.all[i,"v3.Ontology_term"], "\\,")[[1]])
  if (length(a) > 1) {
    for(j in 1:length(a)) {
      GO.list.mod = rbind(GO.list.mod,c(markers.all[i,"gene_id"],a[j]),stringsAsFactors = F)
    }
    
  } else {
    GO.list.mod = rbind(GO.list.mod,markers.all[i,c("gene_id","v3.Ontology_term")],stringsAsFactors = F)
  }
} 
GO.list.mod = na.omit(GO.list.mod)


#link GO terms with IDs

Gaussian.marker = as.data.frame(read_delim(delim = "\t","/g/arendt/gerber/Analyses/NematostellaTomoSeq/Danila_Gaussian_Markers_C17_Annotation.tsv",col_names = F))
Gaussian.marker = Gaussian.marker[,c("X1","X3")]
Gaussian.marker = Gaussian.marker[!is.na(Gaussian.marker$X3),]

#Prepare for Goseq
assayed.genes = intersect(rownames(tomo.harm),unique(GO.list.mod$gene_id))

gene.vector=as.integer(assayed.genes %in% Gaussian.marker$X1)
names(gene.vector)=assayed.genes
head(gene.vector)

gene.lengths <- length.df$length
names(gene.lengths) <- length.df$gene

# subset by assayed genes and genes with GO.
gene.lengths <- gene.lengths[assayed.genes]
gene.lengths <- gene.lengths[which(names(gene.lengths) %in% unique(GO.list.mod$gene_id))]
#gene.lengths <- gene.lengths[which(names(gene.lengths) %in% unique(GO.list.mod$transcript_ID))]

library(goseq)

pwf <- nullp(gene.vector, bias.data=gene.lengths, plot.fit = F)

go <- goseq(pwf, gene2cat = GO.list.mod)

enriched_go <- go %>% filter(over_represented_pvalue < 0.05)

write.csv(file = "./tomo_Harmony_GOseq_DanilaGaussian_C17.csv",enriched_go[,c("category","over_represented_pvalue","numDEInCat")],row.names = F)

#link GO terms with IDs C16

Gaussian.marker = as.data.frame(read_delim(delim = "\t","/g/arendt/gerber/Analyses/NematostellaTomoSeq/Danila_Gaussian_Markers_C16_Annotation.tsv",col_names = F))
Gaussian.marker = Gaussian.marker[,c("X1","X3")]
Gaussian.marker = Gaussian.marker[!is.na(Gaussian.marker$X3),]

#Prepare for Goseq
assayed.genes = intersect(rownames(tomo.harm),unique(GO.list.mod$gene_id))

gene.vector=as.integer(assayed.genes %in% Gaussian.marker$X1)
names(gene.vector)=assayed.genes
head(gene.vector)

gene.lengths <- length.df$length
names(gene.lengths) <- length.df$gene

# subset by assayed genes and genes with GO.
gene.lengths <- gene.lengths[assayed.genes]
gene.lengths <- gene.lengths[which(names(gene.lengths) %in% unique(GO.list.mod$gene_id))]
#gene.lengths <- gene.lengths[which(names(gene.lengths) %in% unique(GO.list.mod$transcript_ID))]

library(goseq)

pwf <- nullp(gene.vector, bias.data=gene.lengths, plot.fit = F)

go <- goseq(pwf, gene2cat = GO.list.mod)

enriched_go <- go %>% filter(over_represented_pvalue < 0.05)

write.csv(file = "./tomo_Harmony_GOseq_DanilaGaussian_C16.csv",enriched_go[,c("category","over_represented_pvalue","numDEInCat")],row.names = F)

#######plot time point specific genes for GO terms

Gaussian.marker = as.data.frame(read_delim(delim = "\t","/g/arendt/gerber/Analyses/NematostellaTomoSeq/Danila_Gaussian_Markers_C17_Annotation.tsv",col_names = F))
Gaussian.marker = Gaussian.marker[!is.na(Gaussian.marker$X3),]

gene_id = Gaussian.marker$X1
a = Gaussian.marker$X2

data = as.matrix(GetAssayData(object = tomo.harm, slot = "scale.data"))
data = data[gene_id,]

hr <- hclust(as.dist(1-cor(t(data),method="spearman")), method="ward")

data = data[match(rownames(data)[hr$order],rownames(data)),]


#downsample and extract names to have an even heatmap
set.seed(24)
tmp.uncut <- subset(tomo.harm, subset = time_point == "uncut")
tmp.uncut = colnames(tmp.uncut[, sample(colnames(tmp.uncut), size =119, replace=F)])
tmp.12hpa <- subset(tomo.harm, subset = time_point == "12hpa")
tmp.12hpa <- colnames(tmp.12hpa[, sample(colnames(tmp.12hpa), size =119, replace=F)])
tmp.24hpa <- subset(tomo.harm, subset = time_point == "24hpa")
tmp.24hpa <- colnames(tmp.24hpa[, sample(colnames(tmp.24hpa), size =119, replace=F)])
tmp.48hpa <- subset(tomo.harm, subset = time_point == "48hpa")
tmp.48hpa <- colnames(tmp.48hpa[, sample(colnames(tmp.48hpa), size =119, replace=F)])
tmp.96hpa <- subset(tomo.harm, subset = time_point == "96hpa")
tmp.96hpa <- colnames(tmp.96hpa[, sample(colnames(tmp.96hpa), size =119, replace=F)])

subsample.slices = c(tmp.uncut,tmp.12hpa,tmp.24hpa,tmp.48hpa,tmp.96hpa)
  
  
#tmp.tomo.harm@meta.data$time_point <- factor(x = tmp.tomo.harm@meta.data$time_point, levels = c("uncut", "12hpa", "24hpa", "48hpa", "96hpa")) # change the order of the factor levels

pdf("tomo_Harmony_GO_GaussianC17_Marker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = subset(tomo.harm, cells = subsample.slices), 
              markers = rownames(data),
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "YlGnBu","YlGn",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.5,0,1))

plot(hr, labels = NULL, hang = 0.1, 
     main = "Cluster dendrogram")
heatmap.2(as.matrix(data),trace="none",density="none",Colv=F, dendrogram="row",Rowv=as.dendrogram(hr), key=T,scale="none")

dev.off()


gene.labs = a
names(gene.labs) = gene_id

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_id)),id.vars=c("time_point","order_time_scale"))

pdf("./tomo_Harmony_GO_GaussianC17_Marker_LinePlot.pdf",width=10,height=10)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.4)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()

Gaussian.marker = as.data.frame(read_delim(delim = "\t","/g/arendt/gerber/Analyses/NematostellaTomoSeq/Danila_Gaussian_Markers_C16_Annotation.tsv",col_names = F))
Gaussian.marker = Gaussian.marker[!is.na(Gaussian.marker$X3),]

gene_id = Gaussian.marker$X1
a = Gaussian.marker$X2

data = as.matrix(GetAssayData(object = tomo.harm, slot = "scale.data"))
data = data[gene_id,]

hr <- hclust(as.dist(1-cor(t(data),method="spearman")), method="ward")

data = data[match(rownames(data)[hr$order],rownames(data)),]


#downsample and extract names to have an even heatmap
set.seed(24)
tmp.uncut <- subset(tomo.harm, subset = time_point == "uncut")
tmp.uncut = colnames(tmp.uncut[, sample(colnames(tmp.uncut), size =119, replace=F)])
tmp.12hpa <- subset(tomo.harm, subset = time_point == "12hpa")
tmp.12hpa <- colnames(tmp.12hpa[, sample(colnames(tmp.12hpa), size =119, replace=F)])
tmp.24hpa <- subset(tomo.harm, subset = time_point == "24hpa")
tmp.24hpa <- colnames(tmp.24hpa[, sample(colnames(tmp.24hpa), size =119, replace=F)])
tmp.48hpa <- subset(tomo.harm, subset = time_point == "48hpa")
tmp.48hpa <- colnames(tmp.48hpa[, sample(colnames(tmp.48hpa), size =119, replace=F)])
tmp.96hpa <- subset(tomo.harm, subset = time_point == "96hpa")
tmp.96hpa <- colnames(tmp.96hpa[, sample(colnames(tmp.96hpa), size =119, replace=F)])

subsample.slices = c(tmp.uncut,tmp.12hpa,tmp.24hpa,tmp.48hpa,tmp.96hpa)
  
  
#tmp.tomo.harm@meta.data$time_point <- factor(x = tmp.tomo.harm@meta.data$time_point, levels = c("uncut", "12hpa", "24hpa", "48hpa", "96hpa")) # change the order of the factor levels

pdf("tomo_Harmony_GO_GaussianC16_Marker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = subset(tomo.harm, cells = subsample.slices), 
              markers = rownames(data),
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "YlGnBu","YlGn",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.5,0,1))

plot(hr, labels = NULL, hang = 0.1, 
     main = "Cluster dendrogram")
heatmap.2(as.matrix(data),trace="none",density="none",Colv=F, dendrogram="row",Rowv=as.dendrogram(hr), key=T,scale="none")

dev.off()


gene.labs = a
names(gene.labs) = gene_id

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_id)),id.vars=c("time_point","order_time_scale"))

pdf("./tomo_Harmony_GO_GaussianC16_Marker_LinePlot.pdf",width=10,height=10)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.4)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()
 
#TUB1 looks promising
#find more correlated genes with tub1

matrix_mod<-as.matrix(GetAssayData(object = tomo.harm, slot = "data"))
gene<-as.numeric(matrix_mod["NVEC200-000574.1",])
correlations<-sort(apply(matrix_mod,1,function(x){cor(gene,x)}))
correlations = as.data.frame(correlations)
correlations$gene = rownames(correlations)
correlations = merge(correlations,gene_annotation, by.x="gene" , by.y="gene_id")
correlations = merge(correlations,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
correlations = merge(correlations,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
correlations = correlations[order(-correlations$correlations),]  


write_delim(correlations,delim = "\t","./tomo_Harmony_TUBA1D_corGenes.tsv")
 
#look for enrichment of cell type based on Sebe-Pedros et al.

tmp = correlations[correlations$correlations > 0.4,c("gene","gene_name_short.x","marker")]
tmp = tmp[!duplicated(tmp),]

table(tmp$marker)
#cnidocytes     larval     muscle    neurons 
#         3          6          5         16 
table(v1_annotation$marker)
#adult cnidocytes      gland     larval     muscle    neurons 
#   65         59         28         95         80        138

plot.df = melt(data.frame(muscle = 5/80,cnidocytes = 3/59,neurons = 16/138))

#hack to get the wrapping done for pie charts...

cp <- coord_polar(theta = "y")
cp$is_free <- function() TRUE


pdf("tomo_Harmony_TUBA1D_corGenes_CellTypeHistogram.pdf",width=15,height=15)
ggplot(plot.df,aes(x = variable,y = value,fill = variable)) + geom_col() + theme_bw() + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1), panel.border = element_blank(), panel.grid.major = element_blank(),
panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + scale_fill_manual(values = c("goldenrod","darkorchid1","dodgerblue"))
ggplot(plot.df,aes(x = "",y = value,fill = variable)) + geom_bar(width = 1, stat = "identity") + theme_bw()  + cp  +  theme(aspect.ratio = 1) + scale_fill_manual(values = c("goldenrod","darkorchid1","dodgerblue"))
dev.off()




 
# run GoSeq for the cor genes > 0.5
correlations.marker = correlations
correlations.marker = correlations.marker[correlations.marker$correlations > 0.4,c("gene","gene_name_short.x","v3.Ontology_term")]
correlations.marker = correlations.marker[!is.na(correlations.marker$v3.Ontology_term),]
correlations.marker = correlations.marker[!duplicated(correlations.marker),]

#Prepare for Goseq
assayed.genes = intersect(rownames(tomo.harm),unique(GO.list.mod$gene_id))

gene.vector=as.integer(assayed.genes %in% correlations.marker$gene)
names(gene.vector)=assayed.genes
head(gene.vector)

gene.lengths <- length.df$length
names(gene.lengths) <- length.df$gene

# subset by assayed genes and genes with GO.
gene.lengths <- gene.lengths[assayed.genes]
gene.lengths <- gene.lengths[which(names(gene.lengths) %in% unique(GO.list.mod$gene_id))]
#gene.lengths <- gene.lengths[which(names(gene.lengths) %in% unique(GO.list.mod$transcript_ID))]

library(goseq)

pwf <- nullp(gene.vector, bias.data=gene.lengths, plot.fit = F)

go <- goseq(pwf, gene2cat = GO.list.mod)

enriched_go <- go %>% filter(over_represented_pvalue < 0.05)

write.csv(file = "./tomo_Harmony_GOseq_DanilaGaussian_C16_TUB1cor.csv",enriched_go[,c("category","over_represented_pvalue","numDEInCat")],row.names = F)

#plot all genes not only GO term related genes
correlations.marker = correlations
correlations.marker = correlations.marker[correlations.marker$correlations > 0.4,c("gene","gene_name_short.x","v3.Ontology_term")]
#correlations.marker = correlations.marker[!is.na(correlations.marker$v3.Ontology_term),]
correlations.marker = correlations.marker[!duplicated(correlations.marker),]


gene_id = correlations.marker$gene
a = correlations.marker$gene_name_short.x

data = as.matrix(GetAssayData(object = tomo.harm, slot = "scale.data"))
data = data[gene_id,]

hr <- hclust(as.dist(1-cor(t(data),method="spearman")), method="ward")

data = data[match(rownames(data)[hr$order],rownames(data)),]


#downsample and extract names to have an even heatmap
set.seed(24)
tmp.uncut <- subset(tomo.harm, subset = time_point == "uncut")
tmp.uncut = colnames(tmp.uncut[, sample(colnames(tmp.uncut), size =119, replace=F)])
tmp.12hpa <- subset(tomo.harm, subset = time_point == "12hpa")
tmp.12hpa <- colnames(tmp.12hpa[, sample(colnames(tmp.12hpa), size =119, replace=F)])
tmp.24hpa <- subset(tomo.harm, subset = time_point == "24hpa")
tmp.24hpa <- colnames(tmp.24hpa[, sample(colnames(tmp.24hpa), size =119, replace=F)])
tmp.48hpa <- subset(tomo.harm, subset = time_point == "48hpa")
tmp.48hpa <- colnames(tmp.48hpa[, sample(colnames(tmp.48hpa), size =119, replace=F)])
tmp.96hpa <- subset(tomo.harm, subset = time_point == "96hpa")
tmp.96hpa <- colnames(tmp.96hpa[, sample(colnames(tmp.96hpa), size =119, replace=F)])

subsample.slices = c(tmp.uncut,tmp.12hpa,tmp.24hpa,tmp.48hpa,tmp.96hpa)
  
  
#tmp.tomo.harm@meta.data$time_point <- factor(x = tmp.tomo.harm@meta.data$time_point, levels = c("uncut", "12hpa", "24hpa", "48hpa", "96hpa")) # change the order of the factor levels

pdf("tomo_Harmony_GO_GaussianC16_TUB1cor_Marker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = subset(tomo.harm, cells = subsample.slices), 
              markers = rownames(data),
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "YlGnBu","YlGn",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.5,0,1))

plot(hr, labels = NULL, hang = 0.1, 
     main = "Cluster dendrogram")
heatmap.2(as.matrix(data),trace="none",density="none",Colv=F, dendrogram="row",Rowv=as.dendrogram(hr), key=T,scale="none")

dev.off()


gene.labs = a
names(gene.labs) = gene_id

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_id)),id.vars=c("time_point","order_time_scale"))

pdf("./tomo_Harmony_GO_GaussianC16_TUB1cor_Marker_LinePlot.pdf",width=20,height=20)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.4)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()

#c2

Gaussian.marker = as.data.frame(read_delim(delim = "\t","/g/arendt/gerber/Analyses/NematostellaTomoSeq/Danila_Gaussian_Markers_C2_Annotation.tsv",col_names = F))
Gaussian.marker = Gaussian.marker[!duplicated(Gaussian.marker),]

gene_id = Gaussian.marker$X1
a = Gaussian.marker$X2

data = as.matrix(GetAssayData(object = tomo.harm, slot = "scale.data"))
data = data[gene_id,]

hr <- hclust(as.dist(1-cor(t(data),method="spearman")), method="ward")

data = data[match(rownames(data)[hr$order],rownames(data)),]


#downsample and extract names to have an even heatmap
set.seed(24)
tmp.uncut <- subset(tomo.harm, subset = time_point == "uncut")
tmp.uncut = colnames(tmp.uncut[, sample(colnames(tmp.uncut), size =119, replace=F)])
tmp.12hpa <- subset(tomo.harm, subset = time_point == "12hpa")
tmp.12hpa <- colnames(tmp.12hpa[, sample(colnames(tmp.12hpa), size =119, replace=F)])
tmp.24hpa <- subset(tomo.harm, subset = time_point == "24hpa")
tmp.24hpa <- colnames(tmp.24hpa[, sample(colnames(tmp.24hpa), size =119, replace=F)])
tmp.48hpa <- subset(tomo.harm, subset = time_point == "48hpa")
tmp.48hpa <- colnames(tmp.48hpa[, sample(colnames(tmp.48hpa), size =119, replace=F)])
tmp.96hpa <- subset(tomo.harm, subset = time_point == "96hpa")
tmp.96hpa <- colnames(tmp.96hpa[, sample(colnames(tmp.96hpa), size =119, replace=F)])

subsample.slices = c(tmp.uncut,tmp.12hpa,tmp.24hpa,tmp.48hpa,tmp.96hpa)
  
  
#tmp.tomo.harm@meta.data$time_point <- factor(x = tmp.tomo.harm@meta.data$time_point, levels = c("uncut", "12hpa", "24hpa", "48hpa", "96hpa")) # change the order of the factor levels

pdf("tomo_Harmony_GO_GaussianC2_Marker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = subset(tomo.harm, cells = subsample.slices), 
              markers = rownames(data),
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "YlGnBu","YlGn",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.5,0,1))

plot(hr, labels = NULL, hang = 0.1, 
     main = "Cluster dendrogram")
heatmap.2(as.matrix(data),trace="none",density="none",Colv=F, dendrogram="row",Rowv=as.dendrogram(hr), key=T,scale="none")

dev.off()


gene.labs = a
names(gene.labs) = gene_id

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_id)),id.vars=c("time_point","order_time_scale"))

pdf("./tomo_Harmony_GO_GaussianC2_Marker_LinePlot.pdf",width=20,height=20)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.4)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()

```

#####Time point markers

```{r}

gtf = read.delim("/g/arendt/gerber/Genomes/Nematostella/NVEC200.20200813.gtf",header = F, sep = "\t")
gtf = gtf[gtf$V3 == "transcript",]
gtf$length = gtf$V5 - gtf$V4
gtf$gene = sapply(strsplit(gtf$V9,split = ";"),"[[",1)
gtf$gene = sapply(strsplit(gtf$gene,split = " "),"[[",2)
gtf$gene = gsub("NVEC200_","NVEC200-",gtf$gene)
gtf$gene = gsub("_1.1",".1",gtf$gene)

length.df = gtf[,c("gene","length")]

#get all markers

markers.all = merge(gene_annotation,blast_annotation, by="gene_id" ,all.x = T,all.y = T)
markers.all = merge(markers.all,v1_annotation, by.x="gene_id" , by.y="Nvec200",all.x = T,all.y = T)
markers.all = markers.all[,c("gene_id","v3.Ontology_term")]
markers.all = na.omit(markers.all)
markers.all = markers.all[!duplicated(markers.all),]

#link many GOs to one gene id on separate rows

GO.list.mod = data.frame(gene_id = NA,v3.Ontology_term = NA)

for (i in 1:nrow(markers.all)) {
  a = unique(strsplit(markers.all[i,"v3.Ontology_term"], "\\,")[[1]])
  if (length(a) > 1) {
    for(j in 1:length(a)) {
      GO.list.mod = rbind(GO.list.mod,c(markers.all[i,"gene_id"],a[j]),stringsAsFactors = F)
    }
    
  } else {
    GO.list.mod = rbind(GO.list.mod,markers.all[i,c("gene_id","v3.Ontology_term")],stringsAsFactors = F)
  }
} 
GO.list.mod = na.omit(GO.list.mod)


#link GO terms with IDs

early.marker = as.data.frame(read_delim(delim = "\t","/g/arendt/gerber/Analyses/NematostellaTomoSeq/tomo_Harmony_12hpa_24hpa_SpecMarker.tsv"))
early.marker = early.marker[,c("v3.Ontology_term","gene")]
early.marker = early.marker[!duplicated(early.marker),]
early.marker = early.marker[!is.na(early.marker$v3.Ontology_term),]
early.marker = early.marker[1:50,]

mid.marker = as.data.frame(read_delim(delim = "\t","/g/arendt/gerber/Analyses/NematostellaTomoSeq/tomo_Harmony_24hpa_48hpa_SpecMarker.tsv"))
mid.marker = mid.marker[,c("v3.Ontology_term","gene")]
mid.marker = mid.marker[!duplicated(mid.marker),]
mid.marker = mid.marker[!is.na(mid.marker$v3.Ontology_term),]
mid.marker = mid.marker[1:50,]

late.marker = as.data.frame(read_delim(delim = "\t","/g/arendt/gerber/Analyses/NematostellaTomoSeq/tomo_Harmony_48hpa_96hpa_SpecMarker.tsv"))
late.marker = late.marker[,c("v3.Ontology_term","gene")]
late.marker = late.marker[!duplicated(late.marker),]
late.marker = late.marker[!is.na(late.marker$v3.Ontology_term),]
late.marker = late.marker[1:50,]

all.marker = rbind(early.marker,mid.marker,late.marker)

#Prepare for Goseq
assayed.genes = intersect(rownames(tomo.harm),unique(GO.list.mod$gene_id))

gene.vector=as.integer(assayed.genes %in% early.marker$gene)
names(gene.vector)=assayed.genes
head(gene.vector)

gene.lengths <- length.df$length
names(gene.lengths) <- length.df$gene

# subset by assayed genes and genes with GO.
gene.lengths <- gene.lengths[assayed.genes]
gene.lengths <- gene.lengths[which(names(gene.lengths) %in% unique(GO.list.mod$gene_id))]
#gene.lengths <- gene.lengths[which(names(gene.lengths) %in% unique(GO.list.mod$transcript_ID))]

library(goseq)

pwf <- nullp(gene.vector, bias.data=gene.lengths, plot.fit = F)

go <- goseq(pwf, gene2cat = GO.list.mod)

enriched_go <- go %>% filter(over_represented_pvalue < 0.05)

write.csv(file = "./tomo_Harmony_GOseq_EarlyTimePointMarker.csv",enriched_go[,c("category","over_represented_pvalue","numDEInCat")],row.names = F)


#extract genes for corresponding GO terms
#cell communication
mid.marker$gene[mid.marker$v3.Ontology_term %in% mid.marker$v3.Ontology_term[grep("7154",mid.marker$v3.Ontology_term)]]
#"NVEC200-013715.1" "NVEC200-011108.1"
late.marker$gene[late.marker$v3.Ontology_term %in% late.marker$v3.Ontology_term[grep("7154",late.marker$v3.Ontology_term)]]
#"NVEC200-011807.1" "NVEC200-010976.1" "NVEC200-011729.1"

cell.com.GO = c("NVEC200-013715.1","NVEC200-011108.1","NVEC200-011807.1","NVEC200-010976.1","NVEC200-011729.1")


#proteolysis
mid.marker$gene[mid.marker$v3.Ontology_term %in% mid.marker$v3.Ontology_term[grep("0006508",mid.marker$v3.Ontology_term)]]
#"NVEC200-005041.1" "NVEC200-015106.1" "NVEC200-004697.1" "NVEC200-003561.1" "NVEC200-009616.1" "NVEC200-010579.1"

early.marker$gene[early.marker$v3.Ontology_term %in% early.marker$v3.Ontology_term[grep("0006508",early.marker$v3.Ontology_term)]]
#"NVEC200-003560.1" "NVEC200-002718.1" "NVEC200-008973.1" "NVEC200-004697.1" "NVEC200-002840.1" "NVEC200-010579.1" "NVEC200-001955.1" "NVEC200-011019.1"

prot.lys.GO = unique(c("NVEC200-005041.1","NVEC200-015106.1","NVEC200-004697.1","NVEC200-003561.1","NVEC200-009616.1","NVEC200-010579.1","NVEC200-003560.1","NVEC200-002718.1","NVEC200-008973.1","NVEC200-004697.1","NVEC200-002840.1","NVEC200-010579.1","NVEC200-001955.1","NVEC200-011019.1"))

#G protein signallin pathway
late.marker$gene[late.marker$v3.Ontology_term %in% late.marker$v3.Ontology_term[grep("0007186",late.marker$v3.Ontology_term)]]
#"NVEC200-002489.1" "NVEC200-011807.1" "NVEC200-002951.1" "NVEC200-002949.1"

G.sig.GO = c("NVEC200-002489.1","NVEC200-011807.1","NVEC200-002951.1","NVEC200-002949.1")

tomo.harm = AddModuleScore(tomo.harm,list(prot.lys = prot.lys.GO, G.sig = G.sig.GO, cell.com =  cell.com.GO),name= c("prot.lys","G.sig","cell.com"))
  
  
```

####GO genes for paper

```{r}
#######plot time point specific genes for GO terms

gene_id = c(prot.lys.GO,G.sig.GO,cell.com.GO)
a = c("HSP90B1",
"Hspa8",
"HSPA5",
"hsp90a",
"HSPA9",
"HSPA8",
"Calr",
#proteolysis
"CPA2",
"CPD",
"CPA4",
"Cpa2",
"zte25",
"AAEL006169",
"NVEC200-002094.1",
"Dmbt1",
"NVEC200-002344.1",
"Adam12",
"ADAM9",
"Adam12",
"ADAMTS12",
"ADAMTS9",
"NVEC200-003858.1",
"Mmp17",
"NVEC200-008598.1",
"Mmp19",
"NVEC200-001825.1",
"PLG",
"NVEC200-002307.1",
"OVCH1",
"CTRB2",
"Klkb1",
"Tmprss9",
"Ctrb1",
"CTRB1",
"NVEC200-002278.1",
"Tmprss9",
"NVEC200-009896.1",
"CL1",
"SAG12",
"CTSZ",
"NVEC200-010051.1",
"NVEC200-010435.1",
"DPEP1",
#iron ion
"Ferritin",
"Ferritin",
"Ferritin",
"Ferritin")

data = as.matrix(GetAssayData(object = tomo.harm, slot = "scale.data"))
data = data[gene_id,]

hr <- hclust(as.dist(1-cor(t(data),method="spearman")), method="ward")

data = data[match(rownames(data)[hr$order],rownames(data)),]


#downsample and extract names to have an even heatmap
set.seed(24)
tmp.uncut <- subset(tomo.harm, subset = time_point == "uncut")
tmp.uncut = colnames(tmp.uncut[, sample(colnames(tmp.uncut), size =119, replace=F)])
tmp.12hpa <- subset(tomo.harm, subset = time_point == "12hpa")
tmp.12hpa <- colnames(tmp.12hpa[, sample(colnames(tmp.12hpa), size =119, replace=F)])
tmp.24hpa <- subset(tomo.harm, subset = time_point == "24hpa")
tmp.24hpa <- colnames(tmp.24hpa[, sample(colnames(tmp.24hpa), size =119, replace=F)])
tmp.48hpa <- subset(tomo.harm, subset = time_point == "48hpa")
tmp.48hpa <- colnames(tmp.48hpa[, sample(colnames(tmp.48hpa), size =119, replace=F)])
tmp.96hpa <- subset(tomo.harm, subset = time_point == "96hpa")
tmp.96hpa <- colnames(tmp.96hpa[, sample(colnames(tmp.96hpa), size =119, replace=F)])

subsample.slices = c(tmp.uncut,tmp.12hpa,tmp.24hpa,tmp.48hpa,tmp.96hpa)
  
  
#tmp.tomo.harm@meta.data$time_point <- factor(x = tmp.tomo.harm@meta.data$time_point, levels = c("uncut", "12hpa", "24hpa", "48hpa", "96hpa")) # change the order of the factor levels

pdf("tomo_Harmony_GO_TimepointMarker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = subset(tomo.harm, cells = subsample.slices), 
              markers = rownames(data),
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "YlGnBu","YlGn",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.5,0,1))

plot(hr, labels = NULL, hang = 0.1, 
     main = "Cluster dendrogram")
heatmap.2(as.matrix(data),trace="none",density="none",Colv=F, dendrogram="row",Rowv=as.dendrogram(hr), key=T,scale="none")

dev.off()

gene.labs = rownames(data)
names(gene.labs) = rownames(data)
#gene.labs = gene.labs[match(rownames(data),names(gene.labs))]

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale","prot.lys1","G.sig2","cell.com3")),id.vars=c("time_point","order_time_scale"))

pdf("./tomo_Harmony_GO_TimePointMarker_LinePlot.pdf",width=10,height=5)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.4)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()

plot.time.df = data.frame(prot.lys = c(mean(plot.df$value[plot.df$time_point == "96hpa" & plot.df$variable == "prot.lys1"]),mean(plot.df$value[plot.df$time_point == "48hpa" & plot.df$variable == "prot.lys1"]),mean(plot.df$value[plot.df$time_point == "24hpa" & plot.df$variable == "prot.lys1"]),mean(plot.df$value[plot.df$time_point == "12hpa" & plot.df$variable == "prot.lys1"]),mean(plot.df$value[plot.df$time_point == "uncut" & plot.df$variable == "prot.lys1"])) , cell.com = c(mean(plot.df$value[plot.df$time_point == "96hpa" & plot.df$variable == "cell.com3"]),mean(plot.df$value[plot.df$time_point == "48hpa" & plot.df$variable == "cell.com3"]),mean(plot.df$value[plot.df$time_point == "24hpa" & plot.df$variable == "cell.com3"]),mean(plot.df$value[plot.df$time_point == "12hpa" & plot.df$variable == "cell.com3"]),mean(plot.df$value[plot.df$time_point == "uncut" & plot.df$variable == "cell.com3"])) , G.sig = c(mean(plot.df$value[plot.df$time_point == "96hpa" & plot.df$variable == "G.sig2"]),mean(plot.df$value[plot.df$time_point == "48hpa" & plot.df$variable == "G.sig2"]),mean(plot.df$value[plot.df$time_point == "24hpa" & plot.df$variable == "G.sig2"]),mean(plot.df$value[plot.df$time_point == "12hpa" & plot.df$variable == "G.sig2"]),mean(plot.df$value[plot.df$time_point == "uncut" & plot.df$variable == "G.sig2"])),timepoint = c("96hpa","48hpa","24hpa","12hpa","uncut"))

plot.time.df = melt(plot.time.df)
plot.time.df$timepoint = factor(plot.time.df$timepoint, levels = c("uncut","12hpa","24hpa","48hpa","96hpa"))
plot.time.df$order = 1
plot.time.df$order[plot.time.df$timepoint == "12hpa"] = 2
plot.time.df$order[plot.time.df$timepoint == "24hpa"] = 3
plot.time.df$order[plot.time.df$timepoint == "48hpa"] = 4
plot.time.df$order[plot.time.df$timepoint == "96hpa"] = 5

pdf("./tomo_Harmony_GO_TimePointMarker_Time_LinePlot.pdf",width=10,height=5)
ggplot(plot.time.df, aes(x = order,y = value, fill = variable, color = variable)) +theme_bw() + geom_point(alpha = 1) + geom_line()  + xlab("Timepoint")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))  + scale_color_manual(values = c("dodgerblue","forestgreen","violetred"))  
dev.off()

+ facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs))



#####plot location specific genes extracted from GOseq

gene_id = c("NVEC200-003593.1","NVEC200-001262.1","NVEC200-011027.1","NVEC200-007397.1")
a = c("HSP90B1",
"Hspa8",
"HSPA5",
"hsp90a")

tmp = FetchData(subset(tomo.harm,subset = time_point == "96hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = tmp
tmp = FetchData(subset(tomo.harm,subset = time_point == "48hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)
tmp = FetchData(subset(tomo.harm,subset = time_point == "24hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)
tmp = FetchData(subset(tomo.harm,subset = time_point == "12hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)

tmp = FetchData(uncut,c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)

meta.order = meta.order[match(colnames(tomo.harm),rownames(meta.order)),]

tomo.harm@meta.data$order_time = meta.order$order
tomo.harm@meta.data$order_time_scale = meta.order$order.scale

pdf("tomo_Harmony_GO_ProtFoldMarker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = gene_id,
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "Spectral" ,"RdYlBu",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.1,0,1))
dev.off()

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_id)),id.vars=c("time_point","order_time_scale"))

gene.labs = a
names(gene.labs) = gene_id

pdf("./tomo_Harmony_GO_ProtFoldMarker_LinePlot.pdf",width=10,height=10)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.4)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()

gene_id = c("NVEC200-002310.1",
"NVEC200-003004.1",
"NVEC200-003362.1",
"NVEC200-003642.1",
"NVEC200-005409.1",
"NVEC200-005841.1",
"NVEC200-005871.1",
"NVEC200-006198.1",
"NVEC200-007434.1",
"NVEC200-011460.1",
"NVEC200-014047.1",
"NVEC200-012595.1")
a = c("OVCH1",
"CPA2",
"CPA4",
"Cpa2",
"CTRB2",
"Klkb1",
"Tmprss9",
"CTSZ",
"Mmp17",
"Ctrb1",
"Mmp19",
"KDELR2")

pdf("tomo_Harmony_GO_ProtLysMarker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = gene_id,
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "Spectral" ,"RdYlBu",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.1,0,1))
dev.off()

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_id)),id.vars=c("time_point","order_time_scale"))

gene.labs = a
names(gene.labs) = gene_id

pdf("./tomo_Harmony_GO_ProtLysMarker_LinePlot.pdf",width=20,height=15)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.4)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()


#### plot all in one heatmap and line plot


gene_id = c("NVEC200-003593.1","NVEC200-001262.1","NVEC200-011027.1","NVEC200-007397.1","NVEC200-006336.1","NVEC200-001426.1","NVEC200-003470.1",
#proteolysis
"NVEC200-003004.1",
"NVEC200-003314.1",
"NVEC200-003362.1",
"NVEC200-003642.1",
"NVEC200-005933.1",
"NVEC200-005983.1",
"NVEC200-002094.1",
"NVEC200-010891.1",
"NVEC200-002344.1",
"NVEC200-003686.1",
"NVEC200-005649.1",
"NVEC200-005686.1",
"NVEC200-012384.1",
"NVEC200-002960.1",
"NVEC200-003858.1",
"NVEC200-007434.1",
"NVEC200-008598.1",
"NVEC200-014047.1",
"NVEC200-001825.1",
"NVEC200-001883.1",
"NVEC200-002307.1",
"NVEC200-002310.1",
"NVEC200-005409.1",
"NVEC200-005841.1",
"NVEC200-011459.1",
"NVEC200-011460.1",
"NVEC200-001571.1",
"NVEC200-002278.1",
"NVEC200-005871.1",
"NVEC200-009896.1",
"NVEC200-005710.1",
"NVEC200-004469.1",
"NVEC200-006198.1",
"NVEC200-010051.1",
"NVEC200-010435.1",
"NVEC200-010522.1",
#iron ion
"NVEC200-006915.1",
"NVEC200-006947.1",
"NVEC200-009028.1",
"NVEC200-009693.1"
)
a = c("HSP90B1",
"Hspa8",
"HSPA5",
"hsp90a",
"HSPA9",
"HSPA8",
"Calr",
#proteolysis
"CPA2",
"CPD",
"CPA4",
"Cpa2",
"zte25",
"AAEL006169",
"NVEC200-002094.1",
"Dmbt1",
"NVEC200-002344.1",
"Adam12",
"ADAM9",
"Adam12",
"ADAMTS12",
"ADAMTS9",
"NVEC200-003858.1",
"Mmp17",
"NVEC200-008598.1",
"Mmp19",
"NVEC200-001825.1",
"PLG",
"NVEC200-002307.1",
"OVCH1",
"CTRB2",
"Klkb1",
"Tmprss9",
"Ctrb1",
"CTRB1",
"NVEC200-002278.1",
"Tmprss9",
"NVEC200-009896.1",
"CL1",
"SAG12",
"CTSZ",
"NVEC200-010051.1",
"NVEC200-010435.1",
"DPEP1",
#iron ion
"Ferritin",
"Ferritin",
"Ferritin",
"Ferritin")

data = as.matrix(GetAssayData(object = tomo.harm, slot = "scale.data"))
data = data[gene_id,]

hr <- hclust(as.dist(1-cor(t(data),method="spearman")), method="ward")

data = data[match(rownames(data)[hr$order],rownames(data)),]


#downsample and extract names to have an even heatmap
set.seed(24)
tmp.uncut <- subset(tomo.harm, subset = time_point == "uncut")
tmp.uncut = colnames(tmp.uncut[, sample(colnames(tmp.uncut), size =119, replace=F)])
tmp.12hpa <- subset(tomo.harm, subset = time_point == "12hpa")
tmp.12hpa <- colnames(tmp.12hpa[, sample(colnames(tmp.12hpa), size =119, replace=F)])
tmp.24hpa <- subset(tomo.harm, subset = time_point == "24hpa")
tmp.24hpa <- colnames(tmp.24hpa[, sample(colnames(tmp.24hpa), size =119, replace=F)])
tmp.48hpa <- subset(tomo.harm, subset = time_point == "48hpa")
tmp.48hpa <- colnames(tmp.48hpa[, sample(colnames(tmp.48hpa), size =119, replace=F)])
tmp.96hpa <- subset(tomo.harm, subset = time_point == "96hpa")
tmp.96hpa <- colnames(tmp.96hpa[, sample(colnames(tmp.96hpa), size =119, replace=F)])

subsample.slices = c(tmp.uncut,tmp.12hpa,tmp.24hpa,tmp.48hpa,tmp.96hpa)
  
  
#tmp.tomo.harm@meta.data$time_point <- factor(x = tmp.tomo.harm@meta.data$time_point, levels = c("uncut", "12hpa", "24hpa", "48hpa", "96hpa")) # change the order of the factor levels

pdf("tomo_Harmony_GO_AllMarker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = subset(tomo.harm, cells = subsample.slices), 
              markers = rownames(data),
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "YlGnBu","YlGn",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.1,0,1))

plot(hr, labels = NULL, hang = 0.1, 
     main = "Cluster dendrogram")
heatmap.2(as.matrix(data),trace="none",density="none",Colv=F, dendrogram="row",Rowv=as.dendrogram(hr), key=T,scale="none")

dev.off()

gene.labs = a
names(gene.labs) = gene_id
gene.labs = gene.labs[match(rownames(data),names(gene.labs))]

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",rownames(data))),id.vars=c("time_point","order_time_scale"))

pdf("./tomo_Harmony_GO_AllMarker_LinePlot.pdf",width=20,height=15)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.4)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()

#find more correlated genes with the interesting heat shock markers

data = as.matrix(GetAssayData(object = tomo.harm, slot = "data"))
data = data[genes,]
 
matrix_mod<-as.matrix(GetAssayData(object = tomo.harm, slot = "data"))
gene<-as.numeric(matrix_mod["NVEC200-003593.1",])
correlations<-sort(apply(matrix_mod,1,function(x){cor(gene,x)}))
correlations = as.data.frame(correlations)
correlations$gene = rownames(correlations)
correlations = merge(correlations,gene_annotation, by.x="gene" , by.y="gene_id")
correlations = merge(correlations,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
correlations = merge(correlations,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
correlations = correlations[order(-correlations$correlations),]  


write_delim(correlations,delim = "\t","./tomo_Harmony_HSPA90B1_corGenes.tsv")
  
data = as.matrix(GetAssayData(object = tomo.harm, slot = "data"))
data = data[genes,]

matrix_mod<-as.matrix(GetAssayData(object = tomo.harm, slot = "data"))
gene<-as.numeric(matrix_mod["NVEC200-011027.1",])
correlations<-sort(apply(matrix_mod,1,function(x){cor(gene,x)}))
correlations = as.data.frame(correlations)
correlations$gene = rownames(correlations)
correlations = merge(correlations,gene_annotation, by.x="gene" , by.y="gene_id")
correlations = merge(correlations,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
correlations = merge(correlations,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
correlations = correlations[order(-correlations$correlations),]  


write_delim(correlations,delim = "\t","./tomo_Harmony_HSPA5_corGenes.tsv")
  
#plot all markers with more than 0.55 correlation in both

gene_id = c("NVEC200-003593.1",
"NVEC200-011027.1",
"NVEC200-009690.1",
"NVEC200-013550.1",
"NVEC200-004672.1",
"NVEC200-007075.1",
"NVEC200-013045.1",
"NVEC200-012414.1",
"NVEC200-009071.1",
"NVEC200-013744.1",
"NVEC200-012595.1",
"NVEC200-012413.1",
"NVEC200-004252.1",
"NVEC200-000574.1",
"NVEC200-007528.1",
"NVEC200-003470.1",
"NVEC200-008969.1",
"NVEC200-007883.1",
"NVEC200-007397.1",
"NVEC200-001964.1",
"NVEC200-000557.1",
"NVEC200-012144.1",
"NVEC200-002609.1")
a = c("HSP90B1",
"HSPA5",
"NVEC200-009690.1",
"Pdia6",
"NVEC200-004672.1",
"NVEC200-007075.1",
"NVEC200-013045.1",
"RGMB",
"Atpalpha",
"NVEC200-013744.1",
"KDELR2",
"PDIA3",
"SLC5A11",
"TUBA1D",
"Slc3a1",
"Calr",
"ATP1B1",
"Slc6a13",
"hsp90a.1",
"NVEC200-001964.1",
"yxiE",
"REEP5",
"Slc6a15")


data = as.matrix(GetAssayData(object = tomo.harm, slot = "scale.data"))
data = data[gene_id,]

hr <- hclust(as.dist(1-cor(t(data),method="spearman")), method="ward")

data = data[match(rownames(data)[hr$order],rownames(data)),]


#downsample and extract names to have an even heatmap
set.seed(24)
tmp.uncut <- subset(tomo.harm, subset = time_point == "uncut")
tmp.uncut = colnames(tmp.uncut[, sample(colnames(tmp.uncut), size =119, replace=F)])
tmp.12hpa <- subset(tomo.harm, subset = time_point == "12hpa")
tmp.12hpa <- colnames(tmp.12hpa[, sample(colnames(tmp.12hpa), size =119, replace=F)])
tmp.24hpa <- subset(tomo.harm, subset = time_point == "24hpa")
tmp.24hpa <- colnames(tmp.24hpa[, sample(colnames(tmp.24hpa), size =119, replace=F)])
tmp.48hpa <- subset(tomo.harm, subset = time_point == "48hpa")
tmp.48hpa <- colnames(tmp.48hpa[, sample(colnames(tmp.48hpa), size =119, replace=F)])
tmp.96hpa <- subset(tomo.harm, subset = time_point == "96hpa")
tmp.96hpa <- colnames(tmp.96hpa[, sample(colnames(tmp.96hpa), size =119, replace=F)])

subsample.slices = c(tmp.uncut,tmp.12hpa,tmp.24hpa,tmp.48hpa,tmp.96hpa)
  
  
#tmp.tomo.harm@meta.data$time_point <- factor(x = tmp.tomo.harm@meta.data$time_point, levels = c("uncut", "12hpa", "24hpa", "48hpa", "96hpa")) # change the order of the factor levels

pdf("tomo_Harmony_HSP_corMarker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = subset(tomo.harm, cells = subsample.slices), 
              markers = rownames(data),
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "YlGnBu","YlGn",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.1,0,1))

plot(hr, labels = NULL, hang = 0.1, 
     main = "Cluster dendrogram")
heatmap.2(as.matrix(data),trace="none",density="none",Colv=F, dendrogram="row",Rowv=as.dendrogram(hr), key=T,scale="none")

dev.off()

gene.labs = a
names(gene.labs) = gene_id
gene.labs = gene.labs[match(rownames(data),names(gene.labs))]

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",rownames(data))),id.vars=c("time_point","order_time_scale"))

pdf("./tomo_Harmony_HSP_corMarker_LinePlot.pdf",width=20,height=15)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.4)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()



tmp = read.delim("tomo_Harmony_Timepoint_Marker_Plot.tsv",sep = "\t",header = T)
tmp = tmp[!duplicated(tmp[,c("gene.id","gene.name")]),]
gene_id = tmp$gene.id
a = tmp$gene.name

data = as.matrix(GetAssayData(object = tomo.harm, slot = "scale.data"))
data = data[gene_id,]

hr <- hclust(as.dist(1-cor(t(data),method="spearman")), method="ward")

#data = data[match(rownames(data)[hr$order],rownames(data)),]


#downsample and extract names to have an even heatmap
set.seed(24)
tmp.uncut <- subset(tomo.harm, subset = time_point == "uncut")
tmp.uncut = colnames(tmp.uncut[, sample(colnames(tmp.uncut), size =119, replace=F)])
tmp.12hpa <- subset(tomo.harm, subset = time_point == "12hpa")
tmp.12hpa <- colnames(tmp.12hpa[, sample(colnames(tmp.12hpa), size =119, replace=F)])
tmp.24hpa <- subset(tomo.harm, subset = time_point == "24hpa")
tmp.24hpa <- colnames(tmp.24hpa[, sample(colnames(tmp.24hpa), size =119, replace=F)])
tmp.48hpa <- subset(tomo.harm, subset = time_point == "48hpa")
tmp.48hpa <- colnames(tmp.48hpa[, sample(colnames(tmp.48hpa), size =119, replace=F)])
tmp.96hpa <- subset(tomo.harm, subset = time_point == "96hpa")
tmp.96hpa <- colnames(tmp.96hpa[, sample(colnames(tmp.96hpa), size =119, replace=F)])

subsample.slices = c(tmp.uncut,tmp.12hpa,tmp.24hpa,tmp.48hpa,tmp.96hpa)
  
  
#tmp.tomo.harm@meta.data$time_point <- factor(x = tmp.tomo.harm@meta.data$time_point, levels = c("uncut", "12hpa", "24hpa", "48hpa", "96hpa")) # change the order of the factor levels

pdf("tomo_Harmony_Timepoint_Marker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = subset(tomo.harm, cells = subsample.slices), 
              markers = rownames(data),
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "YlGnBu","YlGn",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.5,0,2))

plot(hr, labels = NULL, hang = 0.1, 
     main = "Cluster dendrogram")
heatmap.2(as.matrix(data),trace="none",density="none",Colv=F, dendrogram="row",Rowv=as.dendrogram(hr), key=T,scale="none")

dev.off()

gene.labs = a
names(gene.labs) = gene_id
gene.labs = gene.labs[match(rownames(data),names(gene.labs))]

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",rownames(data))),id.vars=c("time_point","order_time_scale"))

pdf("./tomo_Harmony_Timepoint_Marker_LinePlot.pdf",width=20,height=15)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.4)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()



```


###Plot tip DE marker

```{r}

gene_id = c("NVEC200-014969.1",
"NVEC200-002332.1",
"NVEC200-006858.1",
"NVEC200-008102.1",
"NVEC200-009625.1",
"NVEC200-006651.1",
"NVEC200-012353.1",
"NVEC200-009085.1",
"NVEC200-002734.1",
"NVEC200-012833.1",
"NVEC200-013274.1",
"NVEC200-007856.1",
"NVEC200-008584.1",
"NVEC200-011114.1",
"NVEC200-003350.1",
"NVEC200-005145.1",
"NVEC200-008668.1",
"NVEC200-003556.1",
"NVEC200-002577.1",
"NVEC200-012369.1",
"NVEC200-009386.1",
"NVEC200-007405.1",
"NVEC200-004121.1",
"NVEC200-011056.1",
"NVEC200-009924.1",
"NVEC200-004225.1",
"NVEC200-003643.1",
"NVEC200-012418.1",
"NVEC200-002436.1",
"NVEC200-010038.1",
"NVEC200-008300.1",
"NVEC200-002690.1",
"NVEC200-001872.1",
"NVEC200-009066.1",
"NVEC200-008000.1",
"NVEC200-000769.1",
"NVEC200-008471.1",
"NVEC200-012338.1",
"NVEC200-004471.1",
"NVEC200-012416.1",
"NVEC200-005474.1",
"NVEC200-011554.1",
"NVEC200-006253.1",
"NVEC200-008991.1",
"NVEC200-009456.1",
"NVEC200-012055.1",
"NVEC200-014249.1",
"NVEC200-012375.1",
"NVEC200-009076.1",
"NVEC200-000882.1",
"NVEC200-015196.1",
"NVEC200-003264.1",
"NVEC200-001685.1",
"NVEC200-002739.1",
"NVEC200-001954.1",
"NVEC200-008774.1",
"NVEC200-005628.1",
"NVEC200-002951.1",
"NVEC200-009921.1",
"NVEC200-011064.1")
a = c("COL6A5",
"slc16a10",
"SLC16A3",
"CHRNA7",
"ADAMTS6",
"nas-13",
"Col12a1",
"Fat4",
"blt801",
"TRPA1",
"ADGRD1",
"PTPRD",
"TRPA1",
"CAC",
"Hmcn1",
"Orct",
"DNAH7",
"Mfge8",
"Fbn2",
"COL6A3",
"HMCN1",
"slc16a10",
"EGF1",
"DMBT1",
"Pkd1l2",
"Fat4",
"GLIPR2",
"NAGA",
"GP2",
"P4HTM",
"GLIPR2",
"DMBT1",
"GP2",
"HMCN1",
"Hmcn1",
"col27a1b",
"ADGRL3",
"CHRNA7",
"Dzip3",
"NAGA",
"Hmcn1",
"EGF1",
"TRPA1",
"COL6A5",
"slc16a10",
"Slco4a1",
"TTC28",
"DMBT1",
"ANKRD66",
"col27a1b",
"TRIM45",
"Mfge8",
"CHRNA7",
"blt801",
"nas-13",
"HMCN2",
"ADAMTS6",
"ADGRL3",
"ADGRD1",
"TTC28")


tmp = FetchData(subset(tomo.harm,subset = time_point == "96hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = tmp
tmp = FetchData(subset(tomo.harm,subset = time_point == "48hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)
tmp = FetchData(subset(tomo.harm,subset = time_point == "24hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)
tmp = FetchData(subset(tomo.harm,subset = time_point == "12hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)

tmp = FetchData(uncut,c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)

meta.order = meta.order[match(colnames(tomo.harm),rownames(meta.order)),]

tomo.harm@meta.data$order_time = meta.order$order
tomo.harm@meta.data$order_time_scale = meta.order$order.scale

pdf("tomo_Harmony_TipDEMarker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = gene_id,
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "Spectral" ,"RdYlBu",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.1,0,1))
dev.off()

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_id)),id.vars=c("time_point","order_time_scale"))

gene.labs = a
names(gene.labs) = gene_id

pdf("./tomo_Harmony_TipDEMarker_LinePlot.pdf",width=15,height=13)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.5)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()


#with foot exclusion

gene_id = c("NVEC200-014969.1",
"NVEC200-008102.1",
"NVEC200-009625.1",
"NVEC200-002332.1",
"NVEC200-006651.1",
"NVEC200-014611.1",
"NVEC200-012833.1",
"NVEC200-015068.1",
"NVEC200-002556.1",
"NVEC200-002307.1",
"NVEC200-007405.1",
"NVEC200-009386.1",
"NVEC200-011056.1",
"NVEC200-004121.1",
"NVEC200-004225.1",
"NVEC200-009924.1",
"NVEC200-003643.1",
"NVEC200-012418.1",
"NVEC200-013274.1",
"NVEC200-010038.1",
"NVEC200-008300.1",
"NVEC200-002690.1",
"NVEC200-009066.1",
"NVEC200-008000.1",
"NVEC200-008471.1",
"NVEC200-000769.1",
"NVEC200-012338.1",
"NVEC200-012416.1",
"NVEC200-004471.1",
"NVEC200-013715.1",
"NVEC200-005474.1",
"NVEC200-008991.1",
"NVEC200-006253.1",
"NVEC200-011554.1",
"NVEC200-009456.1",
"NVEC200-014249.1",
"NVEC200-009066.1",
"NVEC200-012055.1",
"NVEC200-013469.1",
"NVEC200-009076.1",
"NVEC200-015196.1",
"NVEC200-003264.1",
"NVEC200-008774.1",
"NVEC200-001685.1",
"NVEC200-001954.1",
"NVEC200-002951.1",
"NVEC200-005628.1",
"NVEC200-002404.1",
"NVEC200-008104.1",
"NVEC200-009921.1")
a = c("COL6A5",
"CHRNA7",
"ADAMTS6",
"slc16a10",
"nas-13",
"TRIM45",
"TRPA1",
"PTPN13",
"NVEC200-002556.1",
"NVEC200-002307.1",
"slc16a10",
"HMCN1",
"DMBT1",
"EGF1",
"Fat4",
"Pkd1l2",
"GLIPR2",
"NAGA",
"ADGRD1",
"P4HTM",
"GLIPR2",
"DMBT1",
"HMCN1",
"Hmcn1",
"ADGRL3",
"col27a1b",
"CHRNA7",
"NAGA",
"Dzip3",
"Fras1",
"Hmcn1",
"COL6A5",
"TRPA1",
"EGF1",
"slc16a10",
"TTC28",
"HMCN1",
"Slco4a1",
"Svep1",
"ANKRD66",
"TRIM45",
"Mfge8",
"HMCN2",
"CHRNA7",
"nas-13",
"ADGRL3",
"ADAMTS6",
"PTPRD",
"Pkd1l2",
"ADGRD1")

pdf("tomo_Harmony_TipDEMarker_wFootExc_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = gene_id,
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "Spectral" ,"RdYlBu",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.1,0,1))
dev.off()

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_id)),id.vars=c("time_point","order_time_scale"))

gene.labs = a
names(gene.labs) = gene_id

pdf("./tomo_Harmony_TipDEMarker_wFootExc_LinePlot.pdf",width=15,height=13)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.5)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()


#plot Danila marker identfied per bulk region

gene_id = c("NVEC200-003805.1",
"NVEC200-003804.1",
"NVEC200-004225.1",
"NVEC200-000800.1",
"NVEC200-013177.1",
"NVEC200-002739.1",
"NVEC200-002154.1",
"NVEC200-003718.1",
"NVEC200-012985.1",
"NVEC200-013420.1",
"NVEC200-005593.1",
"NVEC200-001874.1",
"NVEC200-002875.1")
a = c("LACTBL1","CYP3A24","Fat4","Col1A2","NLRC3","blt801","QRICH2","DET2","gpr54","ALOX5","F5","FAT4","mfsd4b")


pdf("tomo_Harmony_TipDEMarker_Danila_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = gene_id,
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "Spectral" ,"RdYlBu",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.1,0,1))
dev.off()

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_id)),id.vars=c("time_point","order_time_scale"))

gene.labs = a
names(gene.labs) = gene_id

pdf("./tomo_Harmony_TipDEMarker_Danila_LinePlot.pdf",width=15,height=13)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.5)  + xlab("PT") + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()

```

##Plot foot DE marker

```{r}
gene_id = c("NVEC200-006858.1",
"NVEC200-002496.1",
"NVEC200-012353.1",
"NVEC200-002332.1",
"NVEC200-001504.1",
"NVEC200-009085.1",
"NVEC200-012953.1",
"NVEC200-006630.1",
"NVEC200-002734.1",
"NVEC200-002111.1",
"NVEC200-004121.1",
"NVEC200-007405.1",
"NVEC200-011573.1",
"NVEC200-009386.1",
"NVEC200-011056.1",
"NVEC200-004225.1",
"NVEC200-003350.1",
"NVEC200-012638.1",
"NVEC200-009924.1",
"NVEC200-002050.1",
"NVEC200-001872.1",
"NVEC200-002690.1",
"NVEC200-015106.1",
"NVEC200-008300.1",
"NVEC200-003497.1",
"NVEC200-015260.1",
"NVEC200-009450.1",
"NVEC200-012468.1",
"NVEC200-005788.1",
"NVEC200-008433.1",
"NVEC200-002436.1",
"NVEC200-006253.1",
"NVEC200-005474.1",
"NVEC200-009066.1",
"NVEC200-014249.1",
"NVEC200-005041.1",
"NVEC200-000882.1",
"NVEC200-012055.1",
"NVEC200-011554.1",
"NVEC200-009456.1",
"NVEC200-015196.1",
"NVEC200-006417.1",
"NVEC200-002739.1",
"NVEC200-001872.1",
"NVEC200-008774.1",
"NVEC200-014880.1",
"NVEC200-014321.1",
"NVEC200-008104.1",
"NVEC200-006906.1",
"NVEC200-000978.1")
a = c("SLC16A3",
"PF13-0198",
"Col12a1",
"slc16a10",
"ART3",
"Fat4",
"opn4b",
"AM",
"blt801",
"nas-15",
"EGF1",
"slc16a10",
"GP2",
"HMCN1",
"DMBT1",
"Fat4",
"Hmcn1",
"VP1456",
"Pkd1l2",
"TJP1",
"GP2",
"DMBT1",
"nas-15",
"GLIPR2",
"Mfge8",
"Dzip3",
"HMCN2",
"Pkd1l2",
"PF13-0198",
"Moxd1",
"GP2",
"TRPA1",
"Hmcn1",
"HMCN1",
"TTC28",
"nas-15",
"col27a1b",
"Slco4a1",
"EGF1",
"slc16a10",
"TRIM45",
"AM",
"blt801",
"GP2",
"HMCN2",
"PSAP",
"GP2",
"Pkd1l2",
"NPC2",
"Col1a2")


tmp = FetchData(subset(tomo.harm,subset = time_point == "96hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = tmp
tmp = FetchData(subset(tomo.harm,subset = time_point == "48hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)
tmp = FetchData(subset(tomo.harm,subset = time_point == "24hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)
tmp = FetchData(subset(tomo.harm,subset = time_point == "12hpa"),c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)

tmp = FetchData(uncut,c("time_point","pseudo_rank"))
tmp = tmp[order(tmp$pseudo_rank),]
tmp$order = 1:nrow(tmp)
tmp$order.scale = tmp$order/max(tmp$order)
meta.order = rbind(meta.order,tmp,stringsAsFactors = F)

meta.order = meta.order[match(colnames(tomo.harm),rownames(meta.order)),]

tomo.harm@meta.data$order_time = meta.order$order
tomo.harm@meta.data$order_time_scale = meta.order$order.scale

pdf("tomo_Harmony_FootDEMarker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = gene_id,
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "Spectral" ,"RdYlBu",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.1,0,1))
dev.off()

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_id)),id.vars=c("time_point","order_time_scale"))

gene.labs = a
names(gene.labs) = gene_id

pdf("./tomo_Harmony_FootDEMarker_LinePlot.pdf",width=15,height=13)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.5)  + xlab("PT") + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()

#with tip exclusion

gene_id = c("NVEC200-006858.1",
"NVEC200-001504.1",
"NVEC200-003639.1",
"NVEC200-006579.1",
"NVEC200-002734.1",
"NVEC200-010200.1",
"NVEC200-001426.1",
"NVEC200-014058.1",
"NVEC200-002496.1",
"NVEC200-006067.1",
"NVEC200-011573.1",
"NVEC200-004121.1",
"NVEC200-014495.1",
"NVEC200-002436.1",
"NVEC200-009386.1",
"NVEC200-007405.1",
"NVEC200-011056.1",
"NVEC200-006619.1",
"NVEC200-007620.1",
"NVEC200-012638.1",
"NVEC200-001872.1",
"NVEC200-015106.1",
"NVEC200-002690.1",
"NVEC200-014495.1",
"NVEC200-002719.1",
"NVEC200-006417.1",
"NVEC200-008300.1",
"NVEC200-011294.1",
"NVEC200-013773.1",
"NVEC200-015260.1",
"NVEC200-002436.1",
"NVEC200-014858.1",
"NVEC200-005041.1",
"NVEC200-011554.1",
"NVEC200-014535.1",
"NVEC200-006253.1",
"NVEC200-009066.1",
"NVEC200-000882.1",
"NVEC200-012055.1",
"NVEC200-014249.1",
"NVEC200-001872.1",
"NVEC200-006417.1",
"NVEC200-014321.1",
"NVEC200-002739.1",
"NVEC200-014535.1",
"NVEC200-015196.1",
"NVEC200-006906.1",
"NVEC200-002719.1",
"NVEC200-001397.1",
"NVEC200-014880.1")
a = c("SLC16A3",
"ART3",
"NVEC200-003639.1",
"CSRP2",
"blt801",
"v1g167244",
"HSPA8",
"NVEC200-014058.1",
"PF13-0198",
"HK2",
"GP2",
"EGF1",
"NVEC200-014495.1",
"GP2",
"HMCN1",
"slc16a10",
"DMBT1",
"AM",
"FOS-FOX",
"VP1456",
"GP2",
"nas-15",
"DMBT1",
"NVEC200-014495.1",
"GP2",
"AM",
"GLIPR2",
"NVEC200-011294.1",
"SNX11",
"Dzip3",
"GP2",
"NVEC200-014858.1",
"nas-15",
"Hmcn1",
"EGF1",
"NVEC200-014535.1",
"TRPA1",
"HMCN1",
"col27a1b",
"Slco4a1",
"GP2",
"AM",
"GP2",
"blt801",
"NVEC200-014535.1",
"TRIM45",
"NPC2",
"GP2",
"NVEC200-001397.1",
"PSAP")

pdf("tomo_Harmony_FootDEMarker_wTipExc_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = gene_id,
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "Spectral" ,"RdYlBu",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.1,0,1))
dev.off()

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_id)),id.vars=c("time_point","order_time_scale"))

gene.labs = a
names(gene.labs) = gene_id

pdf("./tomo_Harmony_FootDEMarker_wTipExc_LinePlot.pdf",width=15,height=13)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.5)  + xlab("PT") + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()

#diff order

gene_id = c("NVEC200-010080.1",
"NVEC200-005069.1",
"NVEC200-001857.1",
"NVEC200-012966.1",
"NVEC200-004699.1",
"NVEC200-005075.1",
"NVEC200-007209.1",
"NVEC200-009085.1",
"NVEC200-000266.1",
"NVEC200-001353.1",
"NVEC200-004225.1",
"NVEC200-006554.1",
"NVEC200-007405.1",
"NVEC200-005417.1",
"NVEC200-011019.1",
"NVEC200-012103.1",
"NVEC200-006439.1",
"NVEC200-001602.1",
"NVEC200-007425.1",
"NVEC200-011056.1",
"NVEC200-008300.1",
"NVEC200-002154.1",
"NVEC200-007396.1",
"NVEC200-010318.1",
"NVEC200-005593.1",
"NVEC200-009422.1",
"NVEC200-008169.1",
"NVEC200-013715.1",
"NVEC200-006541.1",
"NVEC200-008471.1",
"NVEC200-000849.1",
"NVEC200-005812.1",
"NVEC200-009076.1",
"NVEC200-010651.1",
"NVEC200-009922.1",
"NVEC200-000423.1",
"NVEC200-004068.1",
"NVEC200-001874.1",
"NVEC200-003715.1",
"NVEC200-010106.1",
"NVEC200-013974.1",
"NVEC200-011736.1",
"NVEC200-009921.1",
"NVEC200-005042.1",
"NVEC200-001821.1",
"NVEC200-010581.1",
"NVEC200-004367.1",
"NVEC200-014880.1",
"NVEC200-011945.1",
"NVEC200-003400.1")
a = c("PDPR",
"SEPTIN2",
"mfsd4b",
"ZNF91",
"ALOX5",
"RGS22",
"Tmprss6",
"Fat4",
"TRIM56",
"P65-C",
"Fat4",
"MYC",
"slc16a10",
"YMR196W",
"Tmprss6",
"Ubc",
"TRAF6",
"Fkrp",
"PAPOLA",
"DMBT1",
"GLIPR2",
"QRICH2",
"RYa-R",
"BASS4",
"F5",
"LRRC74A",
"QRFPR",
"Fras1",
"PTGES",
"ADGRL3",
"Orct2",
"Sh",
"ANKRD66",
"SNX16",
"Slc6a19",
"MEP1B",
"DD3-3",
"FAT4",
"w",
"GTSF1",
"ZNF91",
"Srsf4",
"ADGRD1",
"MEP1B",
"pats1",
"nrf-6",
"RP689",
"PSAP",
"Npffr1",
"pdzrn3b")


pdf("tomo_Harmony_FootDEMarker_wTipExc_diff_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = gene_id,
              sort_var = c("time_point","order_time"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "Spectral" ,"RdYlBu",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.1,0,1))
dev.off()

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_id)),id.vars=c("time_point","order_time_scale"))

gene.labs = a
names(gene.labs) = gene_id

pdf("./tomo_Harmony_FootDEMarker_wTipExc_diff_LinePlot.pdf",width=15,height=13)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.5)  + xlab("PT") + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()



```

```{r}
gene_id = c("NVEC200-003593.1","NVEC200-001262.1","NVEC200-011027.1","NVEC200-007397.1")


plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_id)),id.vars=c("time_point","order_time_scale"))

pdf("./tomo_Harmony_test_LinePlot.pdf",width=15,height=13)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.4)  + xlab("PT") + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free") + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()


```



###Plot Danila spatial marker

```{r}
gene.LUT = tomo.harm@assays$RNA@meta.features
danila.heat = read.csv("/g/arendt/gerber/Analyses/NematostellaScRNAseq/202201113_manualClusters.csv")

dan1 = danila.heat[,c("cluster_gene_index","X1")]
dan2 = danila.heat[,c("cluster_gene_index","X2")]
dan3 = danila.heat[,c("cluster_gene_index","X3")]
dan4 = danila.heat[,c("cluster_gene_index","X4")]
dan5 = danila.heat[,c("cluster_gene_index","P")]
dan6 = danila.heat[,c("cluster_gene_index","X6")]
dan7 = danila.heat[,c("cluster_gene_index","X7")]
dan8 = danila.heat[,c("cluster_gene_index","X8")]

dan1 = merge(dan1,gene.LUT,by.x = "X1", by.y = "gene")
dan2 = merge(dan2,gene.LUT,by.x = "X2", by.y = "gene")
dan3 = merge(dan3,gene.LUT,by.x = "X3", by.y = "gene")
dan4 = merge(dan4,gene.LUT,by.x = "X4", by.y = "gene")
dan5 = merge(dan5,gene.LUT,by.x = "P", by.y = "gene")
dan6 = merge(dan6,gene.LUT,by.x = "X6", by.y = "gene")
dan7 = merge(dan7,gene.LUT,by.x = "X7", by.y = "gene")
dan8 = merge(dan8,gene.LUT,by.x = "X8", by.y = "gene")


data = as.matrix(GetAssayData(object = tomo.harm, slot = "data"))
data = data[unique(c(dan1[,"gene_id"],dan2[,"gene_id"],dan3[,"gene_id"],dan4[,"gene_id"],dan5[,"gene_id"],dan6[,"gene_id"],dan7[,"gene_id"],dan8[,"gene_id"])),]
data = data[!duplicated(data),]

#hr <- hclust(as.dist(1-cor(t(data),method="pearson")), method="ward")

#data = data[match(names(sort(cutree(hr,k=8))),rownames(data)),]

tomo.harm@meta.data$barcode = NULL

pdf("tomo_Harmony_DanilaSpatialMarker_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = tomo.harm, 
              markers = unique(rownames(data)),
              sort_var = c("time_point","pseudo_rank"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "Spectral" ,"RdYlBu",c("yellow","red")), hm_colors = viridis_pal(option = "viridis")(3), hm_limit = c(-1,0,1))
dev.off()


#Score for Danila markers


c8_foot <- list(intersect(dan8[,"gene_id"], rownames(tomo.harm)))
tomo.harm <- AddModuleScore(
  object = tomo.harm,
  features = c8_foot,
  name = 'c8_foot'
)
c7_tentacle_base <- list(intersect(dan7[,"gene_id"], rownames(tomo.harm)))
tomo.harm <- AddModuleScore(
  object = tomo.harm,
  features = c7_tentacle_base,
  name = 'c7_tentacle_base'
)
c6_tentacle_base <- list(intersect(dan6[,"gene_id"], rownames(tomo.harm)))
tomo.harm <- AddModuleScore(
  object = tomo.harm,
  features = c6_tentacle_base,
  name = 'c6_tentacle_base'
)
c5_tip <- list(intersect(dan5[,"gene_id"], rownames(tomo.harm)))
tomo.harm <- AddModuleScore(
  object = tomo.harm,
  features = c5_tip,
  name = 'c5_tip'
)
c4_head <- list(intersect(dan4[,"gene_id"], rownames(tomo.harm)))
tomo.harm <- AddModuleScore(
  object = tomo.harm,
  features = c4_head,
  name = 'c4_head'
)
c3_body <- list(intersect(dan3[,"gene_id"], rownames(tomo.harm)))
tomo.harm <- AddModuleScore(
  object = tomo.harm,
  features = c3_body,
  name = 'c3_body'
)

c2_body <- list(intersect(dan2[,"gene_id"], rownames(tomo.harm)))
tomo.harm <- AddModuleScore(
  object = tomo.harm,
  features = c2_body,
  name = 'c2_body'
)
c1_body <- list(intersect(dan1[,"gene_id"], rownames(tomo.harm)))
tomo.harm <- AddModuleScore(
  object = tomo.harm,
  features = c1_body,
  name = 'c1_body'
)

body_sum <- list(intersect(c(dan1[,"gene_id"],dan2[,"gene_id"],dan3[,"gene_id"]), rownames(tomo.harm)))
tomo.harm <- AddModuleScore(
  object = tomo.harm,
  features = body_sum,
  name = 'body_sum'
)

tentacle_sum_all <- list(intersect(c(dan5[,"gene_id"],dan6[,"gene_id"],dan7[,"gene_id"]), rownames(tomo.harm)))
tomo.harm <- AddModuleScore(
  object = tomo.harm,
  features = tentacle_sum_all,
  name = 'tentacle_sum_all'
)

tentacle_sum_noTip <- list(intersect(c(dan6[,"gene_id"],dan7[,"gene_id"]), rownames(tomo.harm)))
tomo.harm <- AddModuleScore(
  object = tomo.harm,
  features = tentacle_sum_noTip,
  name = 'tentacle_sum_noTip'
)


pdf("tomo_Harmony_DanilaSpatialMarkerScores_Feature.pdf",width=10,height=10)
FeaturePlot(tomo.harm, reduction = 'umap', pt.size = 1, features = c("c8_foot1","c7_tentacle_base1","c6_tentacle_base1","c5_tip1","c4_head1","c3_body1","c2_body1","c1_body1","body_sum1","tentacle_sum_all1","tentacle_sum_noTip1"),order = T, cols = c("gray",rev(viridis_pal(option = "viridis")(10)))) & NoLegend() & NoAxes()
dev.off()

```

"CPA4",
"DNAH6",
"ALOX5",
"slc46a1",
"DNAH3",
"NKX6-2",
"amt-1",
"prom1a",
"Sh",
"Gpx5",
"slc16a10",
"OTX1",
"SACS",
"Fras1",
"MYC",
"TLL1",
"STAB2",
"VPS13D",
"Tmtc4",
"MFGE8",
"Ncam2",
"Ggt1",
"TRAF6"



"TRAF6",
"P4HTM",
"PCK2",
"TTN",
"slc15a2",
"MYC",
"slc16a10",
"DMBT1",
"Pkd1l2",
"Fras1",
"DNAH6",
"Fat4",
"Glipr2",
"TJP1",
"rnf213a",
"VP1456",
"SIDT1",
"XDH",
"Flna",
"Ptchd3"

"AM",
"COL6A5",
"CHRNA7",
"Fat4",
"Col12a1",
"blt801",
"slc16a10",
"GP2",
"col27a1b",
"TRPA1",
"CAC",
"Fbn2",
"DNAH7",
"Fat4",
###Plot along gradients

```{r}


meta.data = FetchData(tomo.harm,c("pseudo_rank","time_point",
#c1 butt
"NVEC200-009271.1",
"NVEC200-014866.1",
"NVEC200-013033.1",
"NVEC200-006430.1",
"NVEC200-006423.1",
"NVEC200-010166.1",
#c0 center
"NVEC200-001639.1",
"NVEC200-000819.1",
"NVEC200-014003.1",
"NVEC200-003858.1",
"NVEC200-015133.1",
"NVEC200-002032.1",
#c2 pharynx?
"NVEC200-014418.1",
"NVEC200-003926.1",
"NVEC200-000574.1",
"NVEC200-006121.1",
"NVEC200-004712.1",
"NVEC200-001561.1",
#c3 tip
"NVEC200-007938.1",
"NVEC200-012562.1",
"NVEC200-006310.1",
"NVEC200-015153.1",
"NVEC200-000708.1",
"NVEC200-000772.1",
"NVEC200-012071.1"))

#cyb = "NVEC200-012071.1"
library(RColorBrewer)
library(ggplot2)


plot_list=list()

plot_list <- lapply(colnames(meta.data[,3:ncol(meta.data)]), function(i) { 
  p <- ggplot(meta.data, aes(x = pseudo_rank,y = meta.data[,i] )) +theme_bw() + geom_smooth( se=FALSE, size=1, method = 'loess', span = 2)  + xlab("PT") + ylab("Expression") + ggtitle(i)   + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black") , legend.position = "none") 
})
#+  scale_y_continuous(limits = c(NA, NA))

library(gridExtra)

pdf("./tomo_Harmony_Embedding_PT_ClusterMarker.pdf",width=15,height=20)

do.call("grid.arrange",plot_list)
 
dev.off()


#plot time point distribution along PT ranks

pdf("./tomo_Harmony_Embedding_PT_time_point.pdf",width=10,height=5)
ggplot(meta.data, aes(x = pseudo_rank,fill = time_point, color = time_point)) +theme_bw() + geom_histogram(position="fill",bins = 50)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) 
ggplot(meta.data, aes(x = pseudo_rank)) +theme_bw() + geom_density(aes(color=factor(time_point) ))  + xlab("PT")   + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) 
dev.off()


#plot cluster marker gene expression on jitter split by time point and ordered by PT rank

plot_list=list()

plot_list <- lapply(colnames(meta.data[,3:ncol(meta.data)]), function(i) { 
  p <- ggplot(meta.data, aes(y = pseudo_rank,x = time_point)) +theme_bw() + geom_jitter(aes(color = meta.data[,i]),cex = 1)  + ylab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.position = "none", axis.line = element_line(colour = "black"))  + scale_colour_gradientn(colors = c("gray",rev(viridis_pal(option = "viridis")(12)))) 
})

library(gridExtra)

pdf("./tomo_Harmony_Embedding_PT_Jitter_ClusterMarker.pdf",width=20,height=20)

ggplot(meta.data, aes(y = pseudo_rank,x = time_point, color = time_point)) +theme_bw() + geom_jitter(cex = 5)  + ylab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) 

do.call("grid.arrange",plot_list)
 
dev.off()

#get time point marker

meta.data = FetchData(tomo.harm,c("pseudo_rank","time_point",
#uncut
"NVEC200-005075.1",
"NVEC200-001708.1",
"NVEC200-006970.1",
"NVEC200-008041.1",
"NVEC200-005069.1",
"NVEC200-008352.1",
#12h
"NVEC200-004225.1",
"NVEC200-006554.1",
"NVEC200-011019.1",
"NVEC200-001602.1",
"NVEC200-002988.1",
"NVEC200-012103.1",
#24h
"NVEC200-007396.1",
"NVEC200-010318.1",
"NVEC200-000231.1",
"NVEC200-008169.1",
"NVEC200-001061.1",
"NVEC200-001482.1",
#48h
"NVEC200-005812.1",
"NVEC200-000423.1",
"NVEC200-003715.1",
"NVEC200-010651.1",
"NVEC200-012055.1",
"NVEC200-009922.1",
#96h
"NVEC200-013974.1",
"NVEC200-009921.1",
"NVEC200-011736.1",
"NVEC200-004367.1",
"NVEC200-010581.1",
"NVEC200-005042.1"))


#plot time point marker gene expression on jitter split by time point and ordered by PT rank

plot_list=list()

plot_list <- lapply(colnames(meta.data[,3:ncol(meta.data)]), function(i) { 
  p <- ggplot(meta.data, aes(y = pseudo_rank,x = time_point)) +theme_bw() + geom_jitter(aes(color = meta.data[,i]),cex = 1)  + ylab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.position = "none", axis.line = element_line(colour = "black"))  + scale_colour_gradientn(colors = c("gray",rev(viridis_pal(option = "viridis")(12)))) 
})

library(gridExtra)

pdf("./tomo_Harmony_Embedding_PT_Jitter_TimePointMarker.pdf",width=20,height=20)

ggplot(meta.data, aes(y = pseudo_rank,x = time_point, color = time_point)) +theme_bw() + geom_jitter(cex = 5)  + ylab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) 

do.call("grid.arrange",plot_list)
 
dev.off()

```

##CIBERSORTx

```{r}
cibersort = read.csv("/g/arendt/gerber/Analyses/NematostellaTomoSeq/CIBERSORTx_Job60_Results.csv")


scale.plot = cibersort[,c("c0","c1","c2","c3","c4","c5","c6","c7","c8","c9","c10","c11","c12")]
scale.plot[scale.plot==0]<-NA
scale.plot<-scale(scale.plot)
scale.plot[scale.plot>2]<-2
scale.plot[scale.plot<(-2)]<-(-2)
scale.plot[is.na(scale.plot)]<-(-2)
scale.plot = as.data.frame(scale.plot)




cibersort.norm = cibersort

cibersort.norm[,c("c0","c1","c2","c3","c4","c5","c6","c7","c8","c9","c10","c11","c12")] = scale.plot[,c("c0","c1","c2","c3","c4","c5","c6","c7","c8","c9","c10","c11","c12")]

tmp = FetchData(tomo.harm,c("pseudo_rank","ident"))
tmp$Mixture = rownames(tmp)

cibersort = merge(tmp,cibersort,by = "Mixture")
cibersort.norm = merge(tmp,cibersort.norm,by = "Mixture")

plot2.df = melt(cibersort.norm[,c("Mixture","c0","c1","c2","c3","c4","c5","c6","c7","c8","c9","c10","c11","c12")])
plot2.df = merge(plot2.df,cibersort.norm[,c("Mixture","pseudo_rank","Correlation","RMSE")])

plot.df = melt(cibersort[,c("Mixture","c0","c1","c2","c3","c4","c5","c6","c7","c8","c9","c10","c11","c12")])
plot.df = merge(plot.df,cibersort[,c("Mixture","pseudo_rank","Correlation","RMSE")])



pdf("./tomo_cibersort_result60.pdf",width=10,height=10)
ggplot(plot2.df, aes(fill=variable, y=value, x=pseudo_rank)) + 
    geom_bar(position="stack", stat="identity")
ggplot(plot2.df, aes(fill=variable,color=variable, y=value, x=pseudo_rank)) + 
    geom_smooth(se = F,span = 1)
ggplot(plot.df, aes(fill=variable, y=value, x=pseudo_rank)) + 
    geom_bar(position="stack", stat="identity")
ggplot(plot.df, aes(fill=variable,color=variable, y=value, x=pseudo_rank)) + 
    geom_smooth(se = F,span = 1)
ggplot(plot2.df, aes(fill=variable, y=Correlation, x=pseudo_rank)) + 
    geom_bar(position="stack", stat="identity")
ggplot(plot2.df, aes(fill=variable, y=RMSE, x=pseudo_rank)) + 
    geom_bar(position="stack", stat="identity")
dev.off()

#only uncut b2-42

cibersort = read.csv("/g/arendt/gerber/Analyses/NematostellaTomoSeq/CIBERSORTx_Job60_Results.csv")


tmp = FetchData(tomo.harm,c("pseudo_rank","ident","slice_index","sample_name"))
tmp = tmp[tmp$sample_name == "b2-42-uncut",]
tmp$Mixture = rownames(tmp)
tmp = tmp[order(tmp$slice_index),]
tmp$slice = 1:nrow(tmp)

cibersort = cibersort[cibersort$Mixture %in% tmp$Mixture,]

scale.plot = cibersort[,c("c0","c1","c2","c3","c4","c5","c6","c7","c8","c9","c10","c11","c12")]
scale.plot[scale.plot==0]<-NA
scale.plot<-scale(scale.plot)
scale.plot[scale.plot>2]<-2
scale.plot[scale.plot<(-2)]<-(-2)
scale.plot[is.na(scale.plot)]<-(-2)
scale.plot = as.data.frame(scale.plot)

cibersort.norm = cibersort

cibersort.norm[,c("c0","c1","c2","c3","c4","c5","c6","c7","c8","c9","c10","c11","c12")] = scale.plot[,c("c0","c1","c2","c3","c4","c5","c6","c7","c8","c9","c10","c11","c12")]

cibersort = merge(tmp,cibersort,by = "Mixture")
cibersort.norm = merge(tmp,cibersort.norm,by = "Mixture")

plot2.df = melt(cibersort.norm[,c("Mixture","c0","c1","c2","c3","c4","c5","c6","c7","c8","c9","c10","c11","c12")])
plot2.df = merge(plot2.df,cibersort.norm[,c("Mixture","pseudo_rank","slice","Correlation","RMSE")])

plot.df = melt(cibersort[,c("Mixture","c0","c1","c2","c3","c4","c5","c6","c7","c8","c9","c10","c11","c12")])
plot.df = merge(plot.df,cibersort[,c("Mixture","pseudo_rank","slice","Correlation","RMSE")])

pdf("./tomo_cibersort_uncut_result60.pdf",width=10,height=10)
ggplot(plot2.df, aes(fill=variable, y=value, x=slice)) + 
    geom_bar(position="stack", stat="identity")
ggplot(plot2.df, aes(fill=variable,color=variable, y=value, x=slice)) + 
    geom_smooth(se = F,span = 1)
ggplot(plot.df, aes(fill=variable, y=value, x=slice)) + 
    geom_bar(position="stack", stat="identity")
ggplot(plot.df, aes(fill=variable,color=variable, y=value, x=slice)) + 
    geom_smooth(se = F,span = 1)
ggplot(plot2.df, aes(fill=variable, y=Correlation, x=slice)) + 
    geom_bar(position="stack", stat="identity")
ggplot(plot2.df, aes(fill=variable, y=RMSE, x=slice)) + 
    geom_bar(position="stack", stat="identity")
dev.off()

#read in cibersort result that had only marker gene list as input
#only uncut b2-42

cibersort = read.csv("/g/arendt/gerber/Analyses/NematostellaTomoSeq/CIBERSORTx_Job67_Results.csv")


tmp = FetchData(tomo.harm,c("pseudo_rank","ident","slice_index","sample_name"))
tmp = tmp[tmp$sample_name == "b2-42-uncut",]
tmp$Mixture = rownames(tmp)
tmp = tmp[order(tmp$slice_index),]
tmp$slice = 1:nrow(tmp)

cibersort = cibersort[cibersort$Mixture %in% tmp$Mixture,]

scale.plot = cibersort[,c("c0","c1","c2","c3","c4","c5","c6","c7","c8","c9","c10","c11","c12","c13")]
scale.plot[scale.plot==0]<-NA
scale.plot<-scale(scale.plot)
scale.plot[scale.plot>2]<-2
scale.plot[scale.plot<(-2)]<-(-2)
scale.plot[is.na(scale.plot)]<-(-2)
scale.plot = as.data.frame(scale.plot)

cibersort.norm = cibersort

cibersort.norm[,c("c0","c1","c2","c3","c4","c5","c6","c7","c8","c9","c10","c11","c12","c13")] = scale.plot[,c("c0","c1","c2","c3","c4","c5","c6","c7","c8","c9","c10","c11","c12","c13")]

cibersort = merge(tmp,cibersort,by = "Mixture")
cibersort.norm = merge(tmp,cibersort.norm,by = "Mixture")

plot2.df = melt(cibersort.norm[,c("Mixture","c0","c1","c2","c3","c4","c5","c6","c7","c8","c9","c10","c11","c12","c13")])
plot2.df = merge(plot2.df,cibersort.norm[,c("Mixture","pseudo_rank","slice","Correlation","RMSE")])

plot.df = melt(cibersort[,c("Mixture","c0","c1","c2","c3","c4","c5","c6","c7","c8","c9","c10","c11","c12","c13")])
plot.df = merge(plot.df,cibersort[,c("Mixture","pseudo_rank","slice","Correlation","RMSE")])

pdf("./tomo_cibersort_uncut_result67.pdf",width=10,height=10)
ggplot(plot2.df, aes(fill=variable, y=value, x=slice)) + 
    geom_bar(position="stack", stat="identity")
ggplot(plot2.df, aes(fill=variable,color=variable, y=value, x=slice)) + 
    geom_smooth(se = F,span = 1)
ggplot(plot.df, aes(fill=variable, y=value, x=slice)) + 
    geom_bar(position="stack", stat="identity")
ggplot(plot.df, aes(fill=variable,color=variable, y=value, x=slice)) + 
    geom_smooth(se = F,span = 1)
ggplot(plot2.df, aes(fill=variable, y=Correlation, x=slice)) + 
    geom_bar(position="stack", stat="identity")
ggplot(plot2.df, aes(fill=variable, y=RMSE, x=slice)) + 
    geom_bar(position="stack", stat="identity")
dev.off()


#plot mitosis score as heatmap for b2_42


tmp = FetchData(tomo.harm,c("pseudo_rank","slice_index","sample_name","mitosis1",mitosis.df$gene))
tmp = tmp[tmp$sample_name == "b2-42-uncut",]
tmp$Mixture = rownames(tmp)
tmp = tmp[order(tmp$slice_index),]
tmp$slice = 1:nrow(tmp)


pdf("./tomo_uncut_LinePlot_mitosisScore.pdf",width=10,height=10)
ggplot(tmp, aes(y=mitosis1, x=slice)) + 
    geom_bar(position="stack", stat="identity") + 
    geom_smooth(se = F,span = 1)
dev.off()

#plot heatmap
cor.sc.bulk = as.matrix(t(tmp[,mitosis.df$gene]))
  
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 49)
#my_palette <- c(rep("Blue", 125), my_palette, rep("Red", 125))

pdf("./tomo_uncut_Heatmap_mitosisScoreGenes.pdf",width=15,height=15)
gplots::heatmap.2(cor.sc.bulk, scale="row",dendrogram = "none", #keysize = 0.3,
                  Rowv = F,Colv = F,   
                  trace = "none" ,  key = T, 
                  col = my_palette)
                  #, RowSideColors = sidecolors)
gplots::heatmap.2(cor.sc.bulk, scale="none",dendrogram = "none", #keysize = 0.3,
                  Rowv = F,Colv = F,   
                  trace = "none" ,  key = T, 
                  col = my_palette)
                  #, RowSideColors = sidecolors)
dev.off()

```

##Nematostella visualization
```{r}


set.seed(seed = 300)
meta.data = FetchData(tomo.harm,c("pseudo_rank","time_point","NVEC200-003903.1",
"NVEC200-006089.1",
"NVEC200-003854.1",
"NVEC200-008375.1",
"NVEC200-003672.1",
"NVEC200-006641.1",
"NVEC200-015111.1",
"NVEC200-005216.1",
"NVEC200-011036.1",
"NVEC200-014929.1", 
"NVEC200-006163.1",
"NVEC200-014405.1",
"NVEC200-009518.1",
"NVEC200-014970.1",
"NVEC200-012414.1",
"NVEC200-009923.1",
"NVEC200-011585.1"))


meta.data$jitter = runif(nrow(meta.data), 1, 2)
meta.data$jitter[meta.data$pseudo_rank %in% c(1:30)] = runif(30, 1.1, 1.9)
meta.data$jitter[meta.data$pseudo_rank %in% c(31:50)] = runif(20, 0.9, 2.1)
meta.data$jitter[meta.data$pseudo_rank %in% c(51:120)] = runif(70, 0.8, 2.2)
meta.data$jitter[meta.data$pseudo_rank %in% c(121:150)] = runif(30, 0.9, 2.1)
meta.data$jitter[meta.data$pseudo_rank %in% c(151:180)] = runif(30, 1.1, 1.9)

meta.data$jitter[meta.data$pseudo_rank %in% c(701:750)] = sample(c(runif(nrow(meta.data), 1, 1.3), runif(nrow(meta.data), 1.7, 2)), 50, replace=T) 
meta.data$jitter[meta.data$pseudo_rank %in% c(751:800)] = sample(c(runif(nrow(meta.data), 0.8, 1.1), runif(nrow(meta.data), 1.9, 2.2)), 50, replace=T) 
meta.data$jitter[meta.data$pseudo_rank %in% c(801:850)] = sample(c(runif(nrow(meta.data), 0.6, 0.9), runif(nrow(meta.data), 2.1, 2.4)), 50, replace=T) 
meta.data$jitter[meta.data$pseudo_rank %in% c(851:900)] = sample(c(runif(nrow(meta.data), 0.4, 0.7), runif(nrow(meta.data), 2.3, 2.6)), 50, replace=T) 
meta.data$jitter[meta.data$pseudo_rank %in% c(901:950)] = sample(c(runif(nrow(meta.data), 0.2, 0.5), runif(nrow(meta.data), 2.5, 2.8)), 50, replace=T) 
meta.data$jitter[meta.data$pseudo_rank %in% c(951:nrow(meta.data))] = sample(c(runif(nrow(meta.data), 0, 0.3), runif(nrow(meta.data), 2.7, 3)), 7, replace=T) 


meta.data$x1 = 1
meta.data$x1[meta.data$pseudo_rank %in% c(1:10)] = 1.1
meta.data$x1[meta.data$pseudo_rank %in% c(11:20)] = 1.05
meta.data$x1[meta.data$pseudo_rank %in% c(31:40)] = 0.95
meta.data$x1[meta.data$pseudo_rank %in% c(41:50)] = 0.9
meta.data$x1[meta.data$pseudo_rank %in% c(51:60)] = 0.85
meta.data$x1[meta.data$pseudo_rank %in% c(61:70)] = 0.825
meta.data$x1[meta.data$pseudo_rank %in% c(71:120)] = 0.8
meta.data$x1[meta.data$pseudo_rank %in% c(121:130)] = 0.825
meta.data$x1[meta.data$pseudo_rank %in% c(131:200)] = 0.85
meta.data$x1[meta.data$pseudo_rank %in% c(201:210)] = 0.875
meta.data$x1[meta.data$pseudo_rank %in% c(211:300)] = 0.9
meta.data$x1[meta.data$pseudo_rank %in% c(301:310)] = 0.925
meta.data$x1[meta.data$pseudo_rank %in% c(311:400)] = 0.95
meta.data$x1[meta.data$pseudo_rank %in% c(401:600)] = 0.975

meta.data$x1[meta.data$pseudo_rank %in% c(651:660)] = 1
meta.data$x1[meta.data$pseudo_rank %in% c(661:670)] =0.9
meta.data$x1[meta.data$pseudo_rank %in% c(671:680)] = 0.8
meta.data$x1[meta.data$pseudo_rank %in% c(681:700)] = 0.7
meta.data$x1[meta.data$pseudo_rank %in% c(701:725)] = 0.6
meta.data$x1[meta.data$pseudo_rank %in% c(726:750)] = 0.5
meta.data$x1[meta.data$pseudo_rank %in% c(751:775)] = 0.4
meta.data$x1[meta.data$pseudo_rank %in% c(776:800)] = 0.3
meta.data$x1[meta.data$pseudo_rank %in% c(801:825)] = 0.2
meta.data$x1[meta.data$pseudo_rank %in% c(826:850)] = 0.1
meta.data$x1[meta.data$pseudo_rank %in% c(851:875)] = 0
meta.data$x1[meta.data$pseudo_rank %in% c(876:950)] = -0.1
meta.data$x1[meta.data$pseudo_rank %in% c(951:960)] = -0.075
meta.data$x1[meta.data$pseudo_rank %in% c(961:nrow(meta.data))] = -0.05

meta.data$x2 = 2
meta.data$x2[meta.data$pseudo_rank %in% c(1:10)] = 1.9
meta.data$x2[meta.data$pseudo_rank %in% c(11:20)] = 1.95
meta.data$x2[meta.data$pseudo_rank %in% c(31:40)] = 2.05
meta.data$x2[meta.data$pseudo_rank %in% c(41:50)] = 2.1
meta.data$x2[meta.data$pseudo_rank %in% c(51:60)] = 2.15
meta.data$x2[meta.data$pseudo_rank %in% c(61:70)] = 2.175
meta.data$x2[meta.data$pseudo_rank %in% c(71:120)] = 2.2
meta.data$x2[meta.data$pseudo_rank %in% c(121:130)] = 2.175
meta.data$x2[meta.data$pseudo_rank %in% c(131:200)] = 2.15
meta.data$x2[meta.data$pseudo_rank %in% c(201:210)] = 2.125
meta.data$x2[meta.data$pseudo_rank %in% c(211:300)] = 2.1
meta.data$x2[meta.data$pseudo_rank %in% c(301:310)] = 2.075
meta.data$x2[meta.data$pseudo_rank %in% c(311:400)] = 2.05
meta.data$x2[meta.data$pseudo_rank %in% c(401:600)] = 2.025

meta.data$x2[meta.data$pseudo_rank %in% c(651:660)] = 1.3
meta.data$x2[meta.data$pseudo_rank %in% c(661:670)] = 1.2
meta.data$x2[meta.data$pseudo_rank %in% c(671:680)] = 1.1
meta.data$x2[meta.data$pseudo_rank %in% c(681:700)] = 1
meta.data$x2[meta.data$pseudo_rank %in% c(701:725)] = 0.9
meta.data$x2[meta.data$pseudo_rank %in% c(726:750)] = 0.8
meta.data$x2[meta.data$pseudo_rank %in% c(751:775)] = 0.7
meta.data$x2[meta.data$pseudo_rank %in% c(776:800)] = 0.6
meta.data$x2[meta.data$pseudo_rank %in% c(801:825)] = 0.5
meta.data$x2[meta.data$pseudo_rank %in% c(826:850)] = 0.4
meta.data$x2[meta.data$pseudo_rank %in% c(851:875)] = 0.3
meta.data$x2[meta.data$pseudo_rank %in% c(876:950)] = 0.2
meta.data$x2[meta.data$pseudo_rank %in% c(951:960)] = 0.175
meta.data$x2[meta.data$pseudo_rank %in% c(961:nrow(meta.data))] = 0.15

meta.data$x3 = 1
meta.data$x3[meta.data$pseudo_rank %in% c(1:10)] = 1.1
meta.data$x3[meta.data$pseudo_rank %in% c(11:20)] = 1.05
meta.data$x3[meta.data$pseudo_rank %in% c(31:40)] = 0.95
meta.data$x3[meta.data$pseudo_rank %in% c(41:50)] = 0.9
meta.data$x3[meta.data$pseudo_rank %in% c(51:60)] = 0.85
meta.data$x3[meta.data$pseudo_rank %in% c(61:70)] = 0.825
meta.data$x3[meta.data$pseudo_rank %in% c(71:120)] = 0.8
meta.data$x3[meta.data$pseudo_rank %in% c(121:130)] = 0.825
meta.data$x3[meta.data$pseudo_rank %in% c(131:200)] = 0.85
meta.data$x3[meta.data$pseudo_rank %in% c(201:210)] = 0.875
meta.data$x3[meta.data$pseudo_rank %in% c(211:300)] = 0.9
meta.data$x3[meta.data$pseudo_rank %in% c(301:310)] = 0.925
meta.data$x3[meta.data$pseudo_rank %in% c(311:400)] = 0.95
meta.data$x3[meta.data$pseudo_rank %in% c(401:600)] = 0.975

meta.data$x3[meta.data$pseudo_rank %in% c(651:660)] = 1.7
meta.data$x3[meta.data$pseudo_rank %in% c(661:670)] = 1.8
meta.data$x3[meta.data$pseudo_rank %in% c(671:680)] = 1.9
meta.data$x3[meta.data$pseudo_rank %in% c(681:700)] = 2
meta.data$x3[meta.data$pseudo_rank %in% c(701:725)] = 2.1
meta.data$x3[meta.data$pseudo_rank %in% c(726:750)] = 2.2
meta.data$x3[meta.data$pseudo_rank %in% c(751:775)] = 2.3
meta.data$x3[meta.data$pseudo_rank %in% c(776:800)] = 2.4
meta.data$x3[meta.data$pseudo_rank %in% c(801:825)] = 2.5
meta.data$x3[meta.data$pseudo_rank %in% c(826:850)] = 2.6
meta.data$x3[meta.data$pseudo_rank %in% c(851:875)] = 2.7
meta.data$x3[meta.data$pseudo_rank %in% c(876:950)] = 2.8
meta.data$x3[meta.data$pseudo_rank %in% c(951:960)] = 2.825
meta.data$x3[meta.data$pseudo_rank %in% c(961:nrow(meta.data))] = 2.85

meta.data$x4 = 2
meta.data$x4[meta.data$pseudo_rank %in% c(1:10)] = 1.9
meta.data$x4[meta.data$pseudo_rank %in% c(11:20)] = 1.95
meta.data$x4[meta.data$pseudo_rank %in% c(31:40)] = 2.05
meta.data$x4[meta.data$pseudo_rank %in% c(41:50)] = 2.1
meta.data$x4[meta.data$pseudo_rank %in% c(51:60)] = 2.15
meta.data$x4[meta.data$pseudo_rank %in% c(61:70)] = 2.175
meta.data$x4[meta.data$pseudo_rank %in% c(71:120)] = 2.2
meta.data$x4[meta.data$pseudo_rank %in% c(121:130)] = 2.175
meta.data$x4[meta.data$pseudo_rank %in% c(131:200)] = 2.15
meta.data$x4[meta.data$pseudo_rank %in% c(201:210)] = 2.125
meta.data$x4[meta.data$pseudo_rank %in% c(211:300)] = 2.1
meta.data$x4[meta.data$pseudo_rank %in% c(301:310)] = 2.075
meta.data$x4[meta.data$pseudo_rank %in% c(311:400)] = 2.05
meta.data$x4[meta.data$pseudo_rank %in% c(401:600)] = 2.025

meta.data$x4[meta.data$pseudo_rank %in% c(651:660)] = 2
meta.data$x4[meta.data$pseudo_rank %in% c(661:670)] = 2.1
meta.data$x4[meta.data$pseudo_rank %in% c(671:680)] = 2.2
meta.data$x4[meta.data$pseudo_rank %in% c(681:700)] = 2.3
meta.data$x4[meta.data$pseudo_rank %in% c(701:725)] = 2.4
meta.data$x4[meta.data$pseudo_rank %in% c(726:750)] = 2.5
meta.data$x4[meta.data$pseudo_rank %in% c(751:775)] = 2.6
meta.data$x4[meta.data$pseudo_rank %in% c(776:800)] = 2.7
meta.data$x4[meta.data$pseudo_rank %in% c(801:825)] = 2.8
meta.data$x4[meta.data$pseudo_rank %in% c(826:850)] = 2.9
meta.data$x4[meta.data$pseudo_rank %in% c(851:875)] = 3
meta.data$x4[meta.data$pseudo_rank %in% c(876:950)] = 3.1
meta.data$x4[meta.data$pseudo_rank %in% c(951:960)] = 3.075
meta.data$x4[meta.data$pseudo_rank %in% c(961:nrow(meta.data))] = 3.05


meta.data$x5 = 1
meta.data$x5[meta.data$pseudo_rank %in% c(1:10)] = 1.1
meta.data$x5[meta.data$pseudo_rank %in% c(11:20)] = 1.05
meta.data$x5[meta.data$pseudo_rank %in% c(31:40)] = 0.95
meta.data$x5[meta.data$pseudo_rank %in% c(41:50)] = 0.9
meta.data$x5[meta.data$pseudo_rank %in% c(51:60)] = 0.85
meta.data$x5[meta.data$pseudo_rank %in% c(61:70)] = 0.825
meta.data$x5[meta.data$pseudo_rank %in% c(71:120)] = 0.8
meta.data$x5[meta.data$pseudo_rank %in% c(121:130)] = 0.825
meta.data$x5[meta.data$pseudo_rank %in% c(131:200)] = 0.85
meta.data$x5[meta.data$pseudo_rank %in% c(201:210)] = 0.875
meta.data$x5[meta.data$pseudo_rank %in% c(211:300)] = 0.9
meta.data$x5[meta.data$pseudo_rank %in% c(301:310)] = 0.925
meta.data$x5[meta.data$pseudo_rank %in% c(311:400)] = 0.95
meta.data$x5[meta.data$pseudo_rank %in% c(401:600)] = 0.975

meta.data$x5[meta.data$pseudo_rank %in% c(651:660)] = 1.3
meta.data$x5[meta.data$pseudo_rank %in% c(661:670)] =1.25
meta.data$x5[meta.data$pseudo_rank %in% c(671:680)] = 1.2
meta.data$x5[meta.data$pseudo_rank %in% c(681:700)] = 1.15
meta.data$x5[meta.data$pseudo_rank %in% c(701:725)] = 1.1
meta.data$x5[meta.data$pseudo_rank %in% c(726:750)] = 1.05
meta.data$x5[meta.data$pseudo_rank %in% c(751:775)] = 1
meta.data$x5[meta.data$pseudo_rank %in% c(776:800)] = 0.95
meta.data$x5[meta.data$pseudo_rank %in% c(801:825)] = 0.9
meta.data$x5[meta.data$pseudo_rank %in% c(826:850)] = 0.85
meta.data$x5[meta.data$pseudo_rank %in% c(851:875)] = 0.8
meta.data$x5[meta.data$pseudo_rank %in% c(876:950)] = 0.75
meta.data$x5[meta.data$pseudo_rank %in% c(951:960)] = 0.775
meta.data$x5[meta.data$pseudo_rank %in% c(961:nrow(meta.data))] = 0.8

meta.data$x6 = 2
meta.data$x6[meta.data$pseudo_rank %in% c(1:10)] = 1.9
meta.data$x6[meta.data$pseudo_rank %in% c(11:20)] = 1.95
meta.data$x6[meta.data$pseudo_rank %in% c(31:40)] = 2.05
meta.data$x6[meta.data$pseudo_rank %in% c(41:50)] = 2.1
meta.data$x6[meta.data$pseudo_rank %in% c(51:60)] = 2.15
meta.data$x6[meta.data$pseudo_rank %in% c(61:70)] = 2.175
meta.data$x6[meta.data$pseudo_rank %in% c(71:120)] = 2.2
meta.data$x6[meta.data$pseudo_rank %in% c(121:130)] = 2.175
meta.data$x6[meta.data$pseudo_rank %in% c(131:200)] = 2.15
meta.data$x6[meta.data$pseudo_rank %in% c(201:210)] = 2.125
meta.data$x6[meta.data$pseudo_rank %in% c(211:300)] = 2.1
meta.data$x6[meta.data$pseudo_rank %in% c(301:310)] = 2.075
meta.data$x6[meta.data$pseudo_rank %in% c(311:400)] = 2.05
meta.data$x6[meta.data$pseudo_rank %in% c(401:600)] = 2.025

meta.data$x6[meta.data$pseudo_rank %in% c(651:660)] = 1.6
meta.data$x6[meta.data$pseudo_rank %in% c(661:670)] = 1.55
meta.data$x6[meta.data$pseudo_rank %in% c(671:680)] = 1.5
meta.data$x6[meta.data$pseudo_rank %in% c(681:700)] = 1.45
meta.data$x6[meta.data$pseudo_rank %in% c(701:725)] = 1.4
meta.data$x6[meta.data$pseudo_rank %in% c(726:750)] = 1.35
meta.data$x6[meta.data$pseudo_rank %in% c(751:775)] = 1.3
meta.data$x6[meta.data$pseudo_rank %in% c(776:800)] = 1.25
meta.data$x6[meta.data$pseudo_rank %in% c(801:825)] = 1.2
meta.data$x6[meta.data$pseudo_rank %in% c(826:850)] = 1.15
meta.data$x6[meta.data$pseudo_rank %in% c(851:875)] = 1.1
meta.data$x6[meta.data$pseudo_rank %in% c(876:950)] = 1.05
meta.data$x6[meta.data$pseudo_rank %in% c(951:960)] = 1.025
meta.data$x6[meta.data$pseudo_rank %in% c(961:nrow(meta.data))] = 1


meta.data$x7 = 1
meta.data$x7[meta.data$pseudo_rank %in% c(1:10)] = 1.1
meta.data$x7[meta.data$pseudo_rank %in% c(11:20)] = 1.05
meta.data$x7[meta.data$pseudo_rank %in% c(31:40)] = 0.95
meta.data$x7[meta.data$pseudo_rank %in% c(41:50)] = 0.9
meta.data$x7[meta.data$pseudo_rank %in% c(51:60)] = 0.85
meta.data$x7[meta.data$pseudo_rank %in% c(61:70)] = 0.825
meta.data$x7[meta.data$pseudo_rank %in% c(71:120)] = 0.8
meta.data$x7[meta.data$pseudo_rank %in% c(121:130)] = 0.825
meta.data$x7[meta.data$pseudo_rank %in% c(131:200)] = 0.85
meta.data$x7[meta.data$pseudo_rank %in% c(201:210)] = 0.875
meta.data$x7[meta.data$pseudo_rank %in% c(211:300)] = 0.9
meta.data$x7[meta.data$pseudo_rank %in% c(301:310)] = 0.925
meta.data$x7[meta.data$pseudo_rank %in% c(311:400)] = 0.95
meta.data$x7[meta.data$pseudo_rank %in% c(401:600)] = 0.975

meta.data$x7[meta.data$pseudo_rank %in% c(651:660)] = 1.4
meta.data$x7[meta.data$pseudo_rank %in% c(661:670)] = 1.45
meta.data$x7[meta.data$pseudo_rank %in% c(671:680)] = 1.5
meta.data$x7[meta.data$pseudo_rank %in% c(681:700)] = 1.55
meta.data$x7[meta.data$pseudo_rank %in% c(701:725)] = 1.6
meta.data$x7[meta.data$pseudo_rank %in% c(726:750)] = 1.65
meta.data$x7[meta.data$pseudo_rank %in% c(751:775)] = 1.7
meta.data$x7[meta.data$pseudo_rank %in% c(776:800)] = 1.75
meta.data$x7[meta.data$pseudo_rank %in% c(801:825)] = 1.8
meta.data$x7[meta.data$pseudo_rank %in% c(826:850)] = 1.85
meta.data$x7[meta.data$pseudo_rank %in% c(851:875)] = 1.9
meta.data$x7[meta.data$pseudo_rank %in% c(876:950)] = 1.95
meta.data$x7[meta.data$pseudo_rank %in% c(951:960)] = 1.975
meta.data$x7[meta.data$pseudo_rank %in% c(961:nrow(meta.data))] = 2

meta.data$x8 = 2
meta.data$x8[meta.data$pseudo_rank %in% c(1:10)] = 1.9
meta.data$x8[meta.data$pseudo_rank %in% c(11:20)] = 1.95
meta.data$x8[meta.data$pseudo_rank %in% c(31:40)] = 2.05
meta.data$x8[meta.data$pseudo_rank %in% c(41:50)] = 2.1
meta.data$x8[meta.data$pseudo_rank %in% c(51:60)] = 2.15
meta.data$x8[meta.data$pseudo_rank %in% c(61:70)] = 2.175
meta.data$x8[meta.data$pseudo_rank %in% c(71:120)] = 2.2
meta.data$x8[meta.data$pseudo_rank %in% c(121:130)] = 2.175
meta.data$x8[meta.data$pseudo_rank %in% c(131:200)] = 2.15
meta.data$x8[meta.data$pseudo_rank %in% c(201:210)] = 2.125
meta.data$x8[meta.data$pseudo_rank %in% c(211:300)] = 2.1
meta.data$x8[meta.data$pseudo_rank %in% c(301:310)] = 2.075
meta.data$x8[meta.data$pseudo_rank %in% c(311:400)] = 2.05
meta.data$x8[meta.data$pseudo_rank %in% c(401:600)] = 2.025

meta.data$x8[meta.data$pseudo_rank %in% c(651:660)] = 1.7
meta.data$x8[meta.data$pseudo_rank %in% c(661:670)] = 1.75
meta.data$x8[meta.data$pseudo_rank %in% c(671:680)] = 1.8
meta.data$x8[meta.data$pseudo_rank %in% c(681:700)] = 1.85
meta.data$x8[meta.data$pseudo_rank %in% c(701:725)] = 1.9
meta.data$x8[meta.data$pseudo_rank %in% c(726:750)] = 1.95
meta.data$x8[meta.data$pseudo_rank %in% c(751:775)] = 2
meta.data$x8[meta.data$pseudo_rank %in% c(776:800)] = 2.05
meta.data$x8[meta.data$pseudo_rank %in% c(801:825)] = 2.1
meta.data$x8[meta.data$pseudo_rank %in% c(826:850)] = 2.15
meta.data$x8[meta.data$pseudo_rank %in% c(851:875)] = 2.2
meta.data$x8[meta.data$pseudo_rank %in% c(876:950)] = 2.25
meta.data$x8[meta.data$pseudo_rank %in% c(951:960)] = 2.225
meta.data$x8[meta.data$pseudo_rank %in% c(961:nrow(meta.data))] = 2.2




a = meta.data[,c("pseudo_rank","time_point","x1","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")]
#"NVEC200-009271.1","NVEC200-014866.1","NVEC200-013033.1","NVEC200-006430.1","NVEC200-006423.1","NVEC200-010166.1","NVEC200-001639.1","NVEC200-000819.1","NVEC200-014003.1","NVEC200-003858.1","NVEC200-015133.1","NVEC200-002032.1","NVEC200-014221.1","NVEC200-005215.1","NVEC200-011057.1","NVEC200-009256.1","NVEC200-000547.1","NVEC200-005257.1","NVEC200-014418.1","NVEC200-003926.1","NVEC200-000574.1","NVEC200-006121.1","NVEC200-004712.1","NVEC200-001561.1","NVEC200-007938.1","NVEC200-012562.1","NVEC200-006310.1","NVEC200-015153.1","NVEC200-000708.1","NVEC200-000772.1"

colnames(a) = c("pseudo_rank","time_point","x","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")
a$rank = a$pseudo_rank
a$rank[a$pseudo_rank %in% c(601:nrow(a))] = paste("left",a$pseudo_rank,sep = "_")

b = meta.data[,c("pseudo_rank","time_point","x2","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")]
#"NVEC200-009271.1","NVEC200-014866.1","NVEC200-013033.1","NVEC200-006430.1","NVEC200-006423.1","NVEC200-010166.1","NVEC200-001639.1","NVEC200-000819.1","NVEC200-014003.1","NVEC200-003858.1","NVEC200-015133.1","NVEC200-002032.1","NVEC200-014221.1","NVEC200-005215.1","NVEC200-011057.1","NVEC200-009256.1","NVEC200-000547.1","NVEC200-005257.1","NVEC200-014418.1","NVEC200-003926.1","NVEC200-000574.1","NVEC200-006121.1","NVEC200-004712.1","NVEC200-001561.1","NVEC200-007938.1","NVEC200-012562.1","NVEC200-006310.1","NVEC200-015153.1","NVEC200-000708.1","NVEC200-000772.1"

colnames(b) = c("pseudo_rank","time_point","x","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")
b$rank = b$pseudo_rank
b$rank[b$pseudo_rank %in% c(601:nrow(b))] = paste("left",b$pseudo_rank,sep = "_")


c = meta.data[,c("pseudo_rank","time_point","x3","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")]
#"NVEC200-009271.1","NVEC200-014866.1","NVEC200-013033.1","NVEC200-006430.1","NVEC200-006423.1","NVEC200-010166.1","NVEC200-001639.1","NVEC200-000819.1","NVEC200-014003.1","NVEC200-003858.1","NVEC200-015133.1","NVEC200-002032.1","NVEC200-014221.1","NVEC200-005215.1","NVEC200-011057.1","NVEC200-009256.1","NVEC200-000547.1","NVEC200-005257.1","NVEC200-014418.1","NVEC200-003926.1","NVEC200-000574.1","NVEC200-006121.1","NVEC200-004712.1","NVEC200-001561.1","NVEC200-007938.1","NVEC200-012562.1","NVEC200-006310.1","NVEC200-015153.1","NVEC200-000708.1","NVEC200-000772.1"
colnames(c) = c("pseudo_rank","time_point","x","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")
c$rank = c$pseudo_rank
c$rank[c$pseudo_rank %in% c(601:nrow(c))] = paste("right",c$pseudo_rank,sep = "_")

d = meta.data[,c("pseudo_rank","time_point","x4","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")]
#"NVEC200-009271.1","NVEC200-014866.1","NVEC200-013033.1","NVEC200-006430.1","NVEC200-006423.1","NVEC200-010166.1","NVEC200-001639.1","NVEC200-000819.1","NVEC200-014003.1","NVEC200-003858.1","NVEC200-015133.1","NVEC200-002032.1","NVEC200-014221.1","NVEC200-005215.1","NVEC200-011057.1","NVEC200-009256.1","NVEC200-000547.1","NVEC200-005257.1","NVEC200-014418.1","NVEC200-003926.1","NVEC200-000574.1","NVEC200-006121.1","NVEC200-004712.1","NVEC200-001561.1","NVEC200-007938.1","NVEC200-012562.1","NVEC200-006310.1","NVEC200-015153.1","NVEC200-000708.1","NVEC200-000772.1"
colnames(d) = c("pseudo_rank","time_point","x","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")
d$rank = d$pseudo_rank
d$rank[d$pseudo_rank %in% c(601:nrow(d))] = paste("right",d$pseudo_rank,sep = "_")



e = meta.data[,c("pseudo_rank","time_point","x5","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")]
#"NVEC200-009271.1","NVEC200-014866.1","NVEC200-013033.1","NVEC200-006430.1","NVEC200-006423.1","NVEC200-010166.1","NVEC200-001639.1","NVEC200-000819.1","NVEC200-014003.1","NVEC200-003858.1","NVEC200-015133.1","NVEC200-002032.1","NVEC200-014221.1","NVEC200-005215.1","NVEC200-011057.1","NVEC200-009256.1","NVEC200-000547.1","NVEC200-005257.1","NVEC200-014418.1","NVEC200-003926.1","NVEC200-000574.1","NVEC200-006121.1","NVEC200-004712.1","NVEC200-001561.1","NVEC200-007938.1","NVEC200-012562.1","NVEC200-006310.1","NVEC200-015153.1","NVEC200-000708.1","NVEC200-000772.1"

colnames(e) = c("pseudo_rank","time_point","x","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")
e$rank = e$pseudo_rank
e$rank[e$pseudo_rank %in% c(601:nrow(e))] = paste("leftcenter",e$pseudo_rank,sep = "_")

f = meta.data[,c("pseudo_rank","time_point","x6","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")]
#"NVEC200-009271.1","NVEC200-014866.1","NVEC200-013033.1","NVEC200-006430.1","NVEC200-006423.1","NVEC200-010166.1","NVEC200-001639.1","NVEC200-000819.1","NVEC200-014003.1","NVEC200-003858.1","NVEC200-015133.1","NVEC200-002032.1","NVEC200-014221.1","NVEC200-005215.1","NVEC200-011057.1","NVEC200-009256.1","NVEC200-000547.1","NVEC200-005257.1","NVEC200-014418.1","NVEC200-003926.1","NVEC200-000574.1","NVEC200-006121.1","NVEC200-004712.1","NVEC200-001561.1","NVEC200-007938.1","NVEC200-012562.1","NVEC200-006310.1","NVEC200-015153.1","NVEC200-000708.1","NVEC200-000772.1"

colnames(f) = c("pseudo_rank","time_point","x","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")
f$rank = f$pseudo_rank
f$rank[f$pseudo_rank %in% c(601:nrow(f))] = paste("leftcenter",f$pseudo_rank,sep = "_")


g = meta.data[,c("pseudo_rank","time_point","x7","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")]
#"NVEC200-009271.1","NVEC200-014866.1","NVEC200-013033.1","NVEC200-006430.1","NVEC200-006423.1","NVEC200-010166.1","NVEC200-001639.1","NVEC200-000819.1","NVEC200-014003.1","NVEC200-003858.1","NVEC200-015133.1","NVEC200-002032.1","NVEC200-014221.1","NVEC200-005215.1","NVEC200-011057.1","NVEC200-009256.1","NVEC200-000547.1","NVEC200-005257.1","NVEC200-014418.1","NVEC200-003926.1","NVEC200-000574.1","NVEC200-006121.1","NVEC200-004712.1","NVEC200-001561.1","NVEC200-007938.1","NVEC200-012562.1","NVEC200-006310.1","NVEC200-015153.1","NVEC200-000708.1","NVEC200-000772.1"
colnames(g) = c("pseudo_rank","time_point","x","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")
g$rank = g$pseudo_rank
g$rank[g$pseudo_rank %in% c(601:nrow(g))] = paste("rightcenter",g$pseudo_rank,sep = "_")

h = meta.data[,c("pseudo_rank","time_point","x8","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")]
#"NVEC200-009271.1","NVEC200-014866.1","NVEC200-013033.1","NVEC200-006430.1","NVEC200-006423.1","NVEC200-010166.1","NVEC200-001639.1","NVEC200-000819.1","NVEC200-014003.1","NVEC200-003858.1","NVEC200-015133.1","NVEC200-002032.1","NVEC200-014221.1","NVEC200-005215.1","NVEC200-011057.1","NVEC200-009256.1","NVEC200-000547.1","NVEC200-005257.1","NVEC200-014418.1","NVEC200-003926.1","NVEC200-000574.1","NVEC200-006121.1","NVEC200-004712.1","NVEC200-001561.1","NVEC200-007938.1","NVEC200-012562.1","NVEC200-006310.1","NVEC200-015153.1","NVEC200-000708.1","NVEC200-000772.1"
colnames(h) = c("pseudo_rank","time_point","x","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")
h$rank = h$pseudo_rank
h$rank[h$pseudo_rank %in% c(601:nrow(h))] = paste("rightcenter",h$pseudo_rank,sep = "_")

df = rbind(a,b,c,d,e,f,g,h)


pdf("./tomo_Harmony_Embedding_PT_Jitter_ClusterMarker_Shape.pdf",width=4,height=8)
ggplot(meta.data, aes(y = pseudo_rank,x = jitter, color = time_point)) +theme_bw() + geom_point(aes(color = meta.data[,"NVEC200-009271.1"]))  + ylab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + scale_colour_gradientn(colors = c("gray",rev(viridis_pal(option = "viridis")(12)))) 

ggplot(df, aes(x = x, y = pseudo_rank)) + 
  geom_line(aes(group = rank,color = df[,"gene"]),size=0.25) + scale_colour_gradientn(colors = c("gray",rev(viridis_pal(option = "viridis")(12)))) 

dev.off()


plot_list=list()

plot_list <- lapply(colnames(meta.data[,3:(ncol(meta.data)-1)]), function(i) { 
  p <- ggplot(meta.data, aes(y = pseudo_rank,x = jitter)) +theme_bw() + geom_point(aes(color = meta.data[,i]),cex = 1)  + ylab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.position = "none", axis.line = element_line(colour = "black"))  + scale_colour_gradientn(colors = c("gray",rev(viridis_pal(option = "viridis")(12)))) 
})

library(gridExtra)

pdf("./tomo_Harmony_Embedding_PT_Jitter_ClusterMarker_Shape.pdf",width=10,height=20)

ggplot(meta.data, aes(y = pseudo_rank,x = jitter, color = time_point)) +theme_bw() + geom_point(cex = 5)  + ylab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) 

do.call("grid.arrange",plot_list)
 
dev.off()



plot_list=list()

plot_list <- lapply(colnames(df[,4:(ncol(df)-1)]), function(i) { 
  p <- ggplot(df, aes(x = x, y = pseudo_rank)) + 
  geom_line(aes(group = rank,color = df[,i]),size=0.25) + scale_colour_gradientn(colors = c(rep("darkblue",10),rev(viridis_pal(option = "plasma")(12)))) + ylab("PT") +ggtitle(i) + theme_bw() + theme(axis.line=element_blank(),   axis.text.x=element_blank(), axis.text.y=element_blank(), axis.ticks=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), legend.position="none", panel.background=element_blank(), panel.border=element_blank(), panel.grid.major=element_blank(),
      panel.grid.minor=element_blank(), plot.background=element_blank())
  })

library(gridExtra)

pdf("./tomo_Harmony_Embedding_PT_Lines_ClusterMarker_Shape.pdf",width=7,height=15)
do.call("grid.arrange",plot_list)
 
dev.off()



######make Nematostella shape with illustrator
#only plot a heatmap with lines for expression

tomo.harm = readRDS("./tomo_Harmony.RDS")

set.seed(seed = 300)
meta.data = FetchData(tomo.harm,c("pseudo_rank"))

tmp = as.data.frame(GetAssayData(tomo.harm,"data"),stringsAsFactors = F)
tmp = as.data.frame(t(tmp))
tmp = tmp[,c("NVEC200-001043.1",
"NVEC200-000874.1",
"NVEC200-000872.1",
"NVEC200-010351.1",
"NVEC200-010468.1",
"NVEC200-006988.1",
"NVEC200-001297.1",
"NVEC200-007533.1",
"NVEC200-014111.1",
"NVEC200-006032.1",
"NVEC200-006030.1",
"NVEC200-006642.1",
"NVEC200-006028.1",
"NVEC200-003168.1",
"NVEC200-012164.1",
"NVEC200-009995.1",
"NVEC200-011699.1",
"NVEC200-000491.1",
"NVEC200-002987.1",
"NVEC200-000883.1",
"NVEC200-006580.1",
"NVEC200-005938.1",
"NVEC200-001459.1",
"NVEC200-001458.1",
"NVEC200-009287.1",
"NVEC200-008102.1",
"NVEC200-015113.1",
"NVEC200-015112.1",
"NVEC200-012338.1",
"NVEC200-001364.1",
"NVEC200-001363.1",
"NVEC200-001362.1",
"NVEC200-013954.1",
"NVEC200-011655.1",
"NVEC200-008902.1",
"NVEC200-009545.1",
"NVEC200-003200.1",
"NVEC200-009182.1",
"NVEC200-006274.1",
"NVEC200-001846.1",
"NVEC200-013190.1",
"NVEC200-004425.1",
"NVEC200-006033.1",
"NVEC200-013831.1",
"NVEC200-014326.1",
"NVEC200-008616.1",
"NVEC200-008615.1",
"NVEC200-001014.1",
"NVEC200-000260.1",
"NVEC200-009768.1",
"NVEC200-008308.1",
"NVEC200-000261.1",
"NVEC200-010238.1",
"NVEC200-008832.1",
"NVEC200-005807.1",
"NVEC200-005805.1",
"NVEC200-005804.1",
"NVEC200-001872.1",
"NVEC200-011709.1",
"NVEC200-008022.1",
"NVEC200-012414.1")]

meta.data = cbind(meta.data,tmp,stringsAsFactors = F)

plot.df = melt(meta.data,id.vars = c("pseudo_rank"))

pdf("./tomo_Harmony_PT_InSituMarker_Shape.pdf",width=10,height=15)
ggplot(plot.df,aes(x = pseudo_rank, y = 1, color = value , fill = value )) + geom_col()  + scale_colour_gradientn(colors = c(rep("darkblue",1),(viridis_pal(option = "plasma")(12))))  + scale_fill_gradientn(colors = c(rep("darkblue",1),(viridis_pal(option = "plasma")(12))))   + facet_wrap(variable~. ) + theme_bw() + theme(axis.line=element_blank(),   axis.text.x=element_blank(), axis.text.y=element_blank(), axis.ticks=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), legend.position="none", panel.background=element_blank(), panel.border=element_blank(), panel.grid.major=element_blank(),
      panel.grid.minor=element_blank(), plot.background=element_blank())

dev.off()


pdf("./tomo_Harmony_PT_InSituMarker_Shape_individual.pdf",width=10,height=15)

for (i in c("NVEC200-010351.1",
"NVEC200-010468.1",
"NVEC200-001364.1",
"NVEC200-001363.1",
"NVEC200-001362.1",
"NVEC200-011709.1",
"NVEC200-008308.1",
"NVEC200-000872.1",
"NVEC200-015113.1",
"NVEC200-012414.1")) {
  
  print(ggplot(meta.data,aes(x = pseudo_rank, y = 1, color = meta.data[,i] , fill = meta.data[,i]  )) + geom_col()  + scale_colour_gradientn(colors = c(rep("darkblue",1),(viridis_pal(option = "plasma")(12))))  + scale_fill_gradientn(colors = c(rep("darkblue",1),(viridis_pal(option = "plasma")(12))))  + theme_bw() + ggtitle(i) + theme(axis.line=element_blank(),   axis.text.x=element_blank(), axis.text.y=element_blank(), axis.ticks=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), legend.position="none", panel.background=element_blank(), panel.border=element_blank(), panel.grid.major=element_blank(),
      panel.grid.minor=element_blank(), plot.background=element_blank()))

  
}
dev.off()



```

##Check for reshaping

```{r}


tomo.harm = readRDS("./tomo_Harmony.RDS")


#across time points

#check contribution of timepoints to each cluster
for (i in 0 : max(as.integer(tomo.harm@meta.data$seurat_clusters)-1)) {
  if ( i == 0) {
    a = as.data.frame(table(tomo.harm@meta.data$time_point[tomo.harm@meta.data$seurat_cluster == i]))
    a$cluster = i
  } else {
    b = as.data.frame(table(tomo.harm@meta.data$time_point[tomo.harm@meta.data$seurat_clusters == i]))
    b$cluster = i
    a = rbind(a,b)
  }
}

a$Freq.lib = rep(as.numeric(table(tomo.harm@meta.data$time_point)),5)
a$ratio = a$Freq / a$Freq.lib
a

#hack to get the wrapping done for pie charts...

cp <- coord_polar(theta = "y")
cp$is_free <- function() TRUE


pdf("tomo_Harmony_TimepointClusterContributionHistogram.pdf",width=15,height=15)
ggplot(a,aes(x = Var1,y = ratio,fill = Var1)) + geom_col() + facet_wrap(~cluster, scale = "free") + theme_bw() + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1), panel.border = element_blank(), panel.grid.major = element_blank(),
panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + scale_fill_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))
ggplot(a,aes(x = "",y = ratio,fill = Var1)) + geom_bar(width = 1, stat = "identity") + theme_bw()  + cp + facet_wrap(~cluster, scales = "free")  +  theme(aspect.ratio = 1) + scale_fill_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))
dev.off()


#individual poylps


tomo.harm@meta.data$sample_id <- factor(tomo.harm@meta.data$sample_id, levels = c("HH7T7BGXF-39","HH7T7BGXF-40","HH7T7BGXF-42","HLLWFBGXH-9","HLMT2BGXH-37","HLMTJBGXH-1","HLLWFBGXH-17","HLMKKBGXH-2","HLMMWBGXH-4","HLMTJBGXH-40","HLLWFBGXH-18","HLMKKBGXH-3","HLMMWBGXH-42","HLMKKBGXH-4","HLMMWBGXH-6","HLMT2BGXH-48","HLLWFBGXH-13","HLMMWBGXH-7","HLMT2BGXH-43","HLMTJBGXH-16"))



#check contribution of timepoints to each cluster
for (i in 0 : max(as.integer(tomo.harm@meta.data$seurat_clusters)-1)) {
  if ( i == 0) {
    a = as.data.frame(table(tomo.harm@meta.data[tomo.harm@meta.data$seurat_cluster == i, c("sample_id")]))
    a$cluster = i
    a$time_point = c("uncut","uncut","uncut","uncut","uncut","uncut","12hpa","12hpa","12hpa","12hpa","24hpa","24hpa","24hpa","48hpa","48hpa","48hpa","96hpa","96hpa","96hpa","96hpa")
  } else {
    b = as.data.frame(table(tomo.harm@meta.data[tomo.harm@meta.data$seurat_clusters == i, c("sample_id")]))
    b$cluster = i
    b$time_point = c("uncut","uncut","uncut","uncut","uncut","uncut","12hpa","12hpa","12hpa","12hpa","24hpa","24hpa","24hpa","48hpa","48hpa","48hpa","96hpa","96hpa","96hpa","96hpa")
    a = rbind(a,b)
  }
}

a$Freq.lib = rep(as.numeric(table(tomo.harm@meta.data$sample_id)),5)
a$ratio = a$Freq / a$Freq.lib
a

a$time_point <- factor(a$time_point, levels = c("uncut","12hpa","24hpa","48hpa","96hpa"))

a$cluster <- factor(a$cluster, levels = c(2,0,1,3,4))

#hack to get the wrapping done for pie charts...

cp <- coord_polar(theta = "y")
cp$is_free <- function() TRUE


pdf("tomo_Harmony_PolypClusterContributionHistogram.pdf",width=15,height=3)
ggplot(a,aes(x = Var1,y = ratio,fill = time_point)) + geom_col()  + facet_wrap(~cluster, scale = "free", ncol = 5) + theme_bw() + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1), panel.border = element_blank(), panel.grid.major = element_blank(),
panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + scale_fill_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))
ggplot(a,aes(x = "",y = ratio,fill = time_point)) + geom_bar(width = 1, stat = "identity") + theme_bw()  + cp + facet_wrap(~cluster, scales = "free", ncol = 5)  +  theme(aspect.ratio = 1) + scale_fill_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))
dev.off()



#check for mixing along pseudotime


#50 bins along PT


tomo.harm@meta.data$time_point <- factor(tomo.harm@meta.data$time_point, levels = c("uncut","12hpa","24hpa","48hpa","96hpa"))

meta.order = FetchData(tomo.harm,c("time_point","pseudo_rank"))

count = as.integer(nrow(meta.order) / 50)
bin.size = as.integer(nrow(meta.order) / 50)
for (i in 1:50) {
  if ( i == 1) {
    a = as.data.frame(table(meta.order$time_point[meta.order$pseudo_rank %in% 1:bin.size]),stringsAsFactors = F)
    a$bin = i
  } else {
    b = as.data.frame(table(meta.order$time_point[meta.order$pseudo_rank %in% c((count+1):(count+bin.size))]),stringsAsFactors = F)
    b$bin = i
    count = count + bin.size
    a = rbind(a,b)
  }
}

a$Freq.lib = rep(as.numeric(table(tomo.harm@meta.data$time_point)),50)
a$Freq = as.numeric(a$Freq)
a$ratio = a$Freq / a$Freq.lib


a$Var1 <- factor(a$Var1, levels = c("uncut","12hpa","24hpa","48hpa","96hpa"))


#plot as histogram

pdf("./tomo_Harmony_TimepointContributionPT_Histogram.pdf",width=5,height=5)
ggplot(a, aes(x = bin,y = Freq, fill = Var1, color = Var1)) +theme_bw() + geom_col() + xlab("PT") + ylab("Timepoint frequency") + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + scale_fill_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))

ggplot(a, aes(x = bin,y = ratio, fill = Var1, color = Var1)) +theme_bw() + geom_col(stat="identity", position="dodge") + xlab("PT") + ylab("Normalized timepoint ratio")  + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + scale_fill_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))

ggplot(a, aes(x = bin,y = ratio, fill = Var1, color = Var1)) +theme_bw() + geom_point() + geom_smooth(se = F) + xlab("PT") + ylab("Normalized timepoint ratio")  + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + scale_fill_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4])) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))
dev.off()




```






#Uncut only


```{r}

select.cells = colnames(tomo.harm)[tomo.harm@meta.data$sample_name %in% tomo.harm@meta.data$sample_name[grep("uncut",tomo.harm@meta.data$sample_name)]]

uncut = subset(tomo.harm, cells = select.cells)


##############

#Cluster cells
uncut<- FindNeighbors(uncut, dims = 2:20,reduction = "harmony")
uncut <- FindClusters(uncut, resolution = 0.8)

# Perform Hierarchical Clustering
library(tidyverse)
library(factoextra)
Data.hc <- uncut@reductions$pca@cell.embeddings[,c(2:20)] %>% get_dist(method = "pearson") %>% hclust(method = 'ward.D') ## PCs
Hcls <- data.frame(Clusters = as.factor(cutree(Data.hc, k = 5)))
Hcls <- Hcls %>% mutate(Wells = colnames(uncut))

uncut@meta.data$Hclust = Hcls$Clusters

uncut = RunUMAP(uncut, dims = 2:20,min.dist = 0.25,reduction = "harmony")

#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("uncut_Embedding_PC2_20_res0.8.pdf",width=6,height=5)
DimPlot(object = uncut, shuffle = T, reduction = 'umap', pt.size = 3,cols = c("darkgoldenrod3","turquoise4","red3","dodgerblue","salmon"))
DimPlot(object = uncut, shuffle = T, reduction = 'umap', pt.size = 3, group.by = "sequencing_sample_id")
DimPlot(object = uncut, shuffle = T, reduction = 'umap', pt.size = 3, group.by = "batch")
DimPlot(object = uncut, shuffle = T,reduction = 'umap', pt.size = 3, group.by = "sample_name",cols = c("dodgerblue","turquoise3","darkblue","red3","salmon","violetred2"))
DimPlot(object = uncut, shuffle = T,reduction = 'umap', pt.size = 3, group.by = "sample_name",cols = c("gray40","gray25","gray10","gray90","gray75","gray60"))
DimPlot(object = uncut, shuffle = T,label = F, reduction = 'umap',group.by = "slice_index", pt.size = 3)
DimPlot(object = uncut, shuffle = T,label = F, reduction = 'umap',group.by = "slice_sum", pt.size = 3,cols = viridis_pal(option = "magma")(10))
FeaturePlot(uncut, reduction = 'umap', pt.size = 1, features = c("nFeature_RNA","nCount_RNA"),order = F, cols = c(brewer.pal(9,"Greys")[3],brewer.pal(9,"Blues")[3:9]))
dev.off()

saveRDS(uncut,"./uncut_Harmony.RDS")

```

###Markers

```{r}

gene_annotation = read.csv("/g/arendt/gerber/Analyses/NematostellaTomoSeq/tomo_assays_rna_metafeatures.csv")

emapper_annotation = read.delim(file = "/g/arendt/gerber/Analyses/NematostellaTomoSeq/MM_9krs15um.emapper.annotations.tsv", sep = "\t",header = T)

emapper_annotation$query = gsub("_1.",".",emapper_annotation$query)
emapper_annotation$query = gsub("_","-",emapper_annotation$query)
colnames(emapper_annotation) = gsub("query","gene_id",colnames(emapper_annotation))

gene_annotation = merge(gene_annotation,emapper_annotation, by = "gene_id",all.x = T)

v1_Nvec_LUT = read.delim("/g/arendt/gerber/Genomes/Nematostella/Nvecv1/NvecV1_Nvec200_MatchTable.tsv", sep = "\t", header = T)
v1_annotation = read.delim("/g/arendt/gerber/Analyses/NematostellaScRNAseq/MarkerGenes_SebePedros.tsv", sep = "\t", header = T)
v1_annotation = merge(v1_Nvec_LUT,v1_annotation,by.x = "NvecV1", by.y = "gene",all.x = T)

#v1_annotation = merge(v1_annotation,test6,by = "NvecV1",all.x = T)

blast_annotation = read.csv("/g/arendt/gerber/Analyses/NematostellaScRNAseq/v3query_vs_ncbi100.out.full.csv",header = T)
blast_annotation = blast_annotation[,c("gene_id","v3.Note","v3.Ontology_term","gene_name_short")]
blast_annotation = blast_annotation[!duplicated(blast_annotation),]


markers <- FindAllMarkers(uncut, only.pos = TRUE,  logfc.threshold = 0.3)
markers$diff = markers$pct.1 - markers$pct.2

markers.anno = markers
markers.anno = merge(markers.anno,gene_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
markers.anno = merge(markers.anno,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
markers.anno = markers.anno  %>% arrange(cluster , desc(diff))





write_delim(markers.anno,delim = "\t","./uncut_Harmony_SeuratClusterMarker.tsv")




#test markers of all slice data set tomo.harm


#plot with annotation
gene_ids =c(
"NVEC200-009271.1",
"NVEC200-006430.1",
"NVEC200-014823.1",
"NVEC200-005937.1",
"NVEC200-006423.1",
"NVEC200-003174.1",
"NVEC200-007376.1",
"NVEC200-007166.1",
"NVEC200-002459.1",
"NVEC200-002734.1",
"NVEC200-004198.1",
"NVEC200-005968.1",
"NVEC200-011460.1",
"NVEC200-013726.1",
"NVEC200-010443.1",
"NVEC200-002310.1",
"NVEC200-014047.1",
"NVEC200-003642.1",
"NVEC200-011925.1",
"NVEC200-000547.1",
"NVEC200-001405.1",
"NVEC200-005216.1",
"NVEC200-002397.1",
"NVEC200-011742.1",
"NVEC200-008616.1",
"NVEC200-002307.1",
"NVEC200-003388.1",
"NVEC200-003926.1",
"NVEC200-000574.1",
"NVEC200-005104.1",
"NVEC200-012414.1",
"NVEC200-001119.1",
"NVEC200-012745.1",
"NVEC200-005455.1",
"NVEC200-007938.1",
"NVEC200-006310.1",
"NVEC200-003993.1",
"NVEC200-007657.1",
"NVEC200-001253.1",
"NVEC200-006391.1",
"NVEC200-000758.1",
"NVEC200-004893.1",
"NVEC200-004526.1",
"NVEC200-000574.1",
"NVEC200-013268.1",
"NVEC200-004653.1",
"NVEC200-012175.1",
"NVEC200-006239.1")


gene_names =c(
"Btg1",
"hmx3",
"PSAP",
"Hoxc4",
"nkx-2.5",
"SELENOP",
"CalpB",
"Trafd1",
"faxc",
"blt801",
"Capsl",
"Gpx2",
"Ctrb1",
"AQP4",
"fabL",
"OVCH1",
"Mmp19",
"Cpa2",
"DSCAM",
"aguA",
"OIT3",
"PRY2",
"ADAM9",
"Npr1",
"Wnt5a",
"CUBlike",
"50 kDa hatching enzyme",
"fezf2",
"TUBA1D",
"PKD1L2",
"RGMB",
"CD63",
"pkd2",
"Csrp2",
"Gsx1",
"Fezf2",
"LYPLAL1",
"Gpx5",
"C1qb",
"NKX6-2",
"COL17A1",
"Adgrd1",
"Tekt1",
"TUBA1D",
"kcnk9",
"TEKT2",
"CHRNA9",
"KCND2")

gg_Fig <- FeaturePlot(uncut, pt.size = 2, features = gene_ids,order = T, cols = c("gray",rev(viridis_pal(option = "mako")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) & NoLegend() & NoAxes()})

pdf("uncut_Harmony_Embedding_AllSliceSeuratClusterMarkerAnno.pdf",width=25,height=20)
CombinePlots( gg_Fig )
dev.off()

#plot uncut marker with annotation
gene_ids =c("NVEC200-006423.1",
"NVEC200-006430.1",
"NVEC200-008673.1",
"NVEC200-009271.1",
"NVEC200-000423.1",
"NVEC200-007658.1",
"NVEC200-001639.1",
"NVEC200-002032.1",
"NVEC200-000547.1",
"NVEC200-011651.1",
"NVEC200-013969.1",
"NVEC200-007045.1",
"NVEC200-003926.1",
"NVEC200-008603.1",
"NVEC200-009005.1",
"NVEC200-000162.1",
"NVEC200-007938.1",
"NVEC200-006310.1",
"NVEC200-007657.1",
"NVEC200-009518.1")

gene_names =c("nkx-2.5",
"hmx3",
"Tll1",
"Btg1",
"MEP1B",
"Gm2a",
"Lytic polysaccharide mono-oxygenase",
"Trypsin",
"aguA",
"Fat4",
"COL6A3",
"EPHA4",
"fezf2",
"dmbx1b",
"AOC1",
"TCF15",
"Gsx1",
"Fezf2",
"Gpx5",
"LAMA2")

gg_Fig <- FeaturePlot(uncut, pt.size = 2, features = gene_ids,order = T, cols = (viridis_pal(option = "plasma")(12)),min.cutoff = "q5",max.cutoff = "q95", repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) & NoLegend() & NoAxes()})

pdf("uncut_Harmony_Embedding_SeuratClusterMarkerAnno.pdf",width=12,height=10)
CombinePlots( gg_Fig )
dev.off()

#plot for paper
gene_ids =c("NVEC200-006423.1",
"NVEC200-007658.1",
"NVEC200-011651.1",
"NVEC200-000162.1",
"NVEC200-007657.1",
"NVEC200-008673.1",
"NVEC200-000423.1",
"NVEC200-013969.1",
"NVEC200-009005.1",
"NVEC200-007938.1")
gene_names =c("nkx-2.5",
"Gm2a",
"Fat4",
"TCF15",
"Gpx5",
"Tll1",
"MEP1B",
"COL6A3",
"AOC1",
"Gsx1")

gg_Fig <- FeaturePlot(uncut, pt.size = 2, features = gene_ids,order = T,ncol = 6, cols = (viridis_pal(option = "plasma")(12)),min.cutoff = "q5",max.cutoff = "q95", repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) & NoAxes()})

pdf("uncut_Harmony_Embedding_PaperMarkerAnno.pdf",width=10,height=4)
CombinePlots( gg_Fig )
FeaturePlot(uncut, pt.size = 1.5, features = gene_ids,order = T,ncol = 5, cols = (viridis_pal(option = "plasma")(12)),min.cutoff = "q5",max.cutoff = "q95") & NoLegend() & NoAxes()
dev.off()



#plot cluster marker as heatmap along pseudotime


genes = unique(c(markers.anno$gene[markers.anno$cluster == 2][1:30],markers.anno$gene[markers.anno$cluster == 0][1:30],markers.anno$gene[markers.anno$cluster == 1][1:30],markers.anno$gene[markers.anno$cluster == 4][1:30],markers.anno$gene[markers.anno$cluster == 3][1:30]))


data = as.matrix(GetAssayData(object = uncut, slot = "data"))
data = data[genes,]

#hr <- hclust(as.dist(1-cor(t(data),method="pearson")), method="ward")

#data = data[match(names(sort(cutree(hr,k=4))),rownames(data)),]

#if (!require(devtools)) {
#  install.packages("devtools")
#}

#devtools::install_github("xmc811/Scillus", ref = "development")
library(Scillus)


pdf("uncut_Harmony_Heatmap_ClusterMarker.pdf",width=15,height=15)
plot_heatmap(dataset = uncut, 
              markers = genes,
              sort_var = c("pseudo_rank"),
              anno_var = c("time_point","sample_name","pseudo_rank","slice_sum","ident"),
              anno_colors = list("Set1","Grays", "RdYlBu" ,"RdYlBu",c("yellow","red")), hm_colors = viridis_pal(option = "plasma")(3), hm_limit = c(-0.5,0,1))
dev.off()

```

###Pseudotime

```{r}
######Calc pseudo time


harmony.embedding = Embeddings(uncut, "harmony")

#make scaled values positive
for (i in 1:ncol(harmony.embedding)) {
  harmony.embedding[,i] = harmony.embedding[,i] + abs(min(harmony.embedding[,i]))
}

harmony.embedding = as.data.frame(harmony.embedding,stringsAsFactor = F)

meta.data = FetchData(object = uncut, vars = c('ident', 'orig.ident','UMAP_1','UMAP_2',"sequencing_sample_id","batch","sample_name","time_point","slice_index","slice_sum"))


#destiny

library("destiny")
library("Rcpp")


cells_bottom = rownames(meta.data)[meta.data$ident %in% c(2)]
cells_center1 = rownames(meta.data)[meta.data$ident %in% c(0,1)]
#cells_center2 = rownames(meta.data)[meta.data$ident %in% c(1)]
cells_top = rownames(meta.data)[meta.data$ident %in% c(3,4)]


#cells_bottom = rownames(meta.data)[meta.data$UMAP_1 > 2]
#cells_center1 = rownames(meta.data)[meta.data$UMAP_1 < 2 & meta.data$UMAP_1 > (-2)]
#cells_center2 = rownames(meta.data)[meta.data$UMAP_1 < (-2) & meta.data$UMAP_2 < 0]
#cells_top = rownames(meta.data)[meta.data$UMAP_1 < (-2) & meta.data$UMAP_2 > 0 ]


#calculate pseudo time separate on branches

harmony.embedding.bottom = as.data.frame(harmony.embedding[cells_bottom,],stringsAsFactors = F)
harmony.embedding.center1 = as.data.frame(harmony.embedding[cells_center1,],stringsAsFactors = F)
#harmony.embedding.center2 = as.data.frame(harmony.embedding[cells_center2,],stringsAsFactors = F)
harmony.embedding.top = as.data.frame(harmony.embedding[cells_top,],stringsAsFactors = F)

library("destiny")

#bottom
DatasetDM <- DiffusionMap(harmony.embedding.bottom[,2:20])

dpt = DPT(DatasetDM )

pseudotime_bottom = dpt[random_root(dpt), ]

harmony.embedding.bottom$pt = as.numeric(pseudotime_bottom)
harmony.embedding.bottom = harmony.embedding.bottom[order(harmony.embedding.bottom$pt),]

harmony.embedding.bottom$pt_adj = rev(harmony.embedding.bottom$pt)
harmony.embedding.bottom$pt_adj = harmony.embedding.bottom$pt_adj

harmony.embedding.bottom$rank = rev(1:nrow(harmony.embedding.bottom))


#center1
DatasetDM <- DiffusionMap(harmony.embedding.center1[,2:20])

dpt = DPT(DatasetDM )

pseudotime_center1 = dpt[random_root(dpt), ]

harmony.embedding.center1$pt = as.numeric(pseudotime_center1)
harmony.embedding.center1 = harmony.embedding.center1[order(harmony.embedding.center1$pt),]

harmony.embedding.center1$pt_adj = (harmony.embedding.center1$pt)
harmony.embedding.center1$pt_adj = harmony.embedding.center1$pt_adj+  max(harmony.embedding.bottom$pt_adj)


harmony.embedding.center1$rank = (1:nrow(harmony.embedding.center1))
harmony.embedding.center1$rank = harmony.embedding.center1$rank  + nrow(harmony.embedding.bottom)

#center2
#DatasetDM <- DiffusionMap(harmony.embedding.center2[,2:20])

#dpt = DPT(DatasetDM )

#pseudotime_center2 = dpt[random_root(dpt), ]

#harmony.embedding.center2$pt = as.numeric(pseudotime_center2)
#harmony.embedding.center2 = harmony.embedding.center2[order(harmony.embedding.center2$pt),]

#harmony.embedding.center2$pt_adj = rev(harmony.embedding.center2$pt)
#harmony.embedding.center2$pt_adj = harmony.embedding.center2$pt_adj+  max(harmony.embedding.bottom$pt_adj)


#harmony.embedding.center2$rank = (1:nrow(harmony.embedding.center2))
#harmony.embedding.center2$rank = harmony.embedding.center2$rank  + nrow(harmony.embedding.center1) + nrow(harmony.embedding.bottom)

#top
DatasetDM <- DiffusionMap(harmony.embedding.top[,2:20])

dpt = DPT(DatasetDM)

pseudotime_top = dpt[random_root(dpt)]

harmony.embedding.top$pt = as.numeric(pseudotime_top)
harmony.embedding.top = harmony.embedding.top[order(harmony.embedding.top$pt),]

harmony.embedding.top$pt_adj = rev(harmony.embedding.top$pt)
harmony.embedding.top$pt_adj = harmony.embedding.top$pt_adj+ max(harmony.embedding.center1$pt_adj)  +  max(harmony.embedding.bottom$pt_adj)

harmony.embedding.top$rank = rev(1:nrow(harmony.embedding.top))
harmony.embedding.top$rank = harmony.embedding.top$rank  + nrow(harmony.embedding.center1)  + nrow(harmony.embedding.bottom)


pseudotime = rbind(harmony.embedding.bottom,harmony.embedding.center1,harmony.embedding.top,stringsAsFactors = F)

plot.pt = cbind(meta.data,rank = pseudotime[match(rownames(meta.data),rownames(pseudotime)),"rank"],pt_adj = pseudotime[match(rownames(meta.data),rownames(pseudotime)),"pt_adj"], stringsAsFactors = F)


uncut@meta.data$pseudo_rank = plot.pt$rank
uncut@meta.data$pseudo_raw = plot.pt$pt_adj

### set some parameters
# cells
cellcex = 1
#color
colramp = colorRampPalette(viridis(12))
colgrad = colramp(101)

pdf("./uncut_Harmony_Embedding_UMAP_PT_Rank.pdf",width=8,height=8)

#ranks
  # normalize expression [0-1]
  exp = plot.pt [,"rank"]/max(plot.pt [,"rank"])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  plot.pt [,"bin"] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(plot.pt$UMAP_1, plot.pt$UMAP_2, cex=cellcex, pch=16, col=colgrad[plot.pt[,"bin"]], xlab="", ylab="", main = "pseudotime rank")
    
    
#raw pt
     # normalize expression [0-1]
  exp = plot.pt [,"pt_adj"]/max(plot.pt [,"pt_adj"])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  plot.pt [,"bin"] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(plot.pt$UMAP_1, plot.pt$UMAP_2, cex=cellcex, pch=16, col=colgrad[plot.pt[,"bin"]], xlab="", ylab="", main = "pseudotime raw")
    
    DimPlot(object = uncut, shuffle = T, reduction = 'umap', pt.size = 3,cols = c("darkgoldenrod3","turquoise4","red3","dodgerblue","salmon"))

dev.off()


#plot cluster marker as heatmap along pseudotime


markers <- FindAllMarkers(uncut, only.pos = TRUE,  logfc.threshold = 0.3)
markers$diff = markers$pct.1 - markers$pct.2
markers = markers  %>% arrange(cluster , desc(diff))


genes = unique(c(markers$gene[markers$cluster == 2][1:35],
                 markers$gene[markers$cluster == 0][1:35],
                 markers$gene[markers$cluster == 1][1:35],
                 #no informative markers
                 markers$gene[markers$cluster == 4][1:35],
                 markers$gene[markers$cluster == 3][1:35]))


data = as.matrix(GetAssayData(object = uncut, slot = "data"))
data = data[genes,]

#hr <- hclust(as.dist(1-cor(t(data),method="pearson")), method="ward")

#data = data[match(names(sort(cutree(hr,k=4))),rownames(data)),]

#if (!require(devtools)) {
#  install.packages("devtools")
#}

#devtools::install_github("xmc811/Scillus", ref = "development")
library(Scillus)


pdf("uncut_Harmony_Heatmap_ClusterMarker_raw.pdf",width=15,height=30)
plot_heatmap(dataset = uncut, 
              markers = genes,
              sort_var = c("pseudo_rank"),
              anno_var = c("time_point","sample_name","pseudo_rank","slice_sum","seurat_clusters"),
              anno_colors = list("Set1","Greys", "RdYlBu" ,"YlGn","Set3"), hm_colors = viridis_pal(option = "plasma")(3), hm_limit = c(-0.5,0,1))
ggplot(FetchData(uncut,c("slice_sum","pseudo_rank")),aes(x = pseudo_rank, color = slice_sum, fill = slice_sum )) + geom_histogram(bins = nrow(FetchData(uncut,c("slice_sum","pseudo_rank")))) + scale_fill_manual(values = rev(colorRampPalette(brewer.pal(9,"YlGn"))(9)))
dev.off()

####only one reasonable markers found for cluster 1 among top35 with louvain
#look manually


matrix_mod<-as.matrix(GetAssayData(object = uncut, slot = "data"))
gene<-as.numeric(matrix_mod["NVEC200-000547.1",])
correlations<-sort(apply(matrix_mod,1,function(x){cor(gene,x)}))
correlations = as.data.frame(correlations)
correlations$gene = rownames(correlations)
correlations = merge(correlations,gene_annotation, by.x="gene" , by.y="gene_id")
correlations = merge(correlations,blast_annotation, by.x="gene" , by.y="gene_id",all.x = T)
correlations = merge(correlations,v1_annotation, by.x="gene" , by.y="Nvec200",all.x = T)
correlations = correlations[order(-correlations$correlations),]  

write_delim(correlations,delim = "\t","./uncut_Harmony_000547_corGenes.tsv")
  
genes = unique(c(markers$gene[markers$cluster == 2][1:35],
                 markers$gene[markers$cluster == 0][1:35],
                 unique(correlations[,"gene"])[1:35],
                 #no informative markers for cluster 1 --> cor genes
                 markers$gene[markers$cluster == 4][1:35],
                 markers$gene[markers$cluster == 3][1:35]))

saveRDS(genes,"./uncut_Harmony_ClusterMarker_final.RDS")

pdf("uncut_Harmony_Heatmap_ClusterMarker_final.pdf",width=15,height=15)
plot_heatmap(dataset = uncut, 
              markers = genes,
              sort_var = c("pseudo_rank"),
              anno_var = c("time_point","sample_name","pseudo_rank","slice_sum","seurat_clusters"),
              anno_colors = list("Set1","Greys", "RdYlBu" ,"YlGn","Set3"), hm_colors = viridis_pal(option = "plasma")(3), hm_limit = c(-0.5,0,1))
ggplot(FetchData(uncut,c("slice_sum","pseudo_rank")),aes(x = pseudo_rank, color = slice_sum, fill = slice_sum )) + geom_histogram(bins = nrow(FetchData(uncut,c("slice_sum","pseudo_rank")))) + scale_fill_manual(values = rev(colorRampPalette(brewer.pal(9,"YlGn"))(9)))
dev.off()


```


##Check alignment of pseudo and slice index

```{r}

tmp = list()
for (i in unique(uncut@meta.data$sample_id)) {
  tmp[[i]] = FetchData(subset(uncut,subset = sample_id == i),c("sample_id","time_point","pseudo_rank","slice_index","nCount_RNA"))
  tmp[[i]] = tmp[[i]][order(tmp[[i]][,"pseudo_rank"]),]
  tmp[[i]]$pseudo_rank_slice =  1:length(tmp[[i]]$pseudo_rank)
  tmp[[i]] = tmp[[i]][order(tmp[[i]][,"slice_index"]),]
  tmp[[i]]$slice_index_slice =  1:length(tmp[[i]]$pseudo_rank)
  tmp[[i]]$grouping = factor(1:length(tmp[[i]]$pseudo_rank))
}

tmp = do.call("rbind", tmp)

tmp$diff = abs(tmp$slice_index_slice - tmp$pseudo_rank_slice)

plot.df = melt(tmp[,c("pseudo_rank_slice","slice_index_slice","grouping","sample_id","time_point")],id.vars = c("grouping","sample_id","time_point"))

pdf("./uncut_Harmony_PT_index_comparison.pdf",width=15,height=15)
ggplot(plot.df,aes(x= value,y = variable, color = time_point, group = grouping)) + geom_point() + geom_line() + facet_wrap(~sample_id)
ggplot(tmp,aes(x= slice_index_slice,y = diff)) + geom_col() + facet_wrap(~sample_id)
ggplot(tmp,aes(x= slice_index_slice,y = nCount_RNA)) + geom_col() + facet_wrap(~sample_id)
ggplot(tmp,aes(x= diff,y = nCount_RNA)) + geom_point() + facet_wrap(~sample_id, scales = "free_y") + geom_smooth(method='lm', formula= y~x)
ggplot(tmp,aes(x= diff,y = nCount_RNA, color = sample_id)) + geom_point()  + geom_smooth(method='lm', formula= y~x, se = F)
ggplot(tmp,aes(x= pseudo_rank_slice,y = slice_index_slice,color = time_point)) + geom_point() + facet_wrap(~sample_id, scales = "free") + geom_smooth(method='lm', formula= y~x)
dev.off()


#Get R and function for regression line and R value

model = list()

for (i in 1:length(unique(tmp$sample_id))) {
  tmp.sub = tmp[tmp$sample_id %in% unique(tmp$sample_id)[i],]
  model[[unique(tmp$sample_id)[i]]] <- summary(lm(pseudo_rank_slice ~ slice_index_slice, data = tmp.sub))

}

saveRDS(model,"./uncut_Harmony_PT_index_comparison_statistics.RDS")
  



ggp = ggplot(tmp,aes(x= diff,y = nCount_RNA)) + geom_point() + stat_smooth()
ggp_data <- ggplot_build(ggp)$data[[2]] 

#get mean summary per sample
aggregate(tmp$diff, list(tmp$sample_id), FUN=median)


#Plot the marker genes in Fig1 heatmap on all polyps with original slice ordering

genes = readRDS("./uncut_Harmony_ClusterMarker_final.RDS")

#devtools::install_github("xmc811/Scillus", ref = "development")
library(Scillus)


pdf("uncut_Harmony_Heatmap_UncutClusterMarker_SliceIndex_byPolyp.pdf",width=30,height=15)
plot_heatmap(dataset = uncut, 
              markers = genes,
              sort_var = c("sample_id","slice_index"),
              anno_var = c("slice_index","sample_name","pseudo_rank","slice_sum","seurat_clusters"),
              anno_colors = list("YlGn","Greys", "RdYlBu" ,"YlGn","Set3"), hm_colors = viridis_pal(option = "plasma")(3), hm_limit = c(-0.5,0,1))
dev.off()





```

##Compare to time warping

```{r}
uncut = readRDS("./uncut_Harmony.RDS")
tomo.harm = readRDS("./tomo_Harmony.RDS")
time_warp = readRDS("/g/arendt/gerber/Analyses/NematostellaTomoSeq/tomoseq_positions_seurat_v3.RDS")

#subset time warp Data 

time_warp = subset(time_warp, subset = sample_id %in% tomo.harm@meta.data$sample_id)

time_warp = subset(time_warp, subset = sample_name %in% tomo.harm@meta.data$sample_name)

DE500 = read.csv("gene_cluster_k12_long.csv")

time_warp@meta.data$sample_id = as.factor(time_warp@meta.data$sample_id)
time_warp@meta.data$sample_id <- factor(time_warp@meta.data$sample_id, levels = c("HH7T7BGXF-39","HH7T7BGXF-40","HH7T7BGXF-42","HLLWFBGXH-9","HLMT2BGXH-37","HLMTJBGXH-1","HLLWFBGXH-17","HLMKKBGXH-2","HLMMWBGXH-4","HLMTJBGXH-40","HLLWFBGXH-18","HLMKKBGXH-3","HLMMWBGXH-42","HLMKKBGXH-4","HLMMWBGXH-6","HLMT2BGXH-48","HLLWFBGXH-13","HLMMWBGXH-7","HLMT2BGXH-43","HLMTJBGXH-16"))

#all tomo slices

#devtools::install_github("xmc811/Scillus", ref = "development")
library(Scillus)


pdf("tomo_TimeWarp_Heatmap_DanilaHVG500Marker_byPolyp.pdf",width=30,height=15)
plot_heatmap(dataset = time_warp, 
              markers = DE500$gene_id,
              sort_var = c("sample_id","pos"),
              anno_var = c("pos","sample_id","time_point"),
              anno_colors = list("YlGn","Greys", "Set1" ,"YlGn","Set3"), hm_colors = viridis_pal(option = "plasma")(3), hm_limit = c(-3,0,4))
dev.off()


#all uncut

time_warp_uncut = subset(time_warp,subset = sample_id %in% c("HH7T7BGXF-39","HH7T7BGXF-40","HH7T7BGXF-42","HLLWFBGXH-9","HLMT2BGXH-37","HLMTJBGXH-1"))
#devtools::install_github("xmc811/Scillus", ref = "development")
library(Scillus)


pdf("uncut_TimeWarp_Heatmap_DanilaHVG500Marker_byPolyp.pdf",width=30,height=15)
plot_heatmap(dataset = time_warp_uncut, 
              markers = DE500$gene_id,
              sort_var = c("sample_id","pos"),
              anno_var = c("sample_id","time_point"),
              anno_colors = list("Greys", "Set1" ,"YlGn","Set3"), hm_colors = viridis_pal(option = "plasma")(3), hm_limit = c(-4,0,4),
              row_font_size = 4)
dev.off()

#plot expression across time and polyps for the HCR marker genes

meta.data = FetchData(time_warp_uncut,c("pos","sample_name"))

tmp = as.data.frame(GetAssayData(time_warp_uncut,"data"),stringsAsFactors = F)
tmp = as.data.frame(t(tmp))
tmp = tmp[,c("NVEC200-003903.1",
"NVEC200-006089.1",
"NVEC200-003854.1",
"NVEC200-008375.1",
"NVEC200-003672.1",
"NVEC200-006641.1",
"NVEC200-015111.1",
"NVEC200-005216.1",
"NVEC200-011036.1",
"NVEC200-014929.1", 
"NVEC200-006163.1",
"NVEC200-014405.1",
"NVEC200-009518.1",
"NVEC200-014970.1",
"NVEC200-012414.1",
"NVEC200-009923.1",
"NVEC200-011585.1")]

meta.data = cbind(meta.data,tmp,stringsAsFactors = F)

plot.df = melt(meta.data,id.vars = c("pos","sample_name"))

pdf("./uncut_TimeWarp_HCRmarker_Histogram.pdf",width=15,height=7)
ggplot(plot.df,aes(x= pos,y = value,color = sample_name)) + geom_point()  + geom_smooth(se = F,span = 0.5) + facet_wrap(~variable,scales = "free") + scale_color_manual(values = colorRampPalette(brewer.pal(9,"Greys")[2:9])(6))
ggplot(plot.df,aes(x= pos,y = value)) + geom_col() + facet_wrap(~variable,scales = "free") + geom_smooth(se = F,span = 0.5)
dev.off()


#compare time warped data with pseudo index

time_warp_uncut_no0 = subset(time_warp_uncut, subset = nCount_RNA == 0, invert = T)

tmp1 = list()
tmp2 = list()
for (i in unique(uncut@meta.data$sample_id)) {
  
  tmp1[[i]] = FetchData(subset(uncut,subset = sample_id == i),c("sample_id","time_point","pseudo_rank","slice_index","nCount_RNA"))
  tmp2[[i]] = FetchData(subset(time_warp_uncut_no0,subset = sample_id == i),c("sample_id","time_point","pos","nCount_RNA"))
  
  #rank the pseudo index per polyp
  tmp1[[i]] = tmp1[[i]][order(tmp1[[i]][,"pseudo_rank"]),]
  tmp1[[i]]$pseudo_rank_slice =  1:length(tmp1[[i]]$pseudo_rank)
}

tmp1 = do.call("rbind", tmp1)
tmp2 = do.call("rbind", tmp2)

merge(tmp1[,c()])

plot.df = melt(tmp[,c("pseudo_rank_slice","slice_index_slice","grouping","sample_id","time_point")],id.vars = c("grouping","sample_id","time_point"))

pdf("./uncut_Harmony_PT_index_comparison.pdf",width=15,height=15)
ggplot(plot.df,aes(x= value,y = variable, color = time_point, group = grouping)) + geom_point() + geom_line() + facet_wrap(~sample_id)
ggplot(tmp,aes(x= slice_index_slice,y = diff)) + geom_col() + facet_wrap(~sample_id)
ggplot(tmp,aes(x= slice_index_slice,y = nCount_RNA)) + geom_col() + facet_wrap(~sample_id)
ggplot(tmp,aes(x= diff,y = nCount_RNA)) + geom_point() + facet_wrap(~sample_id, scales = "free_y") + geom_smooth(method='lm', formula= y~x)
ggplot(tmp,aes(x= diff,y = nCount_RNA, color = sample_id)) + geom_point()  + geom_smooth(method='lm', formula= y~x, se = F)
ggplot(tmp,aes(x= pseudo_rank_slice,y = slice_index_slice,color = time_point)) + geom_point() + facet_wrap(~sample_id, scales = "free") + geom_smooth(method='lm', formula= y~x)
dev.off()







#only reference polyp

time_warp_42 = subset(time_warp,subset = sample_id %in% "HH7T7BGXF-42")


data = as.matrix(GetAssayData(time_warp_42,slot = "data"))[DE500$gene_id,]

hr <- hclust(as.dist(1-cor(t(data),method="spearman")), method="ward")

data = data[match(rownames(data)[hr$order],rownames(data)),]

pdf("Polyp42_TimeWarp_Heatmap_DanilaHVG500Marker_byPolyp.pdf",width=30,height=15)
plot_heatmap(dataset = time_warp_42, 
              markers = rownames(data),
              sort_var = c("sample_id","pos"),
              anno_var = c("sample_id","time_point"),
              anno_colors = list("Greys", "Set1" ,"YlGn","Set3"), hm_colors = viridis_pal(option = "plasma")(3), hm_limit = c(-1,0,2),
              row_font_size = 4)

plot(hr, labels = NULL, hang = 0.1, 
     main = "Cluster dendrogram")

dev.off()





```


###Search for proliferation marker

```{r}
#check first how known cell cycle markers are expressed across the embedding

#NVEC200_000589_1.1	top2a
#NVEC200_008052_1.1	hmgb2
#NVEC200_006181_1.1	mki67
#NVEC200_010739_1.1	aurka
#NVEC200_010955_1.1	cenpa
#NVEC200-008980_1.1 cdca8

cc =c("NVEC200-008052.1","NVEC200-006181.1","NVEC200-010739.1","NVEC200-010955.1","NVEC200-008980.1")


pdf("uncut_Harmony_Feature_AllCdk.pdf",width=15,height=10)
FeaturePlot(uncut, pt.size = 2, features = gene_annotation[grep("cdk",gene_annotation$gene_name),"gene_id"],order = T, cols = c(viridis_pal(option = "plasma")(50)), repel=T, min.cutoff = "q1") & NoLegend() & NoAxes()
dev.off()
  

#screen manually nematostella annotation list of g2m genes from seurat

#intersect with var genes

uncut = FindVariableFeatures(
  uncut,
  nfeatures = 7000)

#top10 <- head(VariableFeatures(diatom), 10)

gene_ids = intersect(VariableFeatures(uncut),c("NVEC200-0000589.1",
            "NVEC200-008052.1",
            "NVEC200-006181.1",
            "NVEC200-010739.1",
            "NVEC200-010955.1",
            "NVEC200-008980.1",
            "NVEC200-014852.1",
            "NVEC200-005759.1",
            "NVEC200-000144.1",
            "NVEC200-010279.1",
            "NVEC200-014354.1",
            "NVEC200-011097.1"))

gene_ids =c("NVEC200-0000589.1",
            "NVEC200-008052.1",
            "NVEC200-006181.1",
            "NVEC200-010739.1",
            "NVEC200-010955.1",
            "NVEC200-008980.1",
            "NVEC200-014852.1",
            "NVEC200-005759.1",
            "NVEC200-000144.1",
            "NVEC200-010279.1",
            "NVEC200-014354.1",
            "NVEC200-011097.1")
gene_names =c(
"top2a",
"hmgb2",
"mki67",
"aurka",
"cenpa",
"cdca8",
"cdk1",
"tacc3",
"nusap1",
"anln",
"ndc80",
"ttk")

gg_Fig <- FeaturePlot(tomo.harm, pt.size = 2, features = gene_ids,order = T, cols = c(viridis_pal(option = "plasma")(50)), repel=T, min.cutoff = "q5")
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) & NoLegend() & NoAxes()})

pdf("tomo_Harmony_Feature_ManualCC.pdf",width=15,height=10)
CombinePlots( gg_Fig )
dev.off()


pdf("tomo_Harmony_Feature_AllCdk.pdf",width=15,height=10)
FeaturePlot(tomo.harm, pt.size = 2, features = gene_annotation[grep("cdk",gene_annotation$gene_name),"gene_id"],order = T, cols = c(viridis_pal(option = "plasma")(50)), repel=T, min.cutoff = "q1") & NoLegend() & NoAxes()
dev.off()


#Make a score

tomo.harm = AddModuleScore(tomo.harm,list(cc.score = gene_ids),name= c("cc.score"))
  

gene_ids =c(#"NVEC200-0000589.1",
            "NVEC200-008052.1",
            "NVEC200-006181.1",
            "NVEC200-010739.1",
            "NVEC200-010955.1",
            "NVEC200-008980.1",
            "NVEC200-014852.1",
            "NVEC200-005759.1",
            "NVEC200-000144.1",
            "NVEC200-010279.1",
            "NVEC200-014354.1",
            "NVEC200-011097.1",
            "cc.score1")
gene_names =c(
#"top2a",
"hmgb2",
"mki67",
"aurka",
"cenpa",
"cdca8",
"cdk1",
"tacc3",
"nusap1",
"anln",
"ndc80",
"ttk",
"cc.score")

gg_Fig <- FeaturePlot(tomo.harm, pt.size = 2, features = gene_ids,order = T, cols = c(viridis_pal(option = "plasma")(50)), repel=T, min.cutoff = "q5")
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) & NoLegend() & NoAxes()})

pdf("tomo_Harmony_Feature_ManualCC.pdf",width=15,height=10)
CombinePlots( gg_Fig )
dev.off()



gene_ids =c(#"NVEC200-0000589.1",
            "NVEC200-008052.1",
            "NVEC200-006181.1",
            "NVEC200-010739.1",
            "NVEC200-010955.1",
            "NVEC200-008980.1",
            "NVEC200-014852.1",
            "NVEC200-005759.1",
            "NVEC200-000144.1",
            "NVEC200-010279.1",
            "NVEC200-014354.1",
            "NVEC200-011097.1")
a =c(
#"top2a",
"hmgb2",
"mki67",
"aurka",
"cenpa",
"cdca8",
"cdk1",
"tacc3",
"nusap1",
"anln",
"ndc80",
"ttk")

data = as.matrix(GetAssayData(object = tomo.harm, slot = "scale.data"))
data = data[gene_ids,]
#data = rbind(data,t(FetchData(tomo.harm,"cc.score1")))

hr <- hclust(as.dist(1-cor(t(data),method="spearman")), method="ward")

data = data[match(rownames(data)[hr$order],rownames(data)),]


#downsample and extract names to have an even heatmap
set.seed(24)
tmp.uncut <- subset(tomo.harm, subset = time_point == "uncut")
tmp.uncut = colnames(tmp.uncut[, sample(colnames(tmp.uncut), size =119, replace=F)])
tmp.12hpa <- subset(tomo.harm, subset = time_point == "12hpa")
tmp.12hpa <- colnames(tmp.12hpa[, sample(colnames(tmp.12hpa), size =119, replace=F)])
tmp.24hpa <- subset(tomo.harm, subset = time_point == "24hpa")
tmp.24hpa <- colnames(tmp.24hpa[, sample(colnames(tmp.24hpa), size =119, replace=F)])
tmp.48hpa <- subset(tomo.harm, subset = time_point == "48hpa")
tmp.48hpa <- colnames(tmp.48hpa[, sample(colnames(tmp.48hpa), size =119, replace=F)])
tmp.96hpa <- subset(tomo.harm, subset = time_point == "96hpa")
tmp.96hpa <- colnames(tmp.96hpa[, sample(colnames(tmp.96hpa), size =119, replace=F)])

subsample.slices = c(tmp.uncut,tmp.12hpa,tmp.24hpa,tmp.48hpa,tmp.96hpa)
  
  
#tmp.tomo.harm@meta.data$time_point <- factor(x = tmp.tomo.harm@meta.data$time_point, levels = c("uncut", "12hpa", "24hpa", "48hpa", "96hpa")) # change the order of the factor levels

pdf("tomo_Harmony_ManualCC_Heatmap.pdf",width=15,height=15)
plot_heatmap(dataset = subset(tomo.harm, cells = subsample.slices), 
              markers = rownames(data),
              sort_var = c("time_point","pseudo_rank"),
              anno_var = c("time_point","pseudo_rank","slice_sum","cut_info"),
              anno_colors = list("Set1", "YlGnBu","YlGn",c("yellow","red")), hm_colors = c((viridis_pal(option = "plasma")(3))), hm_limit = c(-0.5,0,1))

plot(hr, labels = NULL, hang = 0.1, 
     main = "Cluster dendrogram")
heatmap.2(as.matrix(data),trace="none",density="none",Colv=F, dendrogram="row",Rowv=as.dendrogram(hr), key=T,scale="none")

dev.off()

data = as.matrix(GetAssayData(object = tomo.harm, slot = "scale.data"))
data = data[gene_ids,]
data = rbind(data,t(FetchData(tomo.harm,"cc.score1")))

gene.labs = c(a,"cc.score")
names(gene.labs) = rownames(data)
#gene.labs = gene.labs[match(rownames(data),names(gene.labs))]

plot.df = melt(FetchData(tomo.harm,c("time_point","order_time_scale",gene_ids,"cc.score1")),id.vars=c("time_point","order_time_scale"))

pdf("./tomo_Harmony_ManualCC_LinePlot.pdf",width=10,height=8)
ggplot(plot.df, aes(x = order_time_scale,y = value, fill = time_point, color = time_point)) +theme_bw() + geom_point(alpha = 0.2) + geom_smooth(se = F,span = 0.4)  + xlab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + facet_wrap(~ variable, scale = "free",labeller = labeller(variable = gene.labs)) + scale_color_manual(values = c("gray20",rev(brewer.pal(n = 5, name = "Reds"))[1:4]))  
dev.off()



```


##SPRING


```{r}

harmony.embedding = Embeddings(uncut, "harmony")

#make scaled values positive
for (i in 1:ncol(harmony.embedding)) {
  harmony.embedding[,i] = harmony.embedding[,i] + abs(min(harmony.embedding[,i]))
}

harmony.embedding = as.data.frame(harmony.embedding,stringsAsFactor = F)


#extract norm data from Seurat object

norm.data = GetAssayData(object = uncut, slot = "data")
norm.data = as.matrix(norm.data)


meta.data = FetchData(object = uncut, vars = c('ident', 'orig.ident','UMAP_1','UMAP_2',"sequencing_sample_id","batch","sample_name","time_point","slice_index","slice_sum"))

##############################
#SPRING
##############################

### export for SPRING (export norma and not scaled data)

#cutoff 0.8
y=as.data.frame(t(meta.data[,c('ident','orig.ident','slice_sum',"sample_name")]),stringsAsFactors=F)

write.table(y,sep = ",",col.names = F,file = "SPRING_Export_uncut_meta.csv")

x = as.data.frame(t(harmony.embedding[,c(2:20)]))

write.table(x,sep = ",",col.names = F,file ="SPRING_Export_uncut_HarmonyMatrix.csv")

z=rownames(x)

write.table(z,col.names = F,row.names = F,sep = ",",file = "SPRING_Export_uncut_GeneList.csv")

###Run with : 5 PCs gene / 5 neighbors 

#import SPRING tables

coords = read.csv("/g/arendt/gerber/Analyses/NematostellaTomoSeq/uncut_coordinates.txt",header = F)
colnames(coords) = c("Cell","x","y")
edges = read.csv("/g/arendt/gerber/Analyses/NematostellaTomoSeq/uncut_edge_list.txt",header = F)
colnames(edges) = c("start","end")

rownames(coords) = rownames(harmony.embedding)

### set some parameters
# cells
colstart = "darkblue"
colmed1 = "green"
colmed2 = "yellow"
colmed3 = "orange"
colend = "firebrick"
cellcex = 2
# edges
linecol = adjustcolor("black", alpha.f=0.4)
#linecol = adjustcolor("gray")
linewidth = 0.75

#flip
coords$x = coords$x * -1
coords$y = coords$y * -1

coords$rank=uncut@meta.data$pseudo_rank
coords$slice=uncut@meta.data$slice_sum
coords$slice =  as.numeric(gsub("slice","",coords$slice))
coords$ident=uncut@meta.data$seurat_clusters

coords$color[coords$ident=="0"]= "darkgoldenrod3"
coords$color[coords$ident=="1"]="turquoise4"
coords$color[coords$ident=="2"]="red3"
coords$color[coords$ident=="3"]= "dodgerblue"
coords$color[coords$ident=="4"]="salmon"


plot.data = coords
plot.data$order = sample(1:nrow(plot.data),replace = F)
plot.data = plot.data[order(plot.data$order),]
#plot.data$Cell = rownames(plot.data)

pdf("./uncut_Harmony_SPRING_ident.pdf",width=8,height=4)
plot(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=plot.data$color , xlab="", ylab="")

edge_start_coords = plot.data[match(edges$start, plot.data$Cell), c("x","y")]
edge_end_coords = plot.data[match(edges$end, plot.data$Cell), c("x","y")]
segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
	x1=edge_end_coords[,"x"], y1=edge_end_coords[,"y"], lwd=linewidth, col=linecol)

points(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=plot.data$color)

dev.off()




# cells
cellcex = 2
#color

#colramp = colorRampPalette(c("dodgerblue","violetred2","darkgoldenrod3"))(101)
colramp = colorRampPalette(rev(brewer.pal(n = 9, name = "YlGnBu"))[1:8])(101)
colgrad = colramp

#Get plot for colorpalette
pdf("./PT.scale.pdf",width=8,height=4)
FeaturePlot(uncut, reduction = 'umap', pt.size = 1, features = c("nFeature_RNA","nCount_RNA"),order = F, cols = colorRampPalette(c("dodgerblue","violetred2","darkgoldenrod3"))(101))
FeaturePlot(uncut, reduction = 'umap', pt.size = 1, features = c("nFeature_RNA","nCount_RNA"),order = F, cols = colorRampPalette(rev(brewer.pal(n = 9, name = "YlGnBu"))[1:8])(101))
dev.off()

pdf("./uncut_Harmony_SPRING_PT.pdf",width=8,height=4)

  # normalize expression [0-1]
  exp = plot.data[,"rank"]/max(plot.data[,"rank"])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  plot.data[,"bin"] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=colgrad[plot.data[,"bin"]], xlab="", ylab="", main = "pseudotime")

    
edge_start_coords = plot.data[match(edges$start, plot.data$Cell), c("x","y")]
edge_end_coords = plot.data[match(edges$end, plot.data$Cell), c("x","y")]
segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
	x1=edge_end_coords[,"x"], y1=edge_end_coords[,"y"], lwd=linewidth, col=linecol)

    points(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=colgrad[plot.data[,"bin"]], xlab="", ylab="")

dev.off()

cellcex = 2
#color

#colramp = colorRampPalette(c("dodgerblue","violetred2","darkgoldenrod3"))(101)
colramp =  colorRampPalette(rev(brewer.pal(n = 9, name = "YlGn"))[1:8])(101)
colgrad = colramp

#Get plot for colorpalette
pdf("./Bin.scale.pdf",width=8,height=4)
FeaturePlot(uncut, reduction = 'umap', pt.size = 1, features = c("nFeature_RNA","nCount_RNA"),order = F, cols = colorRampPalette(c("dodgerblue","violetred2","darkgoldenrod3"))(101))
FeaturePlot(uncut, reduction = 'umap', pt.size = 1, features = c("nFeature_RNA","nCount_RNA"),order = F, cols = colorRampPalette(rev(brewer.pal(n = 9, name = "YlGn"))[1:8])(101))
dev.off()

pdf("./uncut_Harmony_SPRING_SliceBin.pdf",width=8,height=4)

  # normalize expression [0-1]
  exp = plot.data[,"slice"]/max(plot.data[,"slice"])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  plot.data[,"bin"] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=colgrad[plot.data[,"bin"]], xlab="", ylab="", main = "pseudotime")

    
edge_start_coords = plot.data[match(edges$start, plot.data$Cell), c("x","y")]
edge_end_coords = plot.data[match(edges$end, plot.data$Cell), c("x","y")]
segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
	x1=edge_end_coords[,"x"], y1=edge_end_coords[,"y"], lwd=linewidth, col=linecol)

    points(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=colgrad[plot.data[,"bin"]], xlab="", ylab="")

dev.off()

#plot genes on Embedding



#plot for paper
tmp=c("NVEC200-010351.1",
"NVEC200-000872.1",
"NVEC200-000491.1",
"NVEC200-001364.1",
"NVEC200-001363.1", 
"NVEC200-001362.1", 
"NVEC200-013831.1",
"NVEC200-001043.1", 
"NVEC200-006580.1",
"NVEC200-015113.1")


meta.data = FetchData(object = uncut, vars = c('seurat_clusters', 'orig.ident',tmp))

tmp.name=c("cdh1",
"Anthox7",
"GBX",
"OtxA",
"OtxB",
"OtxC",
"TCF",
"Anthox1a",
"M-lim",
"NvnAChRaC")
names(tmp.name) = tmp

#color.palette = c(colorRampPalette(brewer.pal(9,"Greys")[9:3])(n = 30),colorRampPalette(brewer.pal(9,"Reds")[3:9])(n = 69))

# create 200 colors (colstart --> colend)
colramp1 = colorRampPalette(brewer.pal(9,"Greys")[9:2])
colramp2 = colorRampPalette(brewer.pal(9,"Reds")[2:9])

colgrad = c(colramp1(50),colramp2(51))

colgrad = viridis_pal(option = "plasma")(101)

cellcex = 2.5
# edges
linecol = adjustcolor("black")
linewidth = 0.5

##########

pdf("./uncut_SPRING_PaperMarker.pdf",width=15,height=10)

par(mfrow=c(4,4))

for (i in colnames(meta.data[,tmp])) {

  # normalize expression [0-1]
  exp = meta.data[,i]/max(meta.data[,i])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  coords[,i] = round(exp, digits=2) *100 +1
  
  coords = coords[order(coords[,i]),]
  #plot gradient

    plot(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="", main = tmp.name[i],bty="n",yaxt="n",xaxt="n")

    edge_start_coords = coords[match(edges$start, coords$Cell), c("x","y")]
    edge_end_coords = coords[match(edges$end, coords$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="")

    coords = coords[match(colnames(uncut),rownames(coords)),]
}

dev.off()


#Plot All in situ markers

tmp=c("NVEC200-001043.1",
"NVEC200-000874.1",
"NVEC200-000872.1",
"NVEC200-010351.1",
"NVEC200-010468.1",
"NVEC200-006988.1",
"NVEC200-001297.1",
"NVEC200-007533.1",
"NVEC200-014111.1",
"NVEC200-006032.1",
"NVEC200-006030.1",
"NVEC200-006642.1",
"NVEC200-006028.1",
"NVEC200-003168.1",
"NVEC200-012164.1",
"NVEC200-009995.1",
"NVEC200-011699.1",
"NVEC200-000491.1",
"NVEC200-002987.1",
"NVEC200-000883.1",
"NVEC200-006580.1",
"NVEC200-005938.1",
"NVEC200-001459.1",
"NVEC200-001458.1",
"NVEC200-009287.1",
"NVEC200-008102.1",
"NVEC200-015113.1",
"NVEC200-015112.1",
"NVEC200-012338.1",
"NVEC200-001364.1",
"NVEC200-001363.1",
"NVEC200-001362.1",
"NVEC200-013954.1",
"NVEC200-011655.1",
"NVEC200-008902.1",
"NVEC200-009545.1",
"NVEC200-003200.1",
"NVEC200-009182.1",
"NVEC200-006274.1",
"NVEC200-001846.1",
"NVEC200-013190.1",
"NVEC200-004425.1",
"NVEC200-006033.1",
"NVEC200-013831.1",
"NVEC200-014326.1",
"NVEC200-008616.1",
"NVEC200-008615.1",
"NVEC200-001014.1",
"NVEC200-000260.1",
"NVEC200-009768.1",
"NVEC200-008308.1",
"NVEC200-000261.1",
"NVEC200-010238.1",
"NVEC200-008832.1",
"NVEC200-005807.1",
"NVEC200-005805.1",
"NVEC200-005804.1",
"NVEC200-001872.1",
"NVEC200-011709.1",
"NVEC200-008022.1",
"NVEC200-012414.1")


meta.data = FetchData(object = uncut, vars = c('seurat_clusters', 'orig.ident',tmp))

meta.data = FetchData(object = uncut, vars = c('seurat_clusters', 'orig.ident'))
tmp.mtx = GetAssayData(uncut,slot = "scale.data")
meta.data = cbind(meta.data,as.data.frame(t(tmp.mtx[tmp,])))

meta.data = FetchData(object = uncut, vars = c('seurat_clusters', 'orig.ident'))
tmp.mtx = GetAssayData(uncut,slot = "data")
meta.data = cbind(meta.data,as.data.frame(t(as.matrix(tmp.mtx[tmp,]))))

tmp.name=c("Anthox1a",
"Anthox6",
"Anthox7",
"cdh1",
"cdh3",
"Churchill",
"Ci",
"delta",
"FGF1A",
"FGF8A",
"FGF8B",
"FGFRa",
"FGFRb",
"Fox1",
"FoxB",
"FoxC ",
"FoxD",
"GBX",
"Hedgehog2",
"HLXB9",
"M-lim",
"MMP 3 inhibitor",
"MoxB_a",
"MoxB_b",
"NCam1",
"NvnAChRaB",
"NvnAChRaC",
"NvnAChRaD",
"NvnAChRaU",
"OtxA",
"OtxB",
"OtxC",
"PaxA",
"PaxD3",
"PL10",
"Repo",
"snailA",
"Sox1",
"Sox2",
"Sox3",
"SoxE",
"SoxF1",
"Sprouty",
"TCF",
"Thiamine enzyme",
"wnt 5",
"wnt 7",
"wnt1",
"wnt10",
"wnt11",
"wnt2",
"wnt6",
"wnt8b",
"wntA",
"ZicC",
"ZicD",
"ZicE",
"Uromodulin_a",
"otp",
"wnt16",
"RGM")
names(tmp.name) = tmp

#color.palette = c(colorRampPalette(brewer.pal(9,"Greys")[9:3])(n = 30),colorRampPalette(brewer.pal(9,"Reds")[3:9])(n = 69))

# create 200 colors (colstart --> colend)
colramp1 = colorRampPalette(brewer.pal(9,"Greys")[9:2])
colramp2 = colorRampPalette(brewer.pal(9,"Reds")[2:9])

colgrad = c(colramp1(50),colramp2(51))

colgrad = viridis_pal(option = "plasma")(101)

cellcex = 2.5
# edges
linecol = adjustcolor("black")
linewidth = 0.5

##########

pdf("./uncut_SPRING_AllInsituMarker_scaled.pdf",width=35,height=22)

par(mfrow=c(8,8))

for (i in colnames(meta.data[,tmp])) {

  # normalize expression [0-1]
  meta.data[,i] = meta.data[,i] + abs(min(meta.data[,i]))
  exp = meta.data[,i]/max(meta.data[,i])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  coords[,i] = round(exp, digits=2) *100 +1
  
  coords = coords[order(coords[,i]),]
  #plot gradient

    plot(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="", main = tmp.name[i],bty="n",yaxt="n",xaxt="n")

    edge_start_coords = coords[match(edges$start, coords$Cell), c("x","y")]
    edge_end_coords = coords[match(edges$end, coords$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="")

    coords = coords[match(colnames(uncut),rownames(coords)),]
}

dev.off()


pdf("./uncut_SPRING_AllInsituMarker_norm.pdf",width=25,height=22)

par(mfrow=c(8,8))

for (i in colnames(meta.data[,tmp])) {

  # normalize expression [0-1]
  exp = meta.data[,i]/max(meta.data[,i])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  coords[,i] = round(exp, digits=2) *100 +1
  
  coords = coords[order(coords[,i]),]
  #plot gradient

    plot(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="", main = tmp.name[i],bty="n",yaxt="n",xaxt="n")

    edge_start_coords = coords[match(edges$start, coords$Cell), c("x","y")]
    edge_end_coords = coords[match(edges$end, coords$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="")

    coords = coords[match(colnames(uncut),rownames(coords)),]
}

dev.off()

```


##Nematostella visualization
```{r}
#plot as histogram over pseudo time

set.seed(seed = 300)
meta.data = FetchData(uncut,c("pseudo_rank","time_point"))

tmp = as.data.frame(GetAssayData(uncut,"data"),stringsAsFactors = F)
tmp = as.data.frame(t(tmp))
tmp = tmp[,c("NVEC200-003903.1",
"NVEC200-006089.1",
"NVEC200-003854.1",
"NVEC200-008375.1",
"NVEC200-003672.1",
"NVEC200-006641.1",
"NVEC200-015111.1",
"NVEC200-005216.1",
"NVEC200-011036.1",
"NVEC200-014929.1", 
"NVEC200-006163.1",
"NVEC200-014405.1",
"NVEC200-009518.1",
"NVEC200-014970.1",
"NVEC200-012414.1",
"NVEC200-009923.1",
"NVEC200-011585.1")]

meta.data = cbind(meta.data,tmp,stringsAsFactors = F)

plot.df = melt(meta.data,id.vars = c("pseudo_rank","time_point"))

pdf("./uncut_Harmony_PT_HCRmarker_Histogram.pdf",width=15,height=7)
ggplot(plot.df,aes(x= pseudo_rank,y = value)) + geom_point()  + geom_smooth(se = F) + facet_wrap(~variable,scales = "free")
ggplot(plot.df,aes(x= pseudo_rank,y = value)) + geom_col() + facet_wrap(~variable,scales = "free") + geom_smooth(se = F,span = 0.8)
dev.off()


#plot virtual polyp

set.seed(seed = 300)
meta.data = FetchData(uncut,c("pseudo_rank","time_point"))

tmp = as.data.frame(GetAssayData(uncut,"scale.data"),stringsAsFactors = F)
tmp = as.data.frame(t(tmp))
tmp = tmp[,c("NVEC200-003903.1",
"NVEC200-006089.1",
"NVEC200-003854.1",
"NVEC200-008375.1",
"NVEC200-003672.1",
"NVEC200-006641.1",
"NVEC200-015111.1",
"NVEC200-005216.1",
"NVEC200-011036.1",
"NVEC200-014929.1", 
"NVEC200-006163.1",
"NVEC200-014405.1",
"NVEC200-009518.1",
"NVEC200-014970.1",
"NVEC200-012414.1",
"NVEC200-009923.1",
"NVEC200-011585.1")]

meta.data = cbind(meta.data,tmp,stringsAsFactors = F)


meta.data$jitter = runif(nrow(meta.data), 1, 2)
meta.data$jitter[meta.data$pseudo_rank %in% c(1:10)] = runif(10, 1.1, 1.9)
meta.data$jitter[meta.data$pseudo_rank %in% c(11:17)] = runif(7, 0.9, 2.1)
meta.data$jitter[meta.data$pseudo_rank %in% c(18:40)] = runif(23, 0.8, 2.2)
meta.data$jitter[meta.data$pseudo_rank %in% c(41:50)] = runif(10, 0.9, 2.1)
meta.data$jitter[meta.data$pseudo_rank %in% c(51:60)] = runif(10, 1.1, 1.9)

meta.data$jitter[meta.data$pseudo_rank %in% c(230:248)] = sample(c(runif(nrow(meta.data), 1, 1.3), runif(nrow(meta.data), 1.7, 2)), 19, replace=T) 
meta.data$jitter[meta.data$pseudo_rank %in% c(249:265)] = sample(c(runif(nrow(meta.data), 0.8, 1.1), runif(nrow(meta.data), 1.9, 2.2)), 17, replace=T) 
meta.data$jitter[meta.data$pseudo_rank %in% c(266:281)] = sample(c(runif(nrow(meta.data), 0.6, 0.9), runif(nrow(meta.data), 2.1, 2.4)), 16, replace=T) 
meta.data$jitter[meta.data$pseudo_rank %in% c(281:298)] = sample(c(runif(nrow(meta.data), 0.4, 0.7), runif(nrow(meta.data), 2.3, 2.6)), 18, replace=T) 
meta.data$jitter[meta.data$pseudo_rank %in% c(299:314)] = sample(c(runif(nrow(meta.data), 0.2, 0.5), runif(nrow(meta.data), 2.5, 2.8)), 16, replace=T) 
meta.data$jitter[meta.data$pseudo_rank %in% c(314:nrow(meta.data))] = sample(c(runif(nrow(meta.data), 0, 0.3), runif(nrow(meta.data), 2.7, 3)), 18, replace=T) 


meta.data$x1 = 1
meta.data$x1[meta.data$pseudo_rank %in% c(1:3)] = 1.1
meta.data$x1[meta.data$pseudo_rank %in% c(4:7)] = 1.05
meta.data$x1[meta.data$pseudo_rank %in% c(8:13)] = 0.95
meta.data$x1[meta.data$pseudo_rank %in% c(14:17)] = 0.9
meta.data$x1[meta.data$pseudo_rank %in% c(18:20)] = 0.85
meta.data$x1[meta.data$pseudo_rank %in% c(21:23)] = 0.825
meta.data$x1[meta.data$pseudo_rank %in% c(24:40)] = 0.8
meta.data$x1[meta.data$pseudo_rank %in% c(41:43)] = 0.825
meta.data$x1[meta.data$pseudo_rank %in% c(44:62)] = 0.85
meta.data$x1[meta.data$pseudo_rank %in% c(63:70)] = 0.875
meta.data$x1[meta.data$pseudo_rank %in% c(71:99)] = 0.9
meta.data$x1[meta.data$pseudo_rank %in% c(100:103)] = 0.925
meta.data$x1[meta.data$pseudo_rank %in% c(104:132)] = 0.95
meta.data$x1[meta.data$pseudo_rank %in% c(133:199)] = 0.975

meta.data$x1[meta.data$pseudo_rank %in% c(215:218)] = 1
meta.data$x1[meta.data$pseudo_rank %in% c(219:222)] =0.9
meta.data$x1[meta.data$pseudo_rank %in% c(223:225)] = 0.8
meta.data$x1[meta.data$pseudo_rank %in% c(226:232)] = 0.7
meta.data$x1[meta.data$pseudo_rank %in% c(233:240)] = 0.6
meta.data$x1[meta.data$pseudo_rank %in% c(241:248)] = 0.5
meta.data$x1[meta.data$pseudo_rank %in% c(249:257)] = 0.4
meta.data$x1[meta.data$pseudo_rank %in% c(258:265)] = 0.3
meta.data$x1[meta.data$pseudo_rank %in% c(266:273)] = 0.2
meta.data$x1[meta.data$pseudo_rank %in% c(274:281)] = 0.1
meta.data$x1[meta.data$pseudo_rank %in% c(282:290)] = 0
meta.data$x1[meta.data$pseudo_rank %in% c(291:325)] = -0.1
meta.data$x1[meta.data$pseudo_rank %in% c(326:328)] = -0.075
meta.data$x1[meta.data$pseudo_rank %in% c(329:nrow(meta.data))] = -0.05

meta.data$x2 = 2
meta.data$x2[meta.data$pseudo_rank %in% c(1:3)] = 1.9
meta.data$x2[meta.data$pseudo_rank %in% c(4:7)] = 1.95
meta.data$x2[meta.data$pseudo_rank %in% c(8:13)] = 2.05
meta.data$x2[meta.data$pseudo_rank %in% c(14:17)] = 2.1
meta.data$x2[meta.data$pseudo_rank %in% c(18:20)] = 2.15
meta.data$x2[meta.data$pseudo_rank %in% c(21:23)] = 2.175
meta.data$x2[meta.data$pseudo_rank %in% c(24:40)] = 2.2
meta.data$x2[meta.data$pseudo_rank %in% c(41:43)] = 2.175
meta.data$x2[meta.data$pseudo_rank %in% c(44:62)] = 2.15
meta.data$x2[meta.data$pseudo_rank %in% c(63:70)] = 2.125
meta.data$x2[meta.data$pseudo_rank %in% c(71:99)] = 2.1
meta.data$x2[meta.data$pseudo_rank %in% c(100:103)] = 2.075
meta.data$x2[meta.data$pseudo_rank %in% c(104:132)] = 2.05
meta.data$x2[meta.data$pseudo_rank %in% c(133:199)] = 2.025

meta.data$x2[meta.data$pseudo_rank %in% c(215:218)] = 1.3
meta.data$x2[meta.data$pseudo_rank %in% c(219:222)] = 1.2
meta.data$x2[meta.data$pseudo_rank %in% c(223:225)] = 1.1
meta.data$x2[meta.data$pseudo_rank %in% c(226:232)] = 1
meta.data$x2[meta.data$pseudo_rank %in% c(233:240)] = 0.9
meta.data$x2[meta.data$pseudo_rank %in% c(241:248)] = 0.8
meta.data$x2[meta.data$pseudo_rank %in% c(249:257)] = 0.7
meta.data$x2[meta.data$pseudo_rank %in% c(258:265)] = 0.6
meta.data$x2[meta.data$pseudo_rank %in% c(266:273)] = 0.5
meta.data$x2[meta.data$pseudo_rank %in% c(274:281)] = 0.4
meta.data$x2[meta.data$pseudo_rank %in% c(282:290)] = 0.3
meta.data$x2[meta.data$pseudo_rank %in% c(291:325)] = 0.2
meta.data$x2[meta.data$pseudo_rank %in% c(326:328)] = 0.175
meta.data$x2[meta.data$pseudo_rank %in% c(329:nrow(meta.data))] = 0.15

meta.data$x3 = 1
meta.data$x3[meta.data$pseudo_rank %in% c(1:3)] = 1.1
meta.data$x3[meta.data$pseudo_rank %in% c(4:7)] = 1.05
meta.data$x3[meta.data$pseudo_rank %in% c(8:13)] = 0.95
meta.data$x3[meta.data$pseudo_rank %in% c(14:17)] = 0.9
meta.data$x3[meta.data$pseudo_rank %in% c(18:20)] = 0.85
meta.data$x3[meta.data$pseudo_rank %in% c(21:23)] = 0.825
meta.data$x3[meta.data$pseudo_rank %in% c(24:40)] = 0.8
meta.data$x3[meta.data$pseudo_rank %in% c(41:43)] = 0.825
meta.data$x3[meta.data$pseudo_rank %in% c(44:62)] = 0.85
meta.data$x3[meta.data$pseudo_rank %in% c(63:70)] = 0.875
meta.data$x3[meta.data$pseudo_rank %in% c(71:99)] = 0.9
meta.data$x3[meta.data$pseudo_rank %in% c(100:103)] = 0.925
meta.data$x3[meta.data$pseudo_rank %in% c(104:132)] = 0.95
meta.data$x3[meta.data$pseudo_rank %in% c(133:199)] = 0.975

meta.data$x3[meta.data$pseudo_rank %in% c(215:218)] = 1.7
meta.data$x3[meta.data$pseudo_rank %in% c(219:222)] = 1.8
meta.data$x3[meta.data$pseudo_rank %in% c(223:225)] = 1.9
meta.data$x3[meta.data$pseudo_rank %in% c(226:232)] = 2
meta.data$x3[meta.data$pseudo_rank %in% c(233:240)] = 2.1
meta.data$x3[meta.data$pseudo_rank %in% c(241:248)] = 2.2
meta.data$x3[meta.data$pseudo_rank %in% c(249:257)] = 2.3
meta.data$x3[meta.data$pseudo_rank %in% c(258:265)] = 2.4
meta.data$x3[meta.data$pseudo_rank %in% c(266:273)] = 2.5
meta.data$x3[meta.data$pseudo_rank %in% c(274:281)] = 2.6
meta.data$x3[meta.data$pseudo_rank %in% c(282:290)] = 2.7
meta.data$x3[meta.data$pseudo_rank %in% c(291:325)] = 2.8
meta.data$x3[meta.data$pseudo_rank %in% c(326:328)] = 2.825
meta.data$x3[meta.data$pseudo_rank %in% c(329:nrow(meta.data))] = 2.85

meta.data$x4 = 2
meta.data$x4[meta.data$pseudo_rank %in% c(1:3)] = 1.9
meta.data$x4[meta.data$pseudo_rank %in% c(4:7)] = 1.95
meta.data$x4[meta.data$pseudo_rank %in% c(8:13)] = 2.05
meta.data$x4[meta.data$pseudo_rank %in% c(14:17)] = 2.1
meta.data$x4[meta.data$pseudo_rank %in% c(18:20)] = 2.15
meta.data$x4[meta.data$pseudo_rank %in% c(21:23)] = 2.175
meta.data$x4[meta.data$pseudo_rank %in% c(24:40)] = 2.2
meta.data$x4[meta.data$pseudo_rank %in% c(41:43)] = 2.175
meta.data$x4[meta.data$pseudo_rank %in% c(44:62)] = 2.15
meta.data$x4[meta.data$pseudo_rank %in% c(63:70)] = 2.125
meta.data$x4[meta.data$pseudo_rank %in% c(71:99)] = 2.1
meta.data$x4[meta.data$pseudo_rank %in% c(100:103)] = 2.075
meta.data$x4[meta.data$pseudo_rank %in% c(104:132)] = 2.05
meta.data$x4[meta.data$pseudo_rank %in% c(133:199)] = 2.025

meta.data$x4[meta.data$pseudo_rank %in% c(215:218)] = 2
meta.data$x4[meta.data$pseudo_rank %in% c(219:222)] = 2.1
meta.data$x4[meta.data$pseudo_rank %in% c(223:225)] = 2.2
meta.data$x4[meta.data$pseudo_rank %in% c(226:232)] = 2.3
meta.data$x4[meta.data$pseudo_rank %in% c(233:240)] = 2.4
meta.data$x4[meta.data$pseudo_rank %in% c(241:248)] = 2.5
meta.data$x4[meta.data$pseudo_rank %in% c(249:257)] = 2.6
meta.data$x4[meta.data$pseudo_rank %in% c(258:265)] = 2.7
meta.data$x4[meta.data$pseudo_rank %in% c(266:273)] = 2.8
meta.data$x4[meta.data$pseudo_rank %in% c(274:281)] = 2.9
meta.data$x4[meta.data$pseudo_rank %in% c(282:290)] = 3
meta.data$x4[meta.data$pseudo_rank %in% c(291:325)] = 3.1
meta.data$x4[meta.data$pseudo_rank %in% c(326:328)] = 3.075
meta.data$x4[meta.data$pseudo_rank %in% c(329:nrow(meta.data))] = 3.05


meta.data$x5 = 1
meta.data$x5[meta.data$pseudo_rank %in% c(1:3)] = 1.1
meta.data$x5[meta.data$pseudo_rank %in% c(4:7)] = 1.05
meta.data$x5[meta.data$pseudo_rank %in% c(8:13)] = 0.95
meta.data$x5[meta.data$pseudo_rank %in% c(14:17)] = 0.9
meta.data$x5[meta.data$pseudo_rank %in% c(18:20)] = 0.85
meta.data$x5[meta.data$pseudo_rank %in% c(21:23)] = 0.825
meta.data$x5[meta.data$pseudo_rank %in% c(24:40)] = 0.8
meta.data$x5[meta.data$pseudo_rank %in% c(41:43)] = 0.825
meta.data$x5[meta.data$pseudo_rank %in% c(44:62)] = 0.85
meta.data$x5[meta.data$pseudo_rank %in% c(63:70)] = 0.875
meta.data$x5[meta.data$pseudo_rank %in% c(71:99)] = 0.9
meta.data$x5[meta.data$pseudo_rank %in% c(100:103)] = 0.925
meta.data$x5[meta.data$pseudo_rank %in% c(104:132)] = 0.95
meta.data$x5[meta.data$pseudo_rank %in% c(133:199)] = 0.975

meta.data$x5[meta.data$pseudo_rank %in% c(215:218)] = 1.3
meta.data$x5[meta.data$pseudo_rank %in% c(219:222)] =1.25
meta.data$x5[meta.data$pseudo_rank %in% c(223:225)] = 1.2
meta.data$x5[meta.data$pseudo_rank %in% c(226:232)] = 1.15
meta.data$x5[meta.data$pseudo_rank %in% c(233:240)] = 1.1
meta.data$x5[meta.data$pseudo_rank %in% c(241:248)] = 1.05
meta.data$x5[meta.data$pseudo_rank %in% c(249:257)] = 1
meta.data$x5[meta.data$pseudo_rank %in% c(258:265)] = 0.95
meta.data$x5[meta.data$pseudo_rank %in% c(266:273)] = 0.9
meta.data$x5[meta.data$pseudo_rank %in% c(274:281)] = 0.85
meta.data$x5[meta.data$pseudo_rank %in% c(282:290)] = 0.8
meta.data$x5[meta.data$pseudo_rank %in% c(291:325)] = 0.75
meta.data$x5[meta.data$pseudo_rank %in% c(326:328)] = 0.775
meta.data$x5[meta.data$pseudo_rank %in% c(329:nrow(meta.data))] = 0.8

meta.data$x6 = 2
meta.data$x6[meta.data$pseudo_rank %in% c(1:3)] = 1.9
meta.data$x6[meta.data$pseudo_rank %in% c(4:7)] = 1.95
meta.data$x6[meta.data$pseudo_rank %in% c(8:13)] = 2.05
meta.data$x6[meta.data$pseudo_rank %in% c(14:17)] = 2.1
meta.data$x6[meta.data$pseudo_rank %in% c(18:20)] = 2.15
meta.data$x6[meta.data$pseudo_rank %in% c(21:23)] = 2.175
meta.data$x6[meta.data$pseudo_rank %in% c(24:40)] = 2.2
meta.data$x6[meta.data$pseudo_rank %in% c(41:43)] = 2.175
meta.data$x6[meta.data$pseudo_rank %in% c(44:62)] = 2.15
meta.data$x6[meta.data$pseudo_rank %in% c(63:70)] = 2.125
meta.data$x6[meta.data$pseudo_rank %in% c(71:99)] = 2.1
meta.data$x6[meta.data$pseudo_rank %in% c(100:103)] = 2.075
meta.data$x6[meta.data$pseudo_rank %in% c(104:132)] = 2.05
meta.data$x6[meta.data$pseudo_rank %in% c(133:199)] = 2.025

meta.data$x6[meta.data$pseudo_rank %in% c(215:218)] = 1.6
meta.data$x6[meta.data$pseudo_rank %in% c(219:222)] = 1.55
meta.data$x6[meta.data$pseudo_rank %in% c(223:225)] = 1.5
meta.data$x6[meta.data$pseudo_rank %in% c(226:232)] = 1.45
meta.data$x6[meta.data$pseudo_rank %in% c(233:240)] = 1.4
meta.data$x6[meta.data$pseudo_rank %in% c(241:248)] = 1.35
meta.data$x6[meta.data$pseudo_rank %in% c(249:257)] = 1.3
meta.data$x6[meta.data$pseudo_rank %in% c(258:265)] = 1.25
meta.data$x6[meta.data$pseudo_rank %in% c(266:273)] = 1.2
meta.data$x6[meta.data$pseudo_rank %in% c(274:281)] = 1.15
meta.data$x6[meta.data$pseudo_rank %in% c(282:290)] = 1.1
meta.data$x6[meta.data$pseudo_rank %in% c(291:325)] = 1.05
meta.data$x6[meta.data$pseudo_rank %in% c(326:328)] = 1.025
meta.data$x6[meta.data$pseudo_rank %in% c(329:nrow(meta.data))] = 1


meta.data$x7 = 1
meta.data$x7[meta.data$pseudo_rank %in% c(1:3)] = 1.1
meta.data$x7[meta.data$pseudo_rank %in% c(4:7)] = 1.05
meta.data$x7[meta.data$pseudo_rank %in% c(8:13)] = 0.95
meta.data$x7[meta.data$pseudo_rank %in% c(14:17)] = 0.9
meta.data$x7[meta.data$pseudo_rank %in% c(18:20)] = 0.85
meta.data$x7[meta.data$pseudo_rank %in% c(21:23)] = 0.825
meta.data$x7[meta.data$pseudo_rank %in% c(24:40)] = 0.8
meta.data$x7[meta.data$pseudo_rank %in% c(41:43)] = 0.825
meta.data$x7[meta.data$pseudo_rank %in% c(44:62)] = 0.85
meta.data$x7[meta.data$pseudo_rank %in% c(63:70)] = 0.875
meta.data$x7[meta.data$pseudo_rank %in% c(71:99)] = 0.9
meta.data$x7[meta.data$pseudo_rank %in% c(100:103)] = 0.925
meta.data$x7[meta.data$pseudo_rank %in% c(104:132)] = 0.95
meta.data$x7[meta.data$pseudo_rank %in% c(133:199)] = 0.975

meta.data$x7[meta.data$pseudo_rank %in% c(215:218)] = 1.4
meta.data$x7[meta.data$pseudo_rank %in% c(219:222)] = 1.45
meta.data$x7[meta.data$pseudo_rank %in% c(223:225)] = 1.5
meta.data$x7[meta.data$pseudo_rank %in% c(226:232)] = 1.55
meta.data$x7[meta.data$pseudo_rank %in% c(233:240)] = 1.6
meta.data$x7[meta.data$pseudo_rank %in% c(241:248)] = 1.65
meta.data$x7[meta.data$pseudo_rank %in% c(249:257)] = 1.7
meta.data$x7[meta.data$pseudo_rank %in% c(258:265)] = 1.75
meta.data$x7[meta.data$pseudo_rank %in% c(266:273)] = 1.8
meta.data$x7[meta.data$pseudo_rank %in% c(274:281)] = 1.85
meta.data$x7[meta.data$pseudo_rank %in% c(282:290)] = 1.9
meta.data$x7[meta.data$pseudo_rank %in% c(291:325)] = 1.95
meta.data$x7[meta.data$pseudo_rank %in% c(326:328)] = 1.975
meta.data$x7[meta.data$pseudo_rank %in% c(329:nrow(meta.data))] = 2

meta.data$x8 = 2
meta.data$x8[meta.data$pseudo_rank %in% c(1:3)] = 1.9
meta.data$x8[meta.data$pseudo_rank %in% c(4:7)] = 1.95
meta.data$x8[meta.data$pseudo_rank %in% c(8:13)] = 2.05
meta.data$x8[meta.data$pseudo_rank %in% c(14:17)] = 2.1
meta.data$x8[meta.data$pseudo_rank %in% c(18:20)] = 2.15
meta.data$x8[meta.data$pseudo_rank %in% c(21:23)] = 2.175
meta.data$x8[meta.data$pseudo_rank %in% c(24:40)] = 2.2
meta.data$x8[meta.data$pseudo_rank %in% c(41:43)] = 2.175
meta.data$x8[meta.data$pseudo_rank %in% c(44:62)] = 2.15
meta.data$x8[meta.data$pseudo_rank %in% c(63:70)] = 2.125
meta.data$x8[meta.data$pseudo_rank %in% c(71:99)] = 2.1
meta.data$x8[meta.data$pseudo_rank %in% c(100:103)] = 2.075
meta.data$x8[meta.data$pseudo_rank %in% c(104:132)] = 2.05
meta.data$x8[meta.data$pseudo_rank %in% c(133:199)] = 2.025

meta.data$x8[meta.data$pseudo_rank %in% c(215:218)] = 1.7
meta.data$x8[meta.data$pseudo_rank %in% c(219:222)] = 1.75
meta.data$x8[meta.data$pseudo_rank %in% c(223:225)] = 1.8
meta.data$x8[meta.data$pseudo_rank %in% c(226:232)] = 1.85
meta.data$x8[meta.data$pseudo_rank %in% c(233:240)] = 1.9
meta.data$x8[meta.data$pseudo_rank %in% c(241:248)] = 1.95
meta.data$x8[meta.data$pseudo_rank %in% c(249:257)] = 2
meta.data$x8[meta.data$pseudo_rank %in% c(258:265)] = 2.05
meta.data$x8[meta.data$pseudo_rank %in% c(266:273)] = 2.1
meta.data$x8[meta.data$pseudo_rank %in% c(274:281)] = 2.15
meta.data$x8[meta.data$pseudo_rank %in% c(282:290)] = 2.2
meta.data$x8[meta.data$pseudo_rank %in% c(291:325)] = 2.25
meta.data$x8[meta.data$pseudo_rank %in% c(326:328)] = 2.225
meta.data$x8[meta.data$pseudo_rank %in% c(329:nrow(meta.data))] = 2.2




a = meta.data[,c("pseudo_rank","time_point","x1","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")]
#"NVEC200-009271.1","NVEC200-014866.1","NVEC200-013033.1","NVEC200-006430.1","NVEC200-006423.1","NVEC200-010166.1","NVEC200-001639.1","NVEC200-000819.1","NVEC200-014003.1","NVEC200-003858.1","NVEC200-015133.1","NVEC200-002032.1","NVEC200-014221.1","NVEC200-005215.1","NVEC200-011057.1","NVEC200-009256.1","NVEC200-000547.1","NVEC200-005257.1","NVEC200-014418.1","NVEC200-003926.1","NVEC200-000574.1","NVEC200-006121.1","NVEC200-004712.1","NVEC200-001561.1","NVEC200-007938.1","NVEC200-012562.1","NVEC200-006310.1","NVEC200-015153.1","NVEC200-000708.1","NVEC200-000772.1"

colnames(a) = c("pseudo_rank","time_point","x","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")
a$rank = a$pseudo_rank
a$rank[a$pseudo_rank %in% c(200:nrow(a))] = paste("left",a$pseudo_rank,sep = "_")

b = meta.data[,c("pseudo_rank","time_point","x2","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")]
#"NVEC200-009271.1","NVEC200-014866.1","NVEC200-013033.1","NVEC200-006430.1","NVEC200-006423.1","NVEC200-010166.1","NVEC200-001639.1","NVEC200-000819.1","NVEC200-014003.1","NVEC200-003858.1","NVEC200-015133.1","NVEC200-002032.1","NVEC200-014221.1","NVEC200-005215.1","NVEC200-011057.1","NVEC200-009256.1","NVEC200-000547.1","NVEC200-005257.1","NVEC200-014418.1","NVEC200-003926.1","NVEC200-000574.1","NVEC200-006121.1","NVEC200-004712.1","NVEC200-001561.1","NVEC200-007938.1","NVEC200-012562.1","NVEC200-006310.1","NVEC200-015153.1","NVEC200-000708.1","NVEC200-000772.1"

colnames(b) = c("pseudo_rank","time_point","x","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")
b$rank = b$pseudo_rank
b$rank[b$pseudo_rank %in% c(200:nrow(b))] = paste("left",b$pseudo_rank,sep = "_")


c = meta.data[,c("pseudo_rank","time_point","x3","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")]
#"NVEC200-009271.1","NVEC200-014866.1","NVEC200-013033.1","NVEC200-006430.1","NVEC200-006423.1","NVEC200-010166.1","NVEC200-001639.1","NVEC200-000819.1","NVEC200-014003.1","NVEC200-003858.1","NVEC200-015133.1","NVEC200-002032.1","NVEC200-014221.1","NVEC200-005215.1","NVEC200-011057.1","NVEC200-009256.1","NVEC200-000547.1","NVEC200-005257.1","NVEC200-014418.1","NVEC200-003926.1","NVEC200-000574.1","NVEC200-006121.1","NVEC200-004712.1","NVEC200-001561.1","NVEC200-007938.1","NVEC200-012562.1","NVEC200-006310.1","NVEC200-015153.1","NVEC200-000708.1","NVEC200-000772.1"
colnames(c) = c("pseudo_rank","time_point","x","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")
c$rank = c$pseudo_rank
c$rank[c$pseudo_rank %in% c(200:nrow(c))] = paste("right",c$pseudo_rank,sep = "_")

d = meta.data[,c("pseudo_rank","time_point","x4","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")]
#"NVEC200-009271.1","NVEC200-014866.1","NVEC200-013033.1","NVEC200-006430.1","NVEC200-006423.1","NVEC200-010166.1","NVEC200-001639.1","NVEC200-000819.1","NVEC200-014003.1","NVEC200-003858.1","NVEC200-015133.1","NVEC200-002032.1","NVEC200-014221.1","NVEC200-005215.1","NVEC200-011057.1","NVEC200-009256.1","NVEC200-000547.1","NVEC200-005257.1","NVEC200-014418.1","NVEC200-003926.1","NVEC200-000574.1","NVEC200-006121.1","NVEC200-004712.1","NVEC200-001561.1","NVEC200-007938.1","NVEC200-012562.1","NVEC200-006310.1","NVEC200-015153.1","NVEC200-000708.1","NVEC200-000772.1"
colnames(d) = c("pseudo_rank","time_point","x","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")
d$rank = d$pseudo_rank
d$rank[d$pseudo_rank %in% c(200:nrow(d))] = paste("right",d$pseudo_rank,sep = "_")



e = meta.data[,c("pseudo_rank","time_point","x5","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")]
#"NVEC200-009271.1","NVEC200-014866.1","NVEC200-013033.1","NVEC200-006430.1","NVEC200-006423.1","NVEC200-010166.1","NVEC200-001639.1","NVEC200-000819.1","NVEC200-014003.1","NVEC200-003858.1","NVEC200-015133.1","NVEC200-002032.1","NVEC200-014221.1","NVEC200-005215.1","NVEC200-011057.1","NVEC200-009256.1","NVEC200-000547.1","NVEC200-005257.1","NVEC200-014418.1","NVEC200-003926.1","NVEC200-000574.1","NVEC200-006121.1","NVEC200-004712.1","NVEC200-001561.1","NVEC200-007938.1","NVEC200-012562.1","NVEC200-006310.1","NVEC200-015153.1","NVEC200-000708.1","NVEC200-000772.1"

colnames(e) = c("pseudo_rank","time_point","x","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")
e$rank = e$pseudo_rank
e$rank[e$pseudo_rank %in% c(200:nrow(e))] = paste("leftcenter",e$pseudo_rank,sep = "_")

f = meta.data[,c("pseudo_rank","time_point","x6","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")]
#"NVEC200-009271.1","NVEC200-014866.1","NVEC200-013033.1","NVEC200-006430.1","NVEC200-006423.1","NVEC200-010166.1","NVEC200-001639.1","NVEC200-000819.1","NVEC200-014003.1","NVEC200-003858.1","NVEC200-015133.1","NVEC200-002032.1","NVEC200-014221.1","NVEC200-005215.1","NVEC200-011057.1","NVEC200-009256.1","NVEC200-000547.1","NVEC200-005257.1","NVEC200-014418.1","NVEC200-003926.1","NVEC200-000574.1","NVEC200-006121.1","NVEC200-004712.1","NVEC200-001561.1","NVEC200-007938.1","NVEC200-012562.1","NVEC200-006310.1","NVEC200-015153.1","NVEC200-000708.1","NVEC200-000772.1"

colnames(f) = c("pseudo_rank","time_point","x","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")
f$rank = f$pseudo_rank
f$rank[f$pseudo_rank %in% c(200:nrow(f))] = paste("leftcenter",f$pseudo_rank,sep = "_")


g = meta.data[,c("pseudo_rank","time_point","x7","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")]
#"NVEC200-009271.1","NVEC200-014866.1","NVEC200-013033.1","NVEC200-006430.1","NVEC200-006423.1","NVEC200-010166.1","NVEC200-001639.1","NVEC200-000819.1","NVEC200-014003.1","NVEC200-003858.1","NVEC200-015133.1","NVEC200-002032.1","NVEC200-014221.1","NVEC200-005215.1","NVEC200-011057.1","NVEC200-009256.1","NVEC200-000547.1","NVEC200-005257.1","NVEC200-014418.1","NVEC200-003926.1","NVEC200-000574.1","NVEC200-006121.1","NVEC200-004712.1","NVEC200-001561.1","NVEC200-007938.1","NVEC200-012562.1","NVEC200-006310.1","NVEC200-015153.1","NVEC200-000708.1","NVEC200-000772.1"
colnames(g) = c("pseudo_rank","time_point","x","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")
g$rank = g$pseudo_rank
g$rank[g$pseudo_rank %in% c(200:nrow(g))] = paste("rightcenter",g$pseudo_rank,sep = "_")

h = meta.data[,c("pseudo_rank","time_point","x8","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")]
#"NVEC200-009271.1","NVEC200-014866.1","NVEC200-013033.1","NVEC200-006430.1","NVEC200-006423.1","NVEC200-010166.1","NVEC200-001639.1","NVEC200-000819.1","NVEC200-014003.1","NVEC200-003858.1","NVEC200-015133.1","NVEC200-002032.1","NVEC200-014221.1","NVEC200-005215.1","NVEC200-011057.1","NVEC200-009256.1","NVEC200-000547.1","NVEC200-005257.1","NVEC200-014418.1","NVEC200-003926.1","NVEC200-000574.1","NVEC200-006121.1","NVEC200-004712.1","NVEC200-001561.1","NVEC200-007938.1","NVEC200-012562.1","NVEC200-006310.1","NVEC200-015153.1","NVEC200-000708.1","NVEC200-000772.1"
colnames(h) = c("pseudo_rank","time_point","x","NVEC200-003903.1","NVEC200-006089.1","NVEC200-003854.1","NVEC200-008375.1","NVEC200-003672.1","NVEC200-006641.1","NVEC200-015111.1","NVEC200-005216.1","NVEC200-011036.1","NVEC200-014929.1","NVEC200-006163.1","NVEC200-014405.1","NVEC200-009518.1","NVEC200-014970.1","NVEC200-012414.1","NVEC200-009923.1","NVEC200-011585.1")
h$rank = h$pseudo_rank
h$rank[h$pseudo_rank %in% c(200:nrow(h))] = paste("rightcenter",h$pseudo_rank,sep = "_")

df = rbind(a,b,c,d,e,f,g,h)



pdf("./uncut_PT_Jitter_ClusterMarker_Shape.pdf",width=4,height=8)
ggplot(meta.data, aes(y = pseudo_rank,x = jitter, color = time_point)) +theme_bw() + geom_point(aes(color = meta.data[,"NVEC200-003903.1"]))  + ylab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + scale_colour_gradientn(colors = viridis_pal(option = "inferno")(12))

ggplot(df, aes(x = x, y = pseudo_rank)) + 
  geom_line(aes(group = rank,color = df[,"NVEC200-003903.1"]),size=0.25) + scale_colour_gradientn(colors = viridis_pal(option = "inferno")(12))

dev.off()


plot_list=list()

plot_list <- lapply(colnames(meta.data[,3:(ncol(meta.data)-1)]), function(i) { 
  p <- ggplot(meta.data, aes(y = pseudo_rank,x = jitter)) +theme_bw() + geom_point(aes(color = meta.data[,i]),cex = 1)  + ylab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.position = "none", axis.line = element_line(colour = "black"))  + scale_colour_gradientn(colors = viridis_pal(option = "inferno")(12))
})

library(gridExtra)

pdf("./uncut_PT_Jitter_ClusterMarker_Shape.pdf",width=10,height=20)

ggplot(meta.data, aes(y = pseudo_rank,x = jitter, color = time_point)) +theme_bw() + geom_point(cex = 5)  + ylab("PT")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) 

do.call("grid.arrange",plot_list)
 
dev.off()



plot_list=list()

plot_list <- lapply(colnames(df[,4:(ncol(df)-1)]), function(i) { 
  p <- ggplot(df, aes(x = x, y = pseudo_rank)) + 
  geom_line(aes(group = rank,color = df[,i]),size=0.25) + scale_colour_gradientn(colors = c(rep("darkblue",10),rev(viridis_pal(option = "plasma")(12)))) + ylab("PT") +ggtitle(i) + theme_bw() + theme(axis.line=element_blank(),   axis.text.x=element_blank(), axis.text.y=element_blank(), axis.ticks=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), legend.position="none", panel.background=element_blank(), panel.border=element_blank(), panel.grid.major=element_blank(),
      panel.grid.minor=element_blank(), plot.background=element_blank())
  })

  
library(gridExtra)

pdf("./uncut_PT_Lines_ClusterMarker_Shape.pdf",width=7,height=15)
do.call("grid.arrange",plot_list)
 
dev.off()

######make Nematostella shape with illustrator
#only plot a heatmap with lines for expression

uncut = readRDS("./uncut_Harmony.RDS")

set.seed(seed = 300)
meta.data = FetchData(uncut,c("pseudo_rank"))

tmp = as.data.frame(GetAssayData(uncut,"data"),stringsAsFactors = F)
tmp = as.data.frame(t(tmp))
tmp = tmp[,c("NVEC200-001043.1",
"NVEC200-000874.1",
"NVEC200-000872.1",
"NVEC200-010351.1",
"NVEC200-010468.1",
"NVEC200-006988.1",
"NVEC200-001297.1",
"NVEC200-007533.1",
"NVEC200-014111.1",
"NVEC200-006032.1",
"NVEC200-006030.1",
"NVEC200-006642.1",
"NVEC200-006028.1",
"NVEC200-003168.1",
"NVEC200-012164.1",
"NVEC200-009995.1",
"NVEC200-011699.1",
"NVEC200-000491.1",
"NVEC200-002987.1",
"NVEC200-000883.1",
"NVEC200-006580.1",
"NVEC200-005938.1",
"NVEC200-001459.1",
"NVEC200-001458.1",
"NVEC200-009287.1",
"NVEC200-008102.1",
"NVEC200-015113.1",
"NVEC200-015112.1",
"NVEC200-012338.1",
"NVEC200-001364.1",
"NVEC200-001363.1",
"NVEC200-001362.1",
"NVEC200-013954.1",
"NVEC200-011655.1",
"NVEC200-008902.1",
"NVEC200-009545.1",
"NVEC200-003200.1",
"NVEC200-009182.1",
"NVEC200-006274.1",
"NVEC200-001846.1",
"NVEC200-013190.1",
"NVEC200-004425.1",
"NVEC200-006033.1",
"NVEC200-013831.1",
"NVEC200-014326.1",
"NVEC200-008616.1",
"NVEC200-008615.1",
"NVEC200-001014.1",
"NVEC200-000260.1",
"NVEC200-009768.1",
"NVEC200-008308.1",
"NVEC200-000261.1",
"NVEC200-010238.1",
"NVEC200-008832.1",
"NVEC200-005807.1",
"NVEC200-005805.1",
"NVEC200-005804.1",
"NVEC200-001872.1",
"NVEC200-011709.1",
"NVEC200-008022.1",
"NVEC200-012414.1")]

meta.data = cbind(meta.data,tmp,stringsAsFactors = F)

plot.df = melt(meta.data,id.vars = c("pseudo_rank"))

pdf("./uncut_Harmony_PT_InSituMarker_Shape.pdf",width=10,height=15)
ggplot(plot.df,aes(x = pseudo_rank, y = 1, color = value , fill = value )) + geom_col()  + scale_colour_gradientn(colors = c(rep("darkblue",1),(viridis_pal(option = "plasma")(12))))  + scale_fill_gradientn(colors = c(rep("darkblue",1),(viridis_pal(option = "plasma")(12))))   + facet_wrap(variable~. ) + theme_bw() + theme(axis.line=element_blank(),   axis.text.x=element_blank(), axis.text.y=element_blank(), axis.ticks=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), legend.position="none", panel.background=element_blank(), panel.border=element_blank(), panel.grid.major=element_blank(),
      panel.grid.minor=element_blank(), plot.background=element_blank())

dev.off()


pdf("./uncut_Harmony_PT_InSituMarker_Shape_individual.pdf",width=10,height=15)

for (i in c("NVEC200-010351.1",
"NVEC200-010468.1",
"NVEC200-001364.1",
"NVEC200-001363.1",
"NVEC200-001362.1",
"NVEC200-011709.1",
"NVEC200-008308.1",
"NVEC200-000872.1",
"NVEC200-015113.1",
"NVEC200-012414.1")) {
  
  print(ggplot(meta.data,aes(x = pseudo_rank, y = 1, color = meta.data[,i] , fill = meta.data[,i]  )) + geom_col()  + scale_colour_gradientn(colors = c(rep("darkblue",1),(viridis_pal(option = "plasma")(12))))  + scale_fill_gradientn(colors = c(rep("darkblue",1),(viridis_pal(option = "plasma")(12))))  + theme_bw() + ggtitle(i) + theme(axis.line=element_blank(),   axis.text.x=element_blank(), axis.text.y=element_blank(), axis.ticks=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), legend.position="none", panel.background=element_blank(), panel.border=element_blank(), panel.grid.major=element_blank(),
      panel.grid.minor=element_blank(), plot.background=element_blank()))

  
}
dev.off()

ggplot(plot.df,aes(x = pseudo_rank, y = 1, color = value , fill = value )) + geom_col()  + scale_colour_gradientn(colors = c(rep("darkblue",1),(viridis_pal(option = "plasma")(12))))  + scale_fill_gradientn(colors = c(rep("darkblue",1),(viridis_pal(option = "plasma")(12))))  + theme_bw() + theme(axis.line=element_blank(),   axis.text.x=element_blank(), axis.text.y=element_blank(), axis.ticks=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), legend.position="none", panel.background=element_blank(), panel.border=element_blank(), panel.grid.major=element_blank(),
      panel.grid.minor=element_blank(), plot.background=element_blank())





ggplot(df, aes(x = x, y = pseudo_rank)) + geom_line(aes(group = rank,color = df[,i]),size=0.25) + scale_colour_gradientn(colors = c(rep("darkblue",10),rev(viridis_pal(option = "plasma")(12)))) + ylab("PT") +ggtitle(i) + theme_bw() + theme(axis.line=element_blank(),   axis.text.x=element_blank(), axis.text.y=element_blank(), axis.ticks=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), legend.position="none", panel.background=element_blank(), panel.border=element_blank(), panel.grid.major=element_blank(),
      panel.grid.minor=element_blank(), plot.background=element_blank())
dev.off()


```

#Plot ECM remodelling

```{r}

dataVector = read.csv("230328_ColIV_Regen.csv")

pdf("./ECM_remod_ColIV_Velocity.pdf",width=12,height=10)
ggplot(dataVector %>% filter(Timepoint == "t0", Condition == "Cut"), aes(x = Position, y = Velocity))+
    theme_linedraw()+
    xlim(1,0)+
    ylim(-0.1,0.1)+
    geom_hline(yintercept=0, linetype="dashed", color = "black", linewidth = 1.3)+
    xlab("Normalized distance from oral pole")+
    ylab("Relative velocity")+
    geom_point(size = 5)+
    geom_smooth(method = lm, span = 0.5)+
    theme(legend.position = c(0.08, 0.88))+
    theme(legend.background = element_rect(fill="white",
                                           size=0.5, linetype="solid", 
                                           colour ="black"))+
    theme(text = element_text(size=25))

ggplot(dataVector %>% filter(Timepoint == "t0", Condition == "Uncut"), aes(x = Position, y = Velocity))+
    theme_linedraw()+
    xlim(1,0)+
    ylim(-0.1,0.1)+
    geom_hline(yintercept=0, linetype="dashed", color = "black", linewidth = 1.3)+
    xlab("Normalized distance from oral pole")+
    ylab("Relative velocity")+
    geom_point(size = 5)+
    geom_smooth(method = lm, span = 0.5)+
    theme(legend.position = c(0.08, 0.88))+
    theme(legend.background = element_rect(fill="white",
                                           size=0.5, linetype="solid", 
                                           colour ="black"))+
    theme(text = element_text(size=25))

dev.off()



dataVector = read.csv("2303230_representatives_uncut+cut.csv")

pdf("./ECM_remod_ColIV_Location.pdf",width=12,height=10)
ggplot(dataVector %>% filter(Animal == "Animal27"), aes(x = Relative_Distance, y = smoothed_norm_intensity, color = Timepoint))+
    xlab("Normalized distance from oral pole")+
    ylab("Intensity")+
    geom_line(size = 2)+
    theme(legend.background = element_rect(fill="white",
                                           size=0.5, linetype="solid", 
                                           colour ="black"))+
    theme(text = element_text(size=25)) +
    theme_bw()

ggplot(dataVector %>% filter(Animal == "Animal16"), aes(x = Relative_Distance, y = smoothed_norm_intensity, color = Timepoint))+
    xlab("Normalized distance from oral pole")+
    ylab("Intensity")+
    geom_line(size=2 )+
    theme(legend.background = element_rect(fill="white",
                                           size=0.5, linetype="solid", 
                                           colour ="black"))+
    theme(text = element_text(size=25)) +
    theme_bw()

dev.off()

```


#Soham GOseq

```{r}
gtf = read.delim("/g/ikmi/Collaborations/EMBL/Stegle_lab/Soham/tcs_v2.20211130.versioned.eggnog.gtf",header = F, sep = "\t")
gtf = gtf[gtf$V3 == "transcript",]
gtf$length = gtf$V5 - gtf$V4
#extract IDs
gtf$gene = sapply(strsplit(gtf$V9,split = ";"),"[[",1)
gtf$gene = sapply(strsplit(gtf$gene,split = " "),"[[",2)
#filter out none GO entries
gtf = gtf[gtf$V9 %in% gtf$V9[grep("GOs",gtf$V9)],]
#extract GOs
gtf$GOs = sapply(strsplit(gtf$V9,split = "GOs"),"[[",2)
gtf$GOs = sapply(strsplit(gtf$GOs,split = ";"),"[[",1)

#gtf$gene = gsub("NVEC200_","NVEC200-",gtf$gene)
#gtf$gene = gsub("_1.1",".1",gtf$gene)

length.df = gtf[,c("gene","length")]

#link many GOs to one gene id on separate rows
#runs long load instead the csv file I saved

#GO.list.mod = data.frame(gene = NA,GOs = NA)
#count = 0
#for (i in 1:nrow(gtf)) {
#  count = count + 1
#  print(count)
#  a = unique(strsplit(gtf[i,"GOs"], "\\,")[[1]])
#  if (length(a) > 1) {
#    for(j in 1:length(a)) {
#      GO.list.mod = rbind(GO.list.mod,c(gtf[i,"gene"],a[j]),stringsAsFactors = F)
#    }
#    
#  } else {
#    GO.list.mod = rbind(GO.list.mod,gtf[i,c("gene","GOs")],stringsAsFactors = F)
#  }
#} 
#GO.list.mod = na.omit(GO.list.mod)
#write.csv(GO.list.mod,"GO.Soham.singleLines.csv")

GO.list.mod = read.csv("./xxxx/GO.Soham.singleLines.csv")


#link GO terms with IDs

Soham.marker = as.data.frame(read_delim(delim = "\t","/g/arendt/gerber/Analyses/NematostellaTomoSeq/2023-09-12_allgenes_PosvsNeg.txt"))


#Prepare for Goseq

assayed.genes = unique(GO.list.mod$gene)

gene.vector=as.integer(assayed.genes %in% Soham.marker$row[Soham.marker$log2FoldChange>1]) #change this value to select different thresholds for your input genes

names(gene.vector)=assayed.genes
head(gene.vector)

gene.lengths <- length.df$length
names(gene.lengths) <- length.df$gene

# subset by assayed genes and genes with GO.
gene.lengths <- gene.lengths[assayed.genes]
gene.lengths <- gene.lengths[which(names(gene.lengths) %in% unique(GO.list.mod$gene))]


if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("goseq")
library(goseq)

pwf <- nullp(gene.vector, bias.data=gene.lengths, plot.fit = F)

go <- goseq(pwf, gene2cat = GO.list.mod)

enriched_go <- go %>% filter(over_represented_pvalue < 0.001)

write.csv(file = "./GO.enrichement.Soham.csv",enriched_go[,c("category","over_represented_pvalue","numDEInCat")],row.names = F)

```




Emapper run info
Wed Dec 22 13:36:47 2021
emapper-2.1.6
/data/shared/home/emapper/miniconda3/envs/eggnog-mapper-2.1/bin/emapper.py --cpu 20 --mp_start_method forkserver --data_dir /dev/shm/ -o out --output_dir /emapper_web_jobs/emapper_jobs/user_data/MM_9krs15um --temp_dir /emapper_web_jobs/emapper_jobs/user_data/MM_9krs15um --override -m diamond --dmnd_ignore_warnings -i /emapper_web_jobs/emapper_jobs/user_data/MM_9krs15um/queries.fasta --evalue 0.01 --score 40 --pident 20 --query_cover 20 --subject_cover 10 --itype CDS --translate --tax_scope auto --target_orthologs all --go_evidence non-electronic --pfam_realign none --report_orthologs --decorate_gff yes --excel
